// @alicloud/dingtalk@2.1.16/dist/rooms_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryMeetingRoomResponseBody = exports.QueryMeetingRoomRequest = exports.QueryMeetingRoomHeaders = exports.QueryDevicePropertiesResponse = exports.QueryDevicePropertiesResponseBody = exports.QueryDevicePropertiesRequest = exports.QueryDevicePropertiesHeaders = exports.QueryDeviceIpByCodeResponse = exports.QueryDeviceIpByCodeResponseBody = exports.QueryDeviceIpByCodeRequest = exports.QueryDeviceIpByCodeHeaders = exports.QueryDeviceCustomTemplateListResponse = exports.QueryDeviceCustomTemplateListResponseBody = exports.QueryDeviceCustomTemplateListHeaders = exports.QueryDeviceCustomTemplateResponse = exports.QueryDeviceCustomTemplateResponseBody = exports.QueryDeviceCustomTemplateHeaders = exports.DeleteMeetingRoomGroupResponse = exports.DeleteMeetingRoomGroupResponseBody = exports.DeleteMeetingRoomGroupRequest = exports.DeleteMeetingRoomGroupHeaders = exports.DeleteMeetingRoomControlPanelResponse = exports.DeleteMeetingRoomControlPanelResponseBody = exports.DeleteMeetingRoomControlPanelShrinkRequest = exports.DeleteMeetingRoomControlPanelRequest = exports.DeleteMeetingRoomControlPanelHeaders = exports.DeleteMeetingRoomResponse = exports.DeleteMeetingRoomResponseBody = exports.DeleteMeetingRoomRequest = exports.DeleteMeetingRoomHeaders = exports.DeleteDeviceCustomTemplateResponse = exports.DeleteDeviceCustomTemplateResponseBody = exports.DeleteDeviceCustomTemplateRequest = exports.DeleteDeviceCustomTemplateHeaders = exports.CreateMeetingRoomGroupResponse = exports.CreateMeetingRoomGroupResponseBody = exports.CreateMeetingRoomGroupRequest = exports.CreateMeetingRoomGroupHeaders = exports.CreateMeetingRoomControlPanelResponse = exports.CreateMeetingRoomControlPanelResponseBody = exports.CreateMeetingRoomControlPanelRequest = exports.CreateMeetingRoomControlPanelHeaders = exports.CreateMeetingRoomResponse = exports.CreateMeetingRoomResponseBody = exports.CreateMeetingRoomRequest = exports.CreateMeetingRoomHeaders = exports.CreateDeviceCustomTemplateResponse = exports.CreateDeviceCustomTemplateResponseBody = exports.CreateDeviceCustomTemplateRequest = exports.CreateDeviceCustomTemplateHeaders = void 0;
exports.QueryDeviceCustomTemplateListResponseBodyResultDeviceCustomTemplates = exports.QueryDeviceCustomTemplateResponseBodyResult = exports.QueryDeviceCustomTemplateResponseBodyResultDeviceCustomTemplate = exports.DeleteMeetingRoomControlPanelRequestBody = exports.CreateMeetingRoomControlPanelRequestRoomConfig = exports.CreateMeetingRoomControlPanelRequestExtra = exports.CreateMeetingRoomRequestRoomLocation = exports.CreateMeetingRoomRequestReservationAuthority = exports.CreateMeetingRoomRequestReservationAuthorityAuthorizedMembers = exports.UpdateMeetingRoomGroupResponse = exports.UpdateMeetingRoomGroupResponseBody = exports.UpdateMeetingRoomGroupRequest = exports.UpdateMeetingRoomGroupHeaders = exports.UpdateMeetingRoomResponse = exports.UpdateMeetingRoomResponseBody = exports.UpdateMeetingRoomRequest = exports.UpdateMeetingRoomHeaders = exports.UpdateDeviceCustomTemplateResponse = exports.UpdateDeviceCustomTemplateResponseBody = exports.UpdateDeviceCustomTemplateRequest = exports.UpdateDeviceCustomTemplateHeaders = exports.SetSuperUserMeetingRoomResponse = exports.SetSuperUserMeetingRoomResponseBody = exports.SetSuperUserMeetingRoomRequest = exports.SetSuperUserMeetingRoomHeaders = exports.RemoveSuperUserMeetingRoomResponse = exports.RemoveSuperUserMeetingRoomResponseBody = exports.RemoveSuperUserMeetingRoomRequest = exports.RemoveSuperUserMeetingRoomHeaders = exports.QueryMeetingRoomListResponse = exports.QueryMeetingRoomListResponseBody = exports.QueryMeetingRoomListRequest = exports.QueryMeetingRoomListHeaders = exports.QueryMeetingRoomGroupListResponse = exports.QueryMeetingRoomGroupListResponseBody = exports.QueryMeetingRoomGroupListRequest = exports.QueryMeetingRoomGroupListHeaders = exports.QueryMeetingRoomGroupResponse = exports.QueryMeetingRoomGroupResponseBody = exports.QueryMeetingRoomGroupRequest = exports.QueryMeetingRoomGroupHeaders = exports.QueryMeetingRoomDeviceResponse = exports.QueryMeetingRoomDeviceResponseBody = exports.QueryMeetingRoomDeviceRequest = exports.QueryMeetingRoomDeviceHeaders = exports.QueryMeetingRoomControlPanelListResponse = exports.QueryMeetingRoomControlPanelListResponseBody = exports.QueryMeetingRoomControlPanelListRequest = exports.QueryMeetingRoomControlPanelListHeaders = exports.QueryMeetingRoomResponse = void 0;
exports.UpdateMeetingRoomRequestRoomLocation = exports.UpdateMeetingRoomRequestReservationAuthority = exports.UpdateMeetingRoomRequestReservationAuthorityAuthorizedMembers = exports.QueryMeetingRoomListResponseBodyResult = exports.QueryMeetingRoomListResponseBodyResultRoomLocation = exports.QueryMeetingRoomListResponseBodyResultRoomLabels = exports.QueryMeetingRoomListResponseBodyResultRoomGroup = exports.QueryMeetingRoomGroupListResponseBodyResult = exports.QueryMeetingRoomDeviceResponseBodyResult = exports.QueryMeetingRoomDeviceResponseBodyResultControllers = exports.QueryMeetingRoomControlPanelListResponseBodyResult = exports.QueryMeetingRoomControlPanelListResponseBodyResultRoomIotConfig = exports.QueryMeetingRoomResponseBodyResult = exports.QueryMeetingRoomResponseBodyResultRoomLocation = exports.QueryMeetingRoomResponseBodyResultRoomLabels = exports.QueryMeetingRoomResponseBodyResultRoomGroup = exports.QueryMeetingRoomResponseBodyResultReservationAuthority = exports.QueryMeetingRoomResponseBodyResultReservationAuthorityAuthorizedMembers = exports.QueryDevicePropertiesResponseBodyResult = exports.QueryDeviceIpByCodeResponseBodyResult = exports.QueryDeviceCustomTemplateListResponseBodyResult = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CreateDeviceCustomTemplateHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateDeviceCustomTemplateHeaders = CreateDeviceCustomTemplateHeaders;
class CreateDeviceCustomTemplateRequest extends $tea.Model {
    static names() {
        return {
            bgImgList: 'bgImgList',
            bgType: 'bgType',
            bgUrl: 'bgUrl',
            customDoc: 'customDoc',
            desensitizeUserName: 'desensitizeUserName',
            deviceUnionIds: 'deviceUnionIds',
            groupIds: 'groupIds',
            hideServerCodeWhenProjecting: 'hideServerCodeWhenProjecting',
            instruction: 'instruction',
            isPicTop: 'isPicTop',
            logo: 'logo',
            orgName: 'orgName',
            picturePlayInterval: 'picturePlayInterval',
            roomIds: 'roomIds',
            showCalendarCard: 'showCalendarCard',
            showCalendarTitle: 'showCalendarTitle',
            showFunctionCard: 'showFunctionCard',
            templateName: 'templateName',
        };
    }
    static types() {
        return {
            bgImgList: { 'type': 'array', 'itemType': 'string' },
            bgType: 'number',
            bgUrl: 'string',
            customDoc: 'string',
            desensitizeUserName: 'boolean',
            deviceUnionIds: { 'type': 'array', 'itemType': 'string' },
            groupIds: { 'type': 'array', 'itemType': 'number' },
            hideServerCodeWhenProjecting: 'boolean',
            instruction: 'boolean',
            isPicTop: 'number',
            logo: 'string',
            orgName: 'string',
            picturePlayInterval: 'number',
            roomIds: { 'type': 'array', 'itemType': 'string' },
            showCalendarCard: 'boolean',
            showCalendarTitle: 'boolean',
            showFunctionCard: 'boolean',
            templateName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateDeviceCustomTemplateRequest = CreateDeviceCustomTemplateRequest;
class CreateDeviceCustomTemplateResponseBody extends $tea.Model {
    static names() {
        return {
            templateId: 'templateId',
        };
    }
    static types() {
        return {
            templateId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateDeviceCustomTemplateResponseBody = CreateDeviceCustomTemplateResponseBody;
class CreateDeviceCustomTemplateResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateDeviceCustomTemplateResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateDeviceCustomTemplateResponse = CreateDeviceCustomTemplateResponse;
class CreateMeetingRoomHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomHeaders = CreateMeetingRoomHeaders;
class CreateMeetingRoomRequest extends $tea.Model {
    static names() {
        return {
            enableCycleReservation: 'enableCycleReservation',
            groupId: 'groupId',
            isvRoomId: 'isvRoomId',
            reservationAuthority: 'reservationAuthority',
            roomCapacity: 'roomCapacity',
            roomLabelIds: 'roomLabelIds',
            roomLocation: 'roomLocation',
            roomName: 'roomName',
            roomPicture: 'roomPicture',
            roomStatus: 'roomStatus',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            enableCycleReservation: 'boolean',
            groupId: 'number',
            isvRoomId: 'string',
            reservationAuthority: CreateMeetingRoomRequestReservationAuthority,
            roomCapacity: 'number',
            roomLabelIds: { 'type': 'array', 'itemType': 'number' },
            roomLocation: CreateMeetingRoomRequestRoomLocation,
            roomName: 'string',
            roomPicture: 'string',
            roomStatus: 'number',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomRequest = CreateMeetingRoomRequest;
class CreateMeetingRoomResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomResponseBody = CreateMeetingRoomResponseBody;
class CreateMeetingRoomResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateMeetingRoomResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomResponse = CreateMeetingRoomResponse;
class CreateMeetingRoomControlPanelHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomControlPanelHeaders = CreateMeetingRoomControlPanelHeaders;
class CreateMeetingRoomControlPanelRequest extends $tea.Model {
    static names() {
        return {
            extra: 'extra',
            roomConfig: 'roomConfig',
            roomId: 'roomId',
            status: 'status',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            extra: CreateMeetingRoomControlPanelRequestExtra,
            roomConfig: { 'type': 'array', 'itemType': CreateMeetingRoomControlPanelRequestRoomConfig },
            roomId: 'string',
            status: 'number',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomControlPanelRequest = CreateMeetingRoomControlPanelRequest;
class CreateMeetingRoomControlPanelResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomControlPanelResponseBody = CreateMeetingRoomControlPanelResponseBody;
class CreateMeetingRoomControlPanelResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateMeetingRoomControlPanelResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomControlPanelResponse = CreateMeetingRoomControlPanelResponse;
class CreateMeetingRoomGroupHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomGroupHeaders = CreateMeetingRoomGroupHeaders;
class CreateMeetingRoomGroupRequest extends $tea.Model {
    static names() {
        return {
            groupName: 'groupName',
            parentGroupId: 'parentGroupId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            groupName: 'string',
            parentGroupId: 'number',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomGroupRequest = CreateMeetingRoomGroupRequest;
class CreateMeetingRoomGroupResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomGroupResponseBody = CreateMeetingRoomGroupResponseBody;
class CreateMeetingRoomGroupResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateMeetingRoomGroupResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomGroupResponse = CreateMeetingRoomGroupResponse;
class DeleteDeviceCustomTemplateHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteDeviceCustomTemplateHeaders = DeleteDeviceCustomTemplateHeaders;
class DeleteDeviceCustomTemplateRequest extends $tea.Model {
    static names() {
        return {
            templateId: 'templateId',
        };
    }
    static types() {
        return {
            templateId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteDeviceCustomTemplateRequest = DeleteDeviceCustomTemplateRequest;
class DeleteDeviceCustomTemplateResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteDeviceCustomTemplateResponseBody = DeleteDeviceCustomTemplateResponseBody;
class DeleteDeviceCustomTemplateResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteDeviceCustomTemplateResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteDeviceCustomTemplateResponse = DeleteDeviceCustomTemplateResponse;
class DeleteMeetingRoomHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomHeaders = DeleteMeetingRoomHeaders;
class DeleteMeetingRoomRequest extends $tea.Model {
    static names() {
        return {
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomRequest = DeleteMeetingRoomRequest;
class DeleteMeetingRoomResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomResponseBody = DeleteMeetingRoomResponseBody;
class DeleteMeetingRoomResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteMeetingRoomResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomResponse = DeleteMeetingRoomResponse;
class DeleteMeetingRoomControlPanelHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomControlPanelHeaders = DeleteMeetingRoomControlPanelHeaders;
class DeleteMeetingRoomControlPanelRequest extends $tea.Model {
    static names() {
        return {
            body: 'body',
        };
    }
    static types() {
        return {
            body: DeleteMeetingRoomControlPanelRequestBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomControlPanelRequest = DeleteMeetingRoomControlPanelRequest;
class DeleteMeetingRoomControlPanelShrinkRequest extends $tea.Model {
    static names() {
        return {
            bodyShrink: 'body',
        };
    }
    static types() {
        return {
            bodyShrink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomControlPanelShrinkRequest = DeleteMeetingRoomControlPanelShrinkRequest;
class DeleteMeetingRoomControlPanelResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomControlPanelResponseBody = DeleteMeetingRoomControlPanelResponseBody;
class DeleteMeetingRoomControlPanelResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteMeetingRoomControlPanelResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomControlPanelResponse = DeleteMeetingRoomControlPanelResponse;
class DeleteMeetingRoomGroupHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomGroupHeaders = DeleteMeetingRoomGroupHeaders;
class DeleteMeetingRoomGroupRequest extends $tea.Model {
    static names() {
        return {
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomGroupRequest = DeleteMeetingRoomGroupRequest;
class DeleteMeetingRoomGroupResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomGroupResponseBody = DeleteMeetingRoomGroupResponseBody;
class DeleteMeetingRoomGroupResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteMeetingRoomGroupResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomGroupResponse = DeleteMeetingRoomGroupResponse;
class QueryDeviceCustomTemplateHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateHeaders = QueryDeviceCustomTemplateHeaders;
class QueryDeviceCustomTemplateResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QueryDeviceCustomTemplateResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateResponseBody = QueryDeviceCustomTemplateResponseBody;
class QueryDeviceCustomTemplateResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryDeviceCustomTemplateResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateResponse = QueryDeviceCustomTemplateResponse;
class QueryDeviceCustomTemplateListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateListHeaders = QueryDeviceCustomTemplateListHeaders;
class QueryDeviceCustomTemplateListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QueryDeviceCustomTemplateListResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateListResponseBody = QueryDeviceCustomTemplateListResponseBody;
class QueryDeviceCustomTemplateListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryDeviceCustomTemplateListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateListResponse = QueryDeviceCustomTemplateListResponse;
class QueryDeviceIpByCodeHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceIpByCodeHeaders = QueryDeviceIpByCodeHeaders;
class QueryDeviceIpByCodeRequest extends $tea.Model {
    static names() {
        return {
            deviceSn: 'deviceSn',
        };
    }
    static types() {
        return {
            deviceSn: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceIpByCodeRequest = QueryDeviceIpByCodeRequest;
class QueryDeviceIpByCodeResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: QueryDeviceIpByCodeResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceIpByCodeResponseBody = QueryDeviceIpByCodeResponseBody;
class QueryDeviceIpByCodeResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryDeviceIpByCodeResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceIpByCodeResponse = QueryDeviceIpByCodeResponse;
class QueryDevicePropertiesHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDevicePropertiesHeaders = QueryDevicePropertiesHeaders;
class QueryDevicePropertiesRequest extends $tea.Model {
    static names() {
        return {
            propertyNames: 'propertyNames',
            deviceId: 'deviceId',
            deviceUnionId: 'deviceUnionId',
            operatorUnionId: 'operatorUnionId',
        };
    }
    static types() {
        return {
            propertyNames: { 'type': 'array', 'itemType': 'string' },
            deviceId: 'string',
            deviceUnionId: 'string',
            operatorUnionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDevicePropertiesRequest = QueryDevicePropertiesRequest;
class QueryDevicePropertiesResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': QueryDevicePropertiesResponseBodyResult },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDevicePropertiesResponseBody = QueryDevicePropertiesResponseBody;
class QueryDevicePropertiesResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryDevicePropertiesResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDevicePropertiesResponse = QueryDevicePropertiesResponse;
class QueryMeetingRoomHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomHeaders = QueryMeetingRoomHeaders;
class QueryMeetingRoomRequest extends $tea.Model {
    static names() {
        return {
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomRequest = QueryMeetingRoomRequest;
class QueryMeetingRoomResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QueryMeetingRoomResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomResponseBody = QueryMeetingRoomResponseBody;
class QueryMeetingRoomResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryMeetingRoomResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomResponse = QueryMeetingRoomResponse;
class QueryMeetingRoomControlPanelListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomControlPanelListHeaders = QueryMeetingRoomControlPanelListHeaders;
class QueryMeetingRoomControlPanelListRequest extends $tea.Model {
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            roomId: 'roomId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'number',
            roomId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomControlPanelListRequest = QueryMeetingRoomControlPanelListRequest;
class QueryMeetingRoomControlPanelListResponseBody extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            nextToken: 'nextToken',
            result: 'result',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            nextToken: 'number',
            result: { 'type': 'array', 'itemType': QueryMeetingRoomControlPanelListResponseBodyResult },
            totalCount: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomControlPanelListResponseBody = QueryMeetingRoomControlPanelListResponseBody;
class QueryMeetingRoomControlPanelListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryMeetingRoomControlPanelListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomControlPanelListResponse = QueryMeetingRoomControlPanelListResponse;
class QueryMeetingRoomDeviceHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomDeviceHeaders = QueryMeetingRoomDeviceHeaders;
class QueryMeetingRoomDeviceRequest extends $tea.Model {
    static names() {
        return {
            deviceId: 'deviceId',
            deviceUnionId: 'deviceUnionId',
            operatorUnionId: 'operatorUnionId',
        };
    }
    static types() {
        return {
            deviceId: 'string',
            deviceUnionId: 'string',
            operatorUnionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomDeviceRequest = QueryMeetingRoomDeviceRequest;
class QueryMeetingRoomDeviceResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QueryMeetingRoomDeviceResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomDeviceResponseBody = QueryMeetingRoomDeviceResponseBody;
class QueryMeetingRoomDeviceResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryMeetingRoomDeviceResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomDeviceResponse = QueryMeetingRoomDeviceResponse;
class QueryMeetingRoomGroupHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomGroupHeaders = QueryMeetingRoomGroupHeaders;
class QueryMeetingRoomGroupRequest extends $tea.Model {
    static names() {
        return {
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomGroupRequest = QueryMeetingRoomGroupRequest;
class QueryMeetingRoomGroupResponseBody extends $tea.Model {
    static names() {
        return {
            groupId: 'groupId',
            groupName: 'groupName',
            parentId: 'parentId',
        };
    }
    static types() {
        return {
            groupId: 'number',
            groupName: 'string',
            parentId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomGroupResponseBody = QueryMeetingRoomGroupResponseBody;
class QueryMeetingRoomGroupResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryMeetingRoomGroupResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomGroupResponse = QueryMeetingRoomGroupResponse;
class QueryMeetingRoomGroupListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomGroupListHeaders = QueryMeetingRoomGroupListHeaders;
class QueryMeetingRoomGroupListRequest extends $tea.Model {
    static names() {
        return {
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomGroupListRequest = QueryMeetingRoomGroupListRequest;
class QueryMeetingRoomGroupListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': QueryMeetingRoomGroupListResponseBodyResult },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomGroupListResponseBody = QueryMeetingRoomGroupListResponseBody;
class QueryMeetingRoomGroupListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryMeetingRoomGroupListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomGroupListResponse = QueryMeetingRoomGroupListResponse;
class QueryMeetingRoomListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomListHeaders = QueryMeetingRoomListHeaders;
class QueryMeetingRoomListRequest extends $tea.Model {
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'number',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomListRequest = QueryMeetingRoomListRequest;
class QueryMeetingRoomListResponseBody extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            nextToken: 'nextToken',
            result: 'result',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            nextToken: 'number',
            result: { 'type': 'array', 'itemType': QueryMeetingRoomListResponseBodyResult },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomListResponseBody = QueryMeetingRoomListResponseBody;
class QueryMeetingRoomListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryMeetingRoomListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomListResponse = QueryMeetingRoomListResponse;
class RemoveSuperUserMeetingRoomHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveSuperUserMeetingRoomHeaders = RemoveSuperUserMeetingRoomHeaders;
class RemoveSuperUserMeetingRoomRequest extends $tea.Model {
    static names() {
        return {
            roomId: 'roomId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            roomId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveSuperUserMeetingRoomRequest = RemoveSuperUserMeetingRoomRequest;
class RemoveSuperUserMeetingRoomResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveSuperUserMeetingRoomResponseBody = RemoveSuperUserMeetingRoomResponseBody;
class RemoveSuperUserMeetingRoomResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RemoveSuperUserMeetingRoomResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveSuperUserMeetingRoomResponse = RemoveSuperUserMeetingRoomResponse;
class SetSuperUserMeetingRoomHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SetSuperUserMeetingRoomHeaders = SetSuperUserMeetingRoomHeaders;
class SetSuperUserMeetingRoomRequest extends $tea.Model {
    static names() {
        return {
            deptIdWhiteList: 'deptIdWhiteList',
            roomId: 'roomId',
            unionId: 'unionId',
            userIdWhiteList: 'userIdWhiteList',
        };
    }
    static types() {
        return {
            deptIdWhiteList: { 'type': 'array', 'itemType': 'number' },
            roomId: 'string',
            unionId: 'string',
            userIdWhiteList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SetSuperUserMeetingRoomRequest = SetSuperUserMeetingRoomRequest;
class SetSuperUserMeetingRoomResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SetSuperUserMeetingRoomResponseBody = SetSuperUserMeetingRoomResponseBody;
class SetSuperUserMeetingRoomResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SetSuperUserMeetingRoomResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SetSuperUserMeetingRoomResponse = SetSuperUserMeetingRoomResponse;
class UpdateDeviceCustomTemplateHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateDeviceCustomTemplateHeaders = UpdateDeviceCustomTemplateHeaders;
class UpdateDeviceCustomTemplateRequest extends $tea.Model {
    static names() {
        return {
            bgImgList: 'bgImgList',
            bgType: 'bgType',
            bgUrl: 'bgUrl',
            customDoc: 'customDoc',
            desensitizeUserName: 'desensitizeUserName',
            deviceUnionIds: 'deviceUnionIds',
            groupIds: 'groupIds',
            hideServerCodeWhenProjecting: 'hideServerCodeWhenProjecting',
            instruction: 'instruction',
            isPicTop: 'isPicTop',
            logo: 'logo',
            orgName: 'orgName',
            picturePlayInterval: 'picturePlayInterval',
            roomIds: 'roomIds',
            showCalendarCard: 'showCalendarCard',
            showCalendarTitle: 'showCalendarTitle',
            showFunctionCard: 'showFunctionCard',
            templateId: 'templateId',
            templateName: 'templateName',
        };
    }
    static types() {
        return {
            bgImgList: { 'type': 'array', 'itemType': 'string' },
            bgType: 'number',
            bgUrl: 'string',
            customDoc: 'string',
            desensitizeUserName: 'boolean',
            deviceUnionIds: { 'type': 'array', 'itemType': 'string' },
            groupIds: { 'type': 'array', 'itemType': 'number' },
            hideServerCodeWhenProjecting: 'boolean',
            instruction: 'boolean',
            isPicTop: 'number',
            logo: 'string',
            orgName: 'string',
            picturePlayInterval: 'number',
            roomIds: { 'type': 'array', 'itemType': 'string' },
            showCalendarCard: 'boolean',
            showCalendarTitle: 'boolean',
            showFunctionCard: 'boolean',
            templateId: 'number',
            templateName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateDeviceCustomTemplateRequest = UpdateDeviceCustomTemplateRequest;
class UpdateDeviceCustomTemplateResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateDeviceCustomTemplateResponseBody = UpdateDeviceCustomTemplateResponseBody;
class UpdateDeviceCustomTemplateResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateDeviceCustomTemplateResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateDeviceCustomTemplateResponse = UpdateDeviceCustomTemplateResponse;
class UpdateMeetingRoomHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomHeaders = UpdateMeetingRoomHeaders;
class UpdateMeetingRoomRequest extends $tea.Model {
    static names() {
        return {
            enableCycleReservation: 'enableCycleReservation',
            groupId: 'groupId',
            isvRoomId: 'isvRoomId',
            reservationAuthority: 'reservationAuthority',
            roomCapacity: 'roomCapacity',
            roomId: 'roomId',
            roomLabelIds: 'roomLabelIds',
            roomLocation: 'roomLocation',
            roomName: 'roomName',
            roomPicture: 'roomPicture',
            roomStatus: 'roomStatus',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            enableCycleReservation: 'boolean',
            groupId: 'number',
            isvRoomId: 'string',
            reservationAuthority: UpdateMeetingRoomRequestReservationAuthority,
            roomCapacity: 'number',
            roomId: 'string',
            roomLabelIds: { 'type': 'array', 'itemType': 'number' },
            roomLocation: UpdateMeetingRoomRequestRoomLocation,
            roomName: 'string',
            roomPicture: 'string',
            roomStatus: 'number',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomRequest = UpdateMeetingRoomRequest;
class UpdateMeetingRoomResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomResponseBody = UpdateMeetingRoomResponseBody;
class UpdateMeetingRoomResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateMeetingRoomResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomResponse = UpdateMeetingRoomResponse;
class UpdateMeetingRoomGroupHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomGroupHeaders = UpdateMeetingRoomGroupHeaders;
class UpdateMeetingRoomGroupRequest extends $tea.Model {
    static names() {
        return {
            groupId: 'groupId',
            groupName: 'groupName',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            groupId: 'number',
            groupName: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomGroupRequest = UpdateMeetingRoomGroupRequest;
class UpdateMeetingRoomGroupResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomGroupResponseBody = UpdateMeetingRoomGroupResponseBody;
class UpdateMeetingRoomGroupResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateMeetingRoomGroupResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomGroupResponse = UpdateMeetingRoomGroupResponse;
class CreateMeetingRoomRequestReservationAuthorityAuthorizedMembers extends $tea.Model {
    static names() {
        return {
            memberId: 'memberId',
            memberName: 'memberName',
            memberType: 'memberType',
        };
    }
    static types() {
        return {
            memberId: 'string',
            memberName: 'string',
            memberType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomRequestReservationAuthorityAuthorizedMembers = CreateMeetingRoomRequestReservationAuthorityAuthorizedMembers;
class CreateMeetingRoomRequestReservationAuthority extends $tea.Model {
    static names() {
        return {
            authorizedMembers: 'authorizedMembers',
        };
    }
    static types() {
        return {
            authorizedMembers: { 'type': 'array', 'itemType': CreateMeetingRoomRequestReservationAuthorityAuthorizedMembers },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomRequestReservationAuthority = CreateMeetingRoomRequestReservationAuthority;
class CreateMeetingRoomRequestRoomLocation extends $tea.Model {
    static names() {
        return {
            desc: 'desc',
            title: 'title',
        };
    }
    static types() {
        return {
            desc: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomRequestRoomLocation = CreateMeetingRoomRequestRoomLocation;
class CreateMeetingRoomControlPanelRequestExtra extends $tea.Model {
    static names() {
        return {
            param: 'param',
        };
    }
    static types() {
        return {
            param: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomControlPanelRequestExtra = CreateMeetingRoomControlPanelRequestExtra;
class CreateMeetingRoomControlPanelRequestRoomConfig extends $tea.Model {
    static names() {
        return {
            enName: 'enName',
            icon: 'icon',
            name: 'name',
            showTime: 'showTime',
            sort: 'sort',
            url: 'url',
        };
    }
    static types() {
        return {
            enName: 'string',
            icon: 'string',
            name: 'string',
            showTime: 'number',
            sort: 'number',
            url: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateMeetingRoomControlPanelRequestRoomConfig = CreateMeetingRoomControlPanelRequestRoomConfig;
class DeleteMeetingRoomControlPanelRequestBody extends $tea.Model {
    static names() {
        return {
            roomIds: 'roomIds',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            roomIds: { 'type': 'array', 'itemType': 'string' },
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteMeetingRoomControlPanelRequestBody = DeleteMeetingRoomControlPanelRequestBody;
class QueryDeviceCustomTemplateResponseBodyResultDeviceCustomTemplate extends $tea.Model {
    static names() {
        return {
            bgImageList: 'bgImageList',
            bgType: 'bgType',
            bgUrl: 'bgUrl',
            confSubType: 'confSubType',
            confType: 'confType',
            corpId: 'corpId',
            customDoc: 'customDoc',
            desensitizeUserName: 'desensitizeUserName',
            hideServerCodeWhenProjecting: 'hideServerCodeWhenProjecting',
            instruction: 'instruction',
            isPicTop: 'isPicTop',
            logo: 'logo',
            orgName: 'orgName',
            picturePlayInterval: 'picturePlayInterval',
            showCalendarCard: 'showCalendarCard',
            showCalendarTitle: 'showCalendarTitle',
            showFunctionCard: 'showFunctionCard',
            templateId: 'templateId',
            templateName: 'templateName',
        };
    }
    static types() {
        return {
            bgImageList: { 'type': 'array', 'itemType': 'string' },
            bgType: 'number',
            bgUrl: 'string',
            confSubType: 'number',
            confType: 'number',
            corpId: 'string',
            customDoc: 'string',
            desensitizeUserName: 'boolean',
            hideServerCodeWhenProjecting: 'boolean',
            instruction: 'boolean',
            isPicTop: 'number',
            logo: 'string',
            orgName: 'string',
            picturePlayInterval: 'number',
            showCalendarCard: 'boolean',
            showCalendarTitle: 'boolean',
            showFunctionCard: 'boolean',
            templateId: 'number',
            templateName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateResponseBodyResultDeviceCustomTemplate = QueryDeviceCustomTemplateResponseBodyResultDeviceCustomTemplate;
class QueryDeviceCustomTemplateResponseBodyResult extends $tea.Model {
    static names() {
        return {
            deviceCustomTemplate: 'deviceCustomTemplate',
            deviceUnionIds: 'deviceUnionIds',
            groupIds: 'groupIds',
            roomIds: 'roomIds',
        };
    }
    static types() {
        return {
            deviceCustomTemplate: QueryDeviceCustomTemplateResponseBodyResultDeviceCustomTemplate,
            deviceUnionIds: { 'type': 'array', 'itemType': 'string' },
            groupIds: { 'type': 'array', 'itemType': 'number' },
            roomIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateResponseBodyResult = QueryDeviceCustomTemplateResponseBodyResult;
class QueryDeviceCustomTemplateListResponseBodyResultDeviceCustomTemplates extends $tea.Model {
    static names() {
        return {
            bgImageList: 'bgImageList',
            bgType: 'bgType',
            bgUrl: 'bgUrl',
            confSubType: 'confSubType',
            confType: 'confType',
            corpId: 'corpId',
            customDoc: 'customDoc',
            desensitizeUserName: 'desensitizeUserName',
            hideServerCodeWhenProjecting: 'hideServerCodeWhenProjecting',
            instruction: 'instruction',
            isPicTop: 'isPicTop',
            logo: 'logo',
            orgName: 'orgName',
            picturePlayInterval: 'picturePlayInterval',
            showCalendarCard: 'showCalendarCard',
            showCalendarTitle: 'showCalendarTitle',
            showFunctionCard: 'showFunctionCard',
            templateId: 'templateId',
            templateName: 'templateName',
        };
    }
    static types() {
        return {
            bgImageList: { 'type': 'array', 'itemType': 'string' },
            bgType: 'number',
            bgUrl: 'string',
            confSubType: 'number',
            confType: 'number',
            corpId: 'string',
            customDoc: 'string',
            desensitizeUserName: 'boolean',
            hideServerCodeWhenProjecting: 'boolean',
            instruction: 'boolean',
            isPicTop: 'number',
            logo: 'string',
            orgName: 'string',
            picturePlayInterval: 'number',
            showCalendarCard: 'boolean',
            showCalendarTitle: 'boolean',
            showFunctionCard: 'boolean',
            templateId: 'number',
            templateName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateListResponseBodyResultDeviceCustomTemplates = QueryDeviceCustomTemplateListResponseBodyResultDeviceCustomTemplates;
class QueryDeviceCustomTemplateListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            deviceCustomTemplates: 'deviceCustomTemplates',
            deviceTemplateMap: 'deviceTemplateMap',
            groupIdTemplateMap: 'groupIdTemplateMap',
            roomIdTemplateMap: 'roomIdTemplateMap',
        };
    }
    static types() {
        return {
            deviceCustomTemplates: { 'type': 'array', 'itemType': QueryDeviceCustomTemplateListResponseBodyResultDeviceCustomTemplates },
            deviceTemplateMap: { 'type': 'map', 'keyType': 'string', 'valueType': { 'type': 'array', 'itemType': 'string' } },
            groupIdTemplateMap: { 'type': 'map', 'keyType': 'string', 'valueType': { 'type': 'array', 'itemType': 'number' } },
            roomIdTemplateMap: { 'type': 'map', 'keyType': 'string', 'valueType': { 'type': 'array', 'itemType': 'string' } },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceCustomTemplateListResponseBodyResult = QueryDeviceCustomTemplateListResponseBodyResult;
class QueryDeviceIpByCodeResponseBodyResult extends $tea.Model {
    static names() {
        return {
            deviceIp: 'deviceIp',
        };
    }
    static types() {
        return {
            deviceIp: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDeviceIpByCodeResponseBodyResult = QueryDeviceIpByCodeResponseBodyResult;
class QueryDevicePropertiesResponseBodyResult extends $tea.Model {
    static names() {
        return {
            propertyName: 'propertyName',
            propertyValue: 'propertyValue',
        };
    }
    static types() {
        return {
            propertyName: 'string',
            propertyValue: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryDevicePropertiesResponseBodyResult = QueryDevicePropertiesResponseBodyResult;
class QueryMeetingRoomResponseBodyResultReservationAuthorityAuthorizedMembers extends $tea.Model {
    static names() {
        return {
            memberId: 'memberId',
            memberName: 'memberName',
            memberType: 'memberType',
        };
    }
    static types() {
        return {
            memberId: 'string',
            memberName: 'string',
            memberType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomResponseBodyResultReservationAuthorityAuthorizedMembers = QueryMeetingRoomResponseBodyResultReservationAuthorityAuthorizedMembers;
class QueryMeetingRoomResponseBodyResultReservationAuthority extends $tea.Model {
    static names() {
        return {
            authorizedMembers: 'authorizedMembers',
        };
    }
    static types() {
        return {
            authorizedMembers: { 'type': 'array', 'itemType': QueryMeetingRoomResponseBodyResultReservationAuthorityAuthorizedMembers },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomResponseBodyResultReservationAuthority = QueryMeetingRoomResponseBodyResultReservationAuthority;
class QueryMeetingRoomResponseBodyResultRoomGroup extends $tea.Model {
    static names() {
        return {
            groupId: 'groupId',
            groupName: 'groupName',
            parentId: 'parentId',
        };
    }
    static types() {
        return {
            groupId: 'number',
            groupName: 'string',
            parentId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomResponseBodyResultRoomGroup = QueryMeetingRoomResponseBodyResultRoomGroup;
class QueryMeetingRoomResponseBodyResultRoomLabels extends $tea.Model {
    static names() {
        return {
            labelId: 'labelId',
            labelName: 'labelName',
        };
    }
    static types() {
        return {
            labelId: 'number',
            labelName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomResponseBodyResultRoomLabels = QueryMeetingRoomResponseBodyResultRoomLabels;
class QueryMeetingRoomResponseBodyResultRoomLocation extends $tea.Model {
    static names() {
        return {
            desc: 'desc',
            title: 'title',
        };
    }
    static types() {
        return {
            desc: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomResponseBodyResultRoomLocation = QueryMeetingRoomResponseBodyResultRoomLocation;
class QueryMeetingRoomResponseBodyResult extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            deviceUnionIds: 'deviceUnionIds',
            enableCycleReservation: 'enableCycleReservation',
            isvRoomId: 'isvRoomId',
            reservationAuthority: 'reservationAuthority',
            roomCapacity: 'roomCapacity',
            roomGroup: 'roomGroup',
            roomId: 'roomId',
            roomLabels: 'roomLabels',
            roomLocation: 'roomLocation',
            roomName: 'roomName',
            roomPicture: 'roomPicture',
            roomStaffId: 'roomStaffId',
            roomStatus: 'roomStatus',
        };
    }
    static types() {
        return {
            corpId: 'string',
            deviceUnionIds: { 'type': 'array', 'itemType': 'string' },
            enableCycleReservation: 'boolean',
            isvRoomId: 'string',
            reservationAuthority: QueryMeetingRoomResponseBodyResultReservationAuthority,
            roomCapacity: 'number',
            roomGroup: QueryMeetingRoomResponseBodyResultRoomGroup,
            roomId: 'string',
            roomLabels: { 'type': 'array', 'itemType': QueryMeetingRoomResponseBodyResultRoomLabels },
            roomLocation: QueryMeetingRoomResponseBodyResultRoomLocation,
            roomName: 'string',
            roomPicture: 'string',
            roomStaffId: 'string',
            roomStatus: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomResponseBodyResult = QueryMeetingRoomResponseBodyResult;
class QueryMeetingRoomControlPanelListResponseBodyResultRoomIotConfig extends $tea.Model {
    static names() {
        return {
            enName: 'enName',
            icon: 'icon',
            name: 'name',
            showTime: 'showTime',
            sort: 'sort',
            url: 'url',
        };
    }
    static types() {
        return {
            enName: 'string',
            icon: 'string',
            name: 'string',
            showTime: 'number',
            sort: 'number',
            url: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomControlPanelListResponseBodyResultRoomIotConfig = QueryMeetingRoomControlPanelListResponseBodyResultRoomIotConfig;
class QueryMeetingRoomControlPanelListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            roomId: 'roomId',
            roomIotConfig: 'roomIotConfig',
        };
    }
    static types() {
        return {
            roomId: 'string',
            roomIotConfig: { 'type': 'array', 'itemType': QueryMeetingRoomControlPanelListResponseBodyResultRoomIotConfig },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomControlPanelListResponseBodyResult = QueryMeetingRoomControlPanelListResponseBodyResult;
class QueryMeetingRoomDeviceResponseBodyResultControllers extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            deviceId: 'deviceId',
            deviceMac: 'deviceMac',
            deviceModel: 'deviceModel',
            deviceName: 'deviceName',
            deviceServiceId: 'deviceServiceId',
            deviceSn: 'deviceSn',
            deviceStatus: 'deviceStatus',
            deviceType: 'deviceType',
            deviceUnionId: 'deviceUnionId',
            openRoomId: 'openRoomId',
            shareCode: 'shareCode',
        };
    }
    static types() {
        return {
            corpId: 'string',
            deviceId: 'string',
            deviceMac: 'string',
            deviceModel: 'string',
            deviceName: 'string',
            deviceServiceId: 'number',
            deviceSn: 'string',
            deviceStatus: 'string',
            deviceType: 'string',
            deviceUnionId: 'string',
            openRoomId: 'string',
            shareCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomDeviceResponseBodyResultControllers = QueryMeetingRoomDeviceResponseBodyResultControllers;
class QueryMeetingRoomDeviceResponseBodyResult extends $tea.Model {
    static names() {
        return {
            activeTime: 'activeTime',
            controllers: 'controllers',
            corpId: 'corpId',
            creatorUnionId: 'creatorUnionId',
            devCamera: 'devCamera',
            devHdmi: 'devHdmi',
            devMic: 'devMic',
            devMirror: 'devMirror',
            devNetIp: 'devNetIp',
            devNetType: 'devNetType',
            devVoice: 'devVoice',
            devWifiMac: 'devWifiMac',
            devWireMac: 'devWireMac',
            deviceId: 'deviceId',
            deviceMac: 'deviceMac',
            deviceModel: 'deviceModel',
            deviceName: 'deviceName',
            deviceServiceId: 'deviceServiceId',
            deviceSn: 'deviceSn',
            deviceStatus: 'deviceStatus',
            deviceType: 'deviceType',
            deviceUnionId: 'deviceUnionId',
            firmwareVersion: 'firmwareVersion',
            openRoomId: 'openRoomId',
            roomName: 'roomName',
            shareCode: 'shareCode',
            sipAccountName: 'sipAccountName',
            softwareVersion: 'softwareVersion',
        };
    }
    static types() {
        return {
            activeTime: 'number',
            controllers: { 'type': 'array', 'itemType': QueryMeetingRoomDeviceResponseBodyResultControllers },
            corpId: 'string',
            creatorUnionId: 'string',
            devCamera: 'string',
            devHdmi: 'string',
            devMic: 'string',
            devMirror: 'string',
            devNetIp: 'string',
            devNetType: 'string',
            devVoice: 'string',
            devWifiMac: 'string',
            devWireMac: 'string',
            deviceId: 'string',
            deviceMac: 'string',
            deviceModel: 'string',
            deviceName: 'string',
            deviceServiceId: 'number',
            deviceSn: 'string',
            deviceStatus: 'string',
            deviceType: 'string',
            deviceUnionId: 'string',
            firmwareVersion: 'string',
            openRoomId: 'string',
            roomName: 'string',
            shareCode: 'string',
            sipAccountName: 'string',
            softwareVersion: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomDeviceResponseBodyResult = QueryMeetingRoomDeviceResponseBodyResult;
class QueryMeetingRoomGroupListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            groupId: 'groupId',
            groupName: 'groupName',
            parentId: 'parentId',
        };
    }
    static types() {
        return {
            groupId: 'number',
            groupName: 'string',
            parentId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomGroupListResponseBodyResult = QueryMeetingRoomGroupListResponseBodyResult;
class QueryMeetingRoomListResponseBodyResultRoomGroup extends $tea.Model {
    static names() {
        return {
            groupId: 'groupId',
            groupName: 'groupName',
            parentId: 'parentId',
        };
    }
    static types() {
        return {
            groupId: 'number',
            groupName: 'string',
            parentId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomListResponseBodyResultRoomGroup = QueryMeetingRoomListResponseBodyResultRoomGroup;
class QueryMeetingRoomListResponseBodyResultRoomLabels extends $tea.Model {
    static names() {
        return {
            labelId: 'labelId',
            labelName: 'labelName',
        };
    }
    static types() {
        return {
            labelId: 'number',
            labelName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomListResponseBodyResultRoomLabels = QueryMeetingRoomListResponseBodyResultRoomLabels;
class QueryMeetingRoomListResponseBodyResultRoomLocation extends $tea.Model {
    static names() {
        return {
            desc: 'desc',
            title: 'title',
        };
    }
    static types() {
        return {
            desc: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomListResponseBodyResultRoomLocation = QueryMeetingRoomListResponseBodyResultRoomLocation;
class QueryMeetingRoomListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            isvRoomId: 'isvRoomId',
            roomCapacity: 'roomCapacity',
            roomGroup: 'roomGroup',
            roomId: 'roomId',
            roomLabels: 'roomLabels',
            roomLocation: 'roomLocation',
            roomName: 'roomName',
            roomPicture: 'roomPicture',
            roomStaffId: 'roomStaffId',
            roomStatus: 'roomStatus',
        };
    }
    static types() {
        return {
            corpId: 'string',
            isvRoomId: 'string',
            roomCapacity: 'number',
            roomGroup: QueryMeetingRoomListResponseBodyResultRoomGroup,
            roomId: 'string',
            roomLabels: { 'type': 'array', 'itemType': QueryMeetingRoomListResponseBodyResultRoomLabels },
            roomLocation: QueryMeetingRoomListResponseBodyResultRoomLocation,
            roomName: 'string',
            roomPicture: 'string',
            roomStaffId: 'string',
            roomStatus: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryMeetingRoomListResponseBodyResult = QueryMeetingRoomListResponseBodyResult;
class UpdateMeetingRoomRequestReservationAuthorityAuthorizedMembers extends $tea.Model {
    static names() {
        return {
            memberId: 'memberId',
            memberName: 'memberName',
            memberType: 'memberType',
        };
    }
    static types() {
        return {
            memberId: 'string',
            memberName: 'string',
            memberType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomRequestReservationAuthorityAuthorizedMembers = UpdateMeetingRoomRequestReservationAuthorityAuthorizedMembers;
class UpdateMeetingRoomRequestReservationAuthority extends $tea.Model {
    static names() {
        return {
            authorizedMembers: 'authorizedMembers',
        };
    }
    static types() {
        return {
            authorizedMembers: { 'type': 'array', 'itemType': UpdateMeetingRoomRequestReservationAuthorityAuthorizedMembers },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomRequestReservationAuthority = UpdateMeetingRoomRequestReservationAuthority;
class UpdateMeetingRoomRequestRoomLocation extends $tea.Model {
    static names() {
        return {
            desc: 'desc',
            title: 'title',
        };
    }
    static types() {
        return {
            desc: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMeetingRoomRequestRoomLocation = UpdateMeetingRoomRequestRoomLocation;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 
     *
     * @param request CreateDeviceCustomTemplateRequest
     * @param headers CreateDeviceCustomTemplateHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateDeviceCustomTemplateResponse
     */
    async createDeviceCustomTemplateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bgImgList)) {
            body["bgImgList"] = request.bgImgList;
        }
        if (!tea_util_1.default.isUnset(request.bgType)) {
            body["bgType"] = request.bgType;
        }
        if (!tea_util_1.default.isUnset(request.bgUrl)) {
            body["bgUrl"] = request.bgUrl;
        }
        if (!tea_util_1.default.isUnset(request.customDoc)) {
            body["customDoc"] = request.customDoc;
        }
        if (!tea_util_1.default.isUnset(request.desensitizeUserName)) {
            body["desensitizeUserName"] = request.desensitizeUserName;
        }
        if (!tea_util_1.default.isUnset(request.deviceUnionIds)) {
            body["deviceUnionIds"] = request.deviceUnionIds;
        }
        if (!tea_util_1.default.isUnset(request.groupIds)) {
            body["groupIds"] = request.groupIds;
        }
        if (!tea_util_1.default.isUnset(request.hideServerCodeWhenProjecting)) {
            body["hideServerCodeWhenProjecting"] = request.hideServerCodeWhenProjecting;
        }
        if (!tea_util_1.default.isUnset(request.instruction)) {
            body["instruction"] = request.instruction;
        }
        if (!tea_util_1.default.isUnset(request.isPicTop)) {
            body["isPicTop"] = request.isPicTop;
        }
        if (!tea_util_1.default.isUnset(request.logo)) {
            body["logo"] = request.logo;
        }
        if (!tea_util_1.default.isUnset(request.orgName)) {
            body["orgName"] = request.orgName;
        }
        if (!tea_util_1.default.isUnset(request.picturePlayInterval)) {
            body["picturePlayInterval"] = request.picturePlayInterval;
        }
        if (!tea_util_1.default.isUnset(request.roomIds)) {
            body["roomIds"] = request.roomIds;
        }
        if (!tea_util_1.default.isUnset(request.showCalendarCard)) {
            body["showCalendarCard"] = request.showCalendarCard;
        }
        if (!tea_util_1.default.isUnset(request.showCalendarTitle)) {
            body["showCalendarTitle"] = request.showCalendarTitle;
        }
        if (!tea_util_1.default.isUnset(request.showFunctionCard)) {
            body["showFunctionCard"] = request.showFunctionCard;
        }
        if (!tea_util_1.default.isUnset(request.templateName)) {
            body["templateName"] = request.templateName;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateDeviceCustomTemplate",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/devices/screens/templates`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateDeviceCustomTemplateResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CreateDeviceCustomTemplateRequest
     * @return CreateDeviceCustomTemplateResponse
     */
    async createDeviceCustomTemplate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateDeviceCustomTemplateHeaders({});
        return await this.createDeviceCustomTemplateWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CreateMeetingRoomRequest
     * @param headers CreateMeetingRoomHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateMeetingRoomResponse
     */
    async createMeetingRoomWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.enableCycleReservation)) {
            body["enableCycleReservation"] = request.enableCycleReservation;
        }
        if (!tea_util_1.default.isUnset(request.groupId)) {
            body["groupId"] = request.groupId;
        }
        if (!tea_util_1.default.isUnset(request.isvRoomId)) {
            body["isvRoomId"] = request.isvRoomId;
        }
        if (!tea_util_1.default.isUnset(request.reservationAuthority)) {
            body["reservationAuthority"] = request.reservationAuthority;
        }
        if (!tea_util_1.default.isUnset(request.roomCapacity)) {
            body["roomCapacity"] = request.roomCapacity;
        }
        if (!tea_util_1.default.isUnset(request.roomLabelIds)) {
            body["roomLabelIds"] = request.roomLabelIds;
        }
        if (!tea_util_1.default.isUnset(request.roomLocation)) {
            body["roomLocation"] = request.roomLocation;
        }
        if (!tea_util_1.default.isUnset(request.roomName)) {
            body["roomName"] = request.roomName;
        }
        if (!tea_util_1.default.isUnset(request.roomPicture)) {
            body["roomPicture"] = request.roomPicture;
        }
        if (!tea_util_1.default.isUnset(request.roomStatus)) {
            body["roomStatus"] = request.roomStatus;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateMeetingRoom",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/meetingrooms`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateMeetingRoomResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CreateMeetingRoomRequest
     * @return CreateMeetingRoomResponse
     */
    async createMeetingRoom(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateMeetingRoomHeaders({});
        return await this.createMeetingRoomWithOptions(request, headers, runtime);
    }
    /**
     * @summary IOT
     *
     * @param request CreateMeetingRoomControlPanelRequest
     * @param headers CreateMeetingRoomControlPanelHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateMeetingRoomControlPanelResponse
     */
    async createMeetingRoomControlPanelWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.extra)) {
            body["extra"] = request.extra;
        }
        if (!tea_util_1.default.isUnset(request.roomConfig)) {
            body["roomConfig"] = request.roomConfig;
        }
        if (!tea_util_1.default.isUnset(request.roomId)) {
            body["roomId"] = request.roomId;
        }
        if (!tea_util_1.default.isUnset(request.status)) {
            body["status"] = request.status;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateMeetingRoomControlPanel",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/panels`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateMeetingRoomControlPanelResponse({}));
    }
    /**
     * @summary IOT
     *
     * @param request CreateMeetingRoomControlPanelRequest
     * @return CreateMeetingRoomControlPanelResponse
     */
    async createMeetingRoomControlPanel(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateMeetingRoomControlPanelHeaders({});
        return await this.createMeetingRoomControlPanelWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CreateMeetingRoomGroupRequest
     * @param headers CreateMeetingRoomGroupHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateMeetingRoomGroupResponse
     */
    async createMeetingRoomGroupWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupName)) {
            body["groupName"] = request.groupName;
        }
        if (!tea_util_1.default.isUnset(request.parentGroupId)) {
            body["parentGroupId"] = request.parentGroupId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateMeetingRoomGroup",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/groups`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateMeetingRoomGroupResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CreateMeetingRoomGroupRequest
     * @return CreateMeetingRoomGroupResponse
     */
    async createMeetingRoomGroup(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateMeetingRoomGroupHeaders({});
        return await this.createMeetingRoomGroupWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request DeleteDeviceCustomTemplateRequest
     * @param headers DeleteDeviceCustomTemplateHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeleteDeviceCustomTemplateResponse
     */
    async deleteDeviceCustomTemplateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.templateId)) {
            body["templateId"] = request.templateId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "DeleteDeviceCustomTemplate",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/devices/screens/templates/remove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteDeviceCustomTemplateResponse({}));
    }
    /**
     * @summary 
     *
     * @param request DeleteDeviceCustomTemplateRequest
     * @return DeleteDeviceCustomTemplateResponse
     */
    async deleteDeviceCustomTemplate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteDeviceCustomTemplateHeaders({});
        return await this.deleteDeviceCustomTemplateWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request DeleteMeetingRoomRequest
     * @param headers DeleteMeetingRoomHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeleteMeetingRoomResponse
     */
    async deleteMeetingRoomWithOptions(roomId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DeleteMeetingRoom",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/meetingRooms/${roomId}`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteMeetingRoomResponse({}));
    }
    /**
     * @summary 
     *
     * @param request DeleteMeetingRoomRequest
     * @return DeleteMeetingRoomResponse
     */
    async deleteMeetingRoom(roomId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteMeetingRoomHeaders({});
        return await this.deleteMeetingRoomWithOptions(roomId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param tmpReq DeleteMeetingRoomControlPanelRequest
     * @param headers DeleteMeetingRoomControlPanelHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeleteMeetingRoomControlPanelResponse
     */
    async deleteMeetingRoomControlPanelWithOptions(tmpReq, headers, runtime) {
        tea_util_1.default.validateModel(tmpReq);
        let request = new DeleteMeetingRoomControlPanelShrinkRequest({});
        openapi_util_1.default.convert(tmpReq, request);
        if (!tea_util_1.default.isUnset(tmpReq.body)) {
            request.bodyShrink = openapi_util_1.default.arrayToStringWithSpecifiedStyle(tmpReq.body, "body", "json");
        }
        let query = {};
        if (!tea_util_1.default.isUnset(request.bodyShrink)) {
            query["body"] = request.bodyShrink;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DeleteMeetingRoomControlPanel",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/panels/remove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteMeetingRoomControlPanelResponse({}));
    }
    /**
     * @summary 
     *
     * @param request DeleteMeetingRoomControlPanelRequest
     * @return DeleteMeetingRoomControlPanelResponse
     */
    async deleteMeetingRoomControlPanel(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteMeetingRoomControlPanelHeaders({});
        return await this.deleteMeetingRoomControlPanelWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request DeleteMeetingRoomGroupRequest
     * @param headers DeleteMeetingRoomGroupHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeleteMeetingRoomGroupResponse
     */
    async deleteMeetingRoomGroupWithOptions(groupId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DeleteMeetingRoomGroup",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/groups/${groupId}`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteMeetingRoomGroupResponse({}));
    }
    /**
     * @summary 
     *
     * @param request DeleteMeetingRoomGroupRequest
     * @return DeleteMeetingRoomGroupResponse
     */
    async deleteMeetingRoomGroup(groupId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteMeetingRoomGroupHeaders({});
        return await this.deleteMeetingRoomGroupWithOptions(groupId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers QueryDeviceCustomTemplateHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryDeviceCustomTemplateResponse
     */
    async queryDeviceCustomTemplateWithOptions(templateId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "QueryDeviceCustomTemplate",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/devices/screens/templates/${templateId}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryDeviceCustomTemplateResponse({}));
    }
    /**
     * @summary 
     *
     * @return QueryDeviceCustomTemplateResponse
     */
    async queryDeviceCustomTemplate(templateId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryDeviceCustomTemplateHeaders({});
        return await this.queryDeviceCustomTemplateWithOptions(templateId, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers QueryDeviceCustomTemplateListHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryDeviceCustomTemplateListResponse
     */
    async queryDeviceCustomTemplateListWithOptions(headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "QueryDeviceCustomTemplateList",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/devices/screens/templateLists`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryDeviceCustomTemplateListResponse({}));
    }
    /**
     * @summary 
     *
     * @return QueryDeviceCustomTemplateListResponse
     */
    async queryDeviceCustomTemplateList() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryDeviceCustomTemplateListHeaders({});
        return await this.queryDeviceCustomTemplateListWithOptions(headers, runtime);
    }
    /**
     * @summary ip
     *
     * @param request QueryDeviceIpByCodeRequest
     * @param headers QueryDeviceIpByCodeHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryDeviceIpByCodeResponse
     */
    async queryDeviceIpByCodeWithOptions(shareCode, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.deviceSn)) {
            query["deviceSn"] = request.deviceSn;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryDeviceIpByCode",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/devices/shareCodes/${shareCode}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryDeviceIpByCodeResponse({}));
    }
    /**
     * @summary ip
     *
     * @param request QueryDeviceIpByCodeRequest
     * @return QueryDeviceIpByCodeResponse
     */
    async queryDeviceIpByCode(shareCode, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryDeviceIpByCodeHeaders({});
        return await this.queryDeviceIpByCodeWithOptions(shareCode, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request QueryDevicePropertiesRequest
     * @param headers QueryDevicePropertiesHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryDevicePropertiesResponse
     */
    async queryDevicePropertiesWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.deviceId)) {
            query["deviceId"] = request.deviceId;
        }
        if (!tea_util_1.default.isUnset(request.deviceUnionId)) {
            query["deviceUnionId"] = request.deviceUnionId;
        }
        if (!tea_util_1.default.isUnset(request.operatorUnionId)) {
            query["operatorUnionId"] = request.operatorUnionId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.propertyNames)) {
            body["propertyNames"] = request.propertyNames;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "QueryDeviceProperties",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/devices/properties/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryDevicePropertiesResponse({}));
    }
    /**
     * @summary 
     *
     * @param request QueryDevicePropertiesRequest
     * @return QueryDevicePropertiesResponse
     */
    async queryDeviceProperties(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryDevicePropertiesHeaders({});
        return await this.queryDevicePropertiesWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomRequest
     * @param headers QueryMeetingRoomHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryMeetingRoomResponse
     */
    async queryMeetingRoomWithOptions(roomId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryMeetingRoom",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/meetingRooms/${roomId}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryMeetingRoomResponse({}));
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomRequest
     * @return QueryMeetingRoomResponse
     */
    async queryMeetingRoom(roomId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryMeetingRoomHeaders({});
        return await this.queryMeetingRoomWithOptions(roomId, request, headers, runtime);
    }
    /**
     * @summary IOT
     *
     * @param request QueryMeetingRoomControlPanelListRequest
     * @param headers QueryMeetingRoomControlPanelListHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryMeetingRoomControlPanelListResponse
     */
    async queryMeetingRoomControlPanelListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.roomId)) {
            query["roomId"] = request.roomId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryMeetingRoomControlPanelList",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/panels/lists`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryMeetingRoomControlPanelListResponse({}));
    }
    /**
     * @summary IOT
     *
     * @param request QueryMeetingRoomControlPanelListRequest
     * @return QueryMeetingRoomControlPanelListResponse
     */
    async queryMeetingRoomControlPanelList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryMeetingRoomControlPanelListHeaders({});
        return await this.queryMeetingRoomControlPanelListWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomDeviceRequest
     * @param headers QueryMeetingRoomDeviceHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryMeetingRoomDeviceResponse
     */
    async queryMeetingRoomDeviceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.deviceId)) {
            query["deviceId"] = request.deviceId;
        }
        if (!tea_util_1.default.isUnset(request.deviceUnionId)) {
            query["deviceUnionId"] = request.deviceUnionId;
        }
        if (!tea_util_1.default.isUnset(request.operatorUnionId)) {
            query["operatorUnionId"] = request.operatorUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryMeetingRoomDevice",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/devices`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryMeetingRoomDeviceResponse({}));
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomDeviceRequest
     * @return QueryMeetingRoomDeviceResponse
     */
    async queryMeetingRoomDevice(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryMeetingRoomDeviceHeaders({});
        return await this.queryMeetingRoomDeviceWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomGroupRequest
     * @param headers QueryMeetingRoomGroupHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryMeetingRoomGroupResponse
     */
    async queryMeetingRoomGroupWithOptions(groupId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryMeetingRoomGroup",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/groups/${groupId}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryMeetingRoomGroupResponse({}));
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomGroupRequest
     * @return QueryMeetingRoomGroupResponse
     */
    async queryMeetingRoomGroup(groupId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryMeetingRoomGroupHeaders({});
        return await this.queryMeetingRoomGroupWithOptions(groupId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomGroupListRequest
     * @param headers QueryMeetingRoomGroupListHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryMeetingRoomGroupListResponse
     */
    async queryMeetingRoomGroupListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryMeetingRoomGroupList",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/groupLists`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryMeetingRoomGroupListResponse({}));
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomGroupListRequest
     * @return QueryMeetingRoomGroupListResponse
     */
    async queryMeetingRoomGroupList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryMeetingRoomGroupListHeaders({});
        return await this.queryMeetingRoomGroupListWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomListRequest
     * @param headers QueryMeetingRoomListHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryMeetingRoomListResponse
     */
    async queryMeetingRoomListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryMeetingRoomList",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/meetingRoomLists`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryMeetingRoomListResponse({}));
    }
    /**
     * @summary 
     *
     * @param request QueryMeetingRoomListRequest
     * @return QueryMeetingRoomListResponse
     */
    async queryMeetingRoomList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryMeetingRoomListHeaders({});
        return await this.queryMeetingRoomListWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request RemoveSuperUserMeetingRoomRequest
     * @param headers RemoveSuperUserMeetingRoomHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RemoveSuperUserMeetingRoomResponse
     */
    async removeSuperUserMeetingRoomWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.roomId)) {
            query["roomId"] = request.roomId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "RemoveSuperUserMeetingRoom",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/meetingRooms/superUsers/remove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RemoveSuperUserMeetingRoomResponse({}));
    }
    /**
     * @summary 
     *
     * @param request RemoveSuperUserMeetingRoomRequest
     * @return RemoveSuperUserMeetingRoomResponse
     */
    async removeSuperUserMeetingRoom(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RemoveSuperUserMeetingRoomHeaders({});
        return await this.removeSuperUserMeetingRoomWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request SetSuperUserMeetingRoomRequest
     * @param headers SetSuperUserMeetingRoomHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SetSuperUserMeetingRoomResponse
     */
    async setSuperUserMeetingRoomWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.deptIdWhiteList)) {
            body["deptIdWhiteList"] = request.deptIdWhiteList;
        }
        if (!tea_util_1.default.isUnset(request.roomId)) {
            body["roomId"] = request.roomId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        if (!tea_util_1.default.isUnset(request.userIdWhiteList)) {
            body["userIdWhiteList"] = request.userIdWhiteList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SetSuperUserMeetingRoom",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/meetingRooms/superUsers/set`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SetSuperUserMeetingRoomResponse({}));
    }
    /**
     * @summary 
     *
     * @param request SetSuperUserMeetingRoomRequest
     * @return SetSuperUserMeetingRoomResponse
     */
    async setSuperUserMeetingRoom(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SetSuperUserMeetingRoomHeaders({});
        return await this.setSuperUserMeetingRoomWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateDeviceCustomTemplateRequest
     * @param headers UpdateDeviceCustomTemplateHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateDeviceCustomTemplateResponse
     */
    async updateDeviceCustomTemplateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bgImgList)) {
            body["bgImgList"] = request.bgImgList;
        }
        if (!tea_util_1.default.isUnset(request.bgType)) {
            body["bgType"] = request.bgType;
        }
        if (!tea_util_1.default.isUnset(request.bgUrl)) {
            body["bgUrl"] = request.bgUrl;
        }
        if (!tea_util_1.default.isUnset(request.customDoc)) {
            body["customDoc"] = request.customDoc;
        }
        if (!tea_util_1.default.isUnset(request.desensitizeUserName)) {
            body["desensitizeUserName"] = request.desensitizeUserName;
        }
        if (!tea_util_1.default.isUnset(request.deviceUnionIds)) {
            body["deviceUnionIds"] = request.deviceUnionIds;
        }
        if (!tea_util_1.default.isUnset(request.groupIds)) {
            body["groupIds"] = request.groupIds;
        }
        if (!tea_util_1.default.isUnset(request.hideServerCodeWhenProjecting)) {
            body["hideServerCodeWhenProjecting"] = request.hideServerCodeWhenProjecting;
        }
        if (!tea_util_1.default.isUnset(request.instruction)) {
            body["instruction"] = request.instruction;
        }
        if (!tea_util_1.default.isUnset(request.isPicTop)) {
            body["isPicTop"] = request.isPicTop;
        }
        if (!tea_util_1.default.isUnset(request.logo)) {
            body["logo"] = request.logo;
        }
        if (!tea_util_1.default.isUnset(request.orgName)) {
            body["orgName"] = request.orgName;
        }
        if (!tea_util_1.default.isUnset(request.picturePlayInterval)) {
            body["picturePlayInterval"] = request.picturePlayInterval;
        }
        if (!tea_util_1.default.isUnset(request.roomIds)) {
            body["roomIds"] = request.roomIds;
        }
        if (!tea_util_1.default.isUnset(request.showCalendarCard)) {
            body["showCalendarCard"] = request.showCalendarCard;
        }
        if (!tea_util_1.default.isUnset(request.showCalendarTitle)) {
            body["showCalendarTitle"] = request.showCalendarTitle;
        }
        if (!tea_util_1.default.isUnset(request.showFunctionCard)) {
            body["showFunctionCard"] = request.showFunctionCard;
        }
        if (!tea_util_1.default.isUnset(request.templateId)) {
            body["templateId"] = request.templateId;
        }
        if (!tea_util_1.default.isUnset(request.templateName)) {
            body["templateName"] = request.templateName;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateDeviceCustomTemplate",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/devices/screens/templates`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateDeviceCustomTemplateResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateDeviceCustomTemplateRequest
     * @return UpdateDeviceCustomTemplateResponse
     */
    async updateDeviceCustomTemplate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateDeviceCustomTemplateHeaders({});
        return await this.updateDeviceCustomTemplateWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateMeetingRoomRequest
     * @param headers UpdateMeetingRoomHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateMeetingRoomResponse
     */
    async updateMeetingRoomWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.enableCycleReservation)) {
            body["enableCycleReservation"] = request.enableCycleReservation;
        }
        if (!tea_util_1.default.isUnset(request.groupId)) {
            body["groupId"] = request.groupId;
        }
        if (!tea_util_1.default.isUnset(request.isvRoomId)) {
            body["isvRoomId"] = request.isvRoomId;
        }
        if (!tea_util_1.default.isUnset(request.reservationAuthority)) {
            body["reservationAuthority"] = request.reservationAuthority;
        }
        if (!tea_util_1.default.isUnset(request.roomCapacity)) {
            body["roomCapacity"] = request.roomCapacity;
        }
        if (!tea_util_1.default.isUnset(request.roomId)) {
            body["roomId"] = request.roomId;
        }
        if (!tea_util_1.default.isUnset(request.roomLabelIds)) {
            body["roomLabelIds"] = request.roomLabelIds;
        }
        if (!tea_util_1.default.isUnset(request.roomLocation)) {
            body["roomLocation"] = request.roomLocation;
        }
        if (!tea_util_1.default.isUnset(request.roomName)) {
            body["roomName"] = request.roomName;
        }
        if (!tea_util_1.default.isUnset(request.roomPicture)) {
            body["roomPicture"] = request.roomPicture;
        }
        if (!tea_util_1.default.isUnset(request.roomStatus)) {
            body["roomStatus"] = request.roomStatus;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateMeetingRoom",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/meetingRooms`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateMeetingRoomResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateMeetingRoomRequest
     * @return UpdateMeetingRoomResponse
     */
    async updateMeetingRoom(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateMeetingRoomHeaders({});
        return await this.updateMeetingRoomWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateMeetingRoomGroupRequest
     * @param headers UpdateMeetingRoomGroupHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateMeetingRoomGroupResponse
     */
    async updateMeetingRoomGroupWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupId)) {
            body["groupId"] = request.groupId;
        }
        if (!tea_util_1.default.isUnset(request.groupName)) {
            body["groupName"] = request.groupName;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateMeetingRoomGroup",
            version: "rooms_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/rooms/groups`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateMeetingRoomGroupResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateMeetingRoomGroupRequest
     * @return UpdateMeetingRoomGroupResponse
     */
    async updateMeetingRoomGroup(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateMeetingRoomGroupHeaders({});
        return await this.updateMeetingRoomGroupWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
