// @alicloud/dingtalk@2.1.16/dist/crm_2_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRelationUkSettingResponseBodyResult = exports.GetRelationUkSettingResponseBodyResultFormUkSettings = exports.GetRelationUkSettingResponseBodyResultFormUkSettingsFieldList = exports.GetRelationUkSettingResponse = exports.GetRelationUkSettingResponseBody = exports.GetRelationUkSettingRequest = exports.GetRelationUkSettingHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class GetRelationUkSettingHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetRelationUkSettingHeaders = GetRelationUkSettingHeaders;
class GetRelationUkSettingRequest extends $tea.Model {
    static names() {
        return {
            relationType: 'relationType',
        };
    }
    static types() {
        return {
            relationType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetRelationUkSettingRequest = GetRelationUkSettingRequest;
class GetRelationUkSettingResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetRelationUkSettingResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetRelationUkSettingResponseBody = GetRelationUkSettingResponseBody;
class GetRelationUkSettingResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetRelationUkSettingResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetRelationUkSettingResponse = GetRelationUkSettingResponse;
class GetRelationUkSettingResponseBodyResultFormUkSettingsFieldList extends $tea.Model {
    static names() {
        return {
            bizAlias: 'bizAlias',
            fieldId: 'fieldId',
        };
    }
    static types() {
        return {
            bizAlias: 'string',
            fieldId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetRelationUkSettingResponseBodyResultFormUkSettingsFieldList = GetRelationUkSettingResponseBodyResultFormUkSettingsFieldList;
class GetRelationUkSettingResponseBodyResultFormUkSettings extends $tea.Model {
    static names() {
        return {
            fieldList: 'fieldList',
        };
    }
    static types() {
        return {
            fieldList: { 'type': 'array', 'itemType': GetRelationUkSettingResponseBodyResultFormUkSettingsFieldList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetRelationUkSettingResponseBodyResultFormUkSettings = GetRelationUkSettingResponseBodyResultFormUkSettings;
class GetRelationUkSettingResponseBodyResult extends $tea.Model {
    static names() {
        return {
            formUkSettings: 'formUkSettings',
            headerFieldIds: 'headerFieldIds',
        };
    }
    static types() {
        return {
            formUkSettings: { 'type': 'array', 'itemType': GetRelationUkSettingResponseBodyResultFormUkSettings },
            headerFieldIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetRelationUkSettingResponseBodyResult = GetRelationUkSettingResponseBodyResult;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 获取关系数据查重规则
     *
     * @param request GetRelationUkSettingRequest
     * @param headers GetRelationUkSettingHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetRelationUkSettingResponse
     */
    async getRelationUkSettingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.relationType)) {
            query["relationType"] = request.relationType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetRelationUkSetting",
            version: "crm_2.0",
            protocol: "HTTP",
            pathname: `/v2.0/crm/relationUkSettings`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetRelationUkSettingResponse({}));
    }
    /**
     * @summary 获取关系数据查重规则
     *
     * @param request GetRelationUkSettingRequest
     * @return GetRelationUkSettingResponse
     */
    async getRelationUkSetting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetRelationUkSettingHeaders({});
        return await this.getRelationUkSettingWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
