// @alicloud/dingtalk@2.1.16/dist/content_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PageFeedResponseBodyFeedList = exports.ListItemUserDataResponseBodyStudyInfos = exports.GetFeedResponseBodyFeedItem = exports.CreateFeedRequestFeedInfo = exports.CreateFeedRequestFeedInfoRecommends = exports.CreateFeedRequestFeedInfoMediaContents = exports.CreateFeedRequestCourseInfo = exports.CreateFeedRequestCourseInfoPayInfo = exports.CreateFeedRequestCourseInfoPayInfoDiscountInfo = exports.CreateFeedRequestCourseInfoPayInfoCsUserInfo = exports.CreateFeedRequestCourseInfoLectorUserInfo = exports.PageFeedResponse = exports.PageFeedResponseBody = exports.PageFeedRequest = exports.PageFeedHeaders = exports.ListItemUserDataResponse = exports.ListItemUserDataResponseBody = exports.ListItemUserDataRequest = exports.ListItemUserDataHeaders = exports.GetMediaCerficateResponse = exports.GetMediaCerficateResponseBody = exports.GetMediaCerficateRequest = exports.GetMediaCerficateHeaders = exports.GetFeedResponse = exports.GetFeedResponseBody = exports.GetFeedRequest = exports.GetFeedHeaders = exports.CreateFeedResponse = exports.CreateFeedResponseBody = exports.CreateFeedRequest = exports.CreateFeedHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CreateFeedHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedHeaders = CreateFeedHeaders;
class CreateFeedRequest extends $tea.Model {
    static names() {
        return {
            courseInfo: 'courseInfo',
            createUserId: 'createUserId',
            feedInfo: 'feedInfo',
        };
    }
    static types() {
        return {
            courseInfo: CreateFeedRequestCourseInfo,
            createUserId: 'string',
            feedInfo: CreateFeedRequestFeedInfo,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedRequest = CreateFeedRequest;
class CreateFeedResponseBody extends $tea.Model {
    static names() {
        return {
            feedId: 'feedId',
        };
    }
    static types() {
        return {
            feedId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedResponseBody = CreateFeedResponseBody;
class CreateFeedResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateFeedResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedResponse = CreateFeedResponse;
class GetFeedHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFeedHeaders = GetFeedHeaders;
class GetFeedRequest extends $tea.Model {
    static names() {
        return {
            mcnId: 'mcnId',
        };
    }
    static types() {
        return {
            mcnId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFeedRequest = GetFeedRequest;
class GetFeedResponseBody extends $tea.Model {
    static names() {
        return {
            feedId: 'feedId',
            feedItem: 'feedItem',
        };
    }
    static types() {
        return {
            feedId: 'string',
            feedItem: { 'type': 'array', 'itemType': GetFeedResponseBodyFeedItem },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFeedResponseBody = GetFeedResponseBody;
class GetFeedResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetFeedResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFeedResponse = GetFeedResponse;
class GetMediaCerficateHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaCerficateHeaders = GetMediaCerficateHeaders;
class GetMediaCerficateRequest extends $tea.Model {
    static names() {
        return {
            fileName: 'fileName',
            mcnId: 'mcnId',
            mediaId: 'mediaId',
            mediaIntroduction: 'mediaIntroduction',
            mediaTitle: 'mediaTitle',
            thumbUrl: 'thumbUrl',
            userId: 'userId',
        };
    }
    static types() {
        return {
            fileName: 'string',
            mcnId: 'string',
            mediaId: 'string',
            mediaIntroduction: 'string',
            mediaTitle: 'string',
            thumbUrl: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaCerficateRequest = GetMediaCerficateRequest;
class GetMediaCerficateResponseBody extends $tea.Model {
    static names() {
        return {
            mediaId: 'mediaId',
            ossAccessKeyId: 'ossAccessKeyId',
            ossAccessKeySecret: 'ossAccessKeySecret',
            ossBucketName: 'ossBucketName',
            ossEndpoint: 'ossEndpoint',
            ossExpiration: 'ossExpiration',
            ossFileName: 'ossFileName',
            ossSecurityToken: 'ossSecurityToken',
        };
    }
    static types() {
        return {
            mediaId: 'string',
            ossAccessKeyId: 'string',
            ossAccessKeySecret: 'string',
            ossBucketName: 'string',
            ossEndpoint: 'string',
            ossExpiration: 'string',
            ossFileName: 'string',
            ossSecurityToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaCerficateResponseBody = GetMediaCerficateResponseBody;
class GetMediaCerficateResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetMediaCerficateResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaCerficateResponse = GetMediaCerficateResponse;
class ListItemUserDataHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListItemUserDataHeaders = ListItemUserDataHeaders;
class ListItemUserDataRequest extends $tea.Model {
    static names() {
        return {
            body: 'body',
        };
    }
    static types() {
        return {
            body: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListItemUserDataRequest = ListItemUserDataRequest;
class ListItemUserDataResponseBody extends $tea.Model {
    static names() {
        return {
            studyInfos: 'studyInfos',
        };
    }
    static types() {
        return {
            studyInfos: { 'type': 'array', 'itemType': ListItemUserDataResponseBodyStudyInfos },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListItemUserDataResponseBody = ListItemUserDataResponseBody;
class ListItemUserDataResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListItemUserDataResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListItemUserDataResponse = ListItemUserDataResponse;
class PageFeedHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageFeedHeaders = PageFeedHeaders;
class PageFeedRequest extends $tea.Model {
    static names() {
        return {
            body: 'body',
            maxResults: 'maxResults',
            mcnId: 'mcnId',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            body: { 'type': 'array', 'itemType': 'string' },
            maxResults: 'number',
            mcnId: 'string',
            nextToken: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageFeedRequest = PageFeedRequest;
class PageFeedResponseBody extends $tea.Model {
    static names() {
        return {
            feedList: 'feedList',
            hasNext: 'hasNext',
            nextCursor: 'nextCursor',
        };
    }
    static types() {
        return {
            feedList: { 'type': 'array', 'itemType': PageFeedResponseBodyFeedList },
            hasNext: 'boolean',
            nextCursor: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageFeedResponseBody = PageFeedResponseBody;
class PageFeedResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PageFeedResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageFeedResponse = PageFeedResponse;
class CreateFeedRequestCourseInfoLectorUserInfo extends $tea.Model {
    static names() {
        return {
            avatar: 'avatar',
            name: 'name',
            userId: 'userId',
        };
    }
    static types() {
        return {
            avatar: 'string',
            name: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedRequestCourseInfoLectorUserInfo = CreateFeedRequestCourseInfoLectorUserInfo;
class CreateFeedRequestCourseInfoPayInfoCsUserInfo extends $tea.Model {
    static names() {
        return {
            avatar: 'avatar',
            name: 'name',
            userId: 'userId',
        };
    }
    static types() {
        return {
            avatar: 'string',
            name: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedRequestCourseInfoPayInfoCsUserInfo = CreateFeedRequestCourseInfoPayInfoCsUserInfo;
class CreateFeedRequestCourseInfoPayInfoDiscountInfo extends $tea.Model {
    static names() {
        return {
            endTimeMillis: 'endTimeMillis',
            price: 'price',
            startTimeMillis: 'startTimeMillis',
        };
    }
    static types() {
        return {
            endTimeMillis: 'number',
            price: 'number',
            startTimeMillis: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedRequestCourseInfoPayInfoDiscountInfo = CreateFeedRequestCourseInfoPayInfoDiscountInfo;
class CreateFeedRequestCourseInfoPayInfo extends $tea.Model {
    static names() {
        return {
            csUserInfo: 'csUserInfo',
            discountInfo: 'discountInfo',
            price: 'price',
        };
    }
    static types() {
        return {
            csUserInfo: CreateFeedRequestCourseInfoPayInfoCsUserInfo,
            discountInfo: CreateFeedRequestCourseInfoPayInfoDiscountInfo,
            price: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedRequestCourseInfoPayInfo = CreateFeedRequestCourseInfoPayInfo;
class CreateFeedRequestCourseInfo extends $tea.Model {
    static names() {
        return {
            lectorUserInfo: 'lectorUserInfo',
            payInfo: 'payInfo',
            studyGroupName: 'studyGroupName',
        };
    }
    static types() {
        return {
            lectorUserInfo: CreateFeedRequestCourseInfoLectorUserInfo,
            payInfo: CreateFeedRequestCourseInfoPayInfo,
            studyGroupName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedRequestCourseInfo = CreateFeedRequestCourseInfo;
class CreateFeedRequestFeedInfoMediaContents extends $tea.Model {
    static names() {
        return {
            mediaId: 'mediaId',
            title: 'title',
            type: 'type',
        };
    }
    static types() {
        return {
            mediaId: 'string',
            title: 'string',
            type: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedRequestFeedInfoMediaContents = CreateFeedRequestFeedInfoMediaContents;
class CreateFeedRequestFeedInfoRecommends extends $tea.Model {
    static names() {
        return {
            objectId: 'objectId',
            type: 'type',
        };
    }
    static types() {
        return {
            objectId: 'string',
            type: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedRequestFeedInfoRecommends = CreateFeedRequestFeedInfoRecommends;
class CreateFeedRequestFeedInfo extends $tea.Model {
    static names() {
        return {
            actionType: 'actionType',
            belongsTo: 'belongsTo',
            feedCategory: 'feedCategory',
            feedId: 'feedId',
            feedTag: 'feedTag',
            feedType: 'feedType',
            industryId: 'industryId',
            introduction: 'introduction',
            introductionPicUrl: 'introductionPicUrl',
            mcnId: 'mcnId',
            mediaContents: 'mediaContents',
            recommends: 'recommends',
            thumbUrl: 'thumbUrl',
            title: 'title',
        };
    }
    static types() {
        return {
            actionType: 'number',
            belongsTo: 'number',
            feedCategory: 'number',
            feedId: 'string',
            feedTag: 'string',
            feedType: 'number',
            industryId: 'number',
            introduction: 'string',
            introductionPicUrl: 'string',
            mcnId: 'string',
            mediaContents: { 'type': 'array', 'itemType': CreateFeedRequestFeedInfoMediaContents },
            recommends: { 'type': 'array', 'itemType': CreateFeedRequestFeedInfoRecommends },
            thumbUrl: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateFeedRequestFeedInfo = CreateFeedRequestFeedInfo;
class GetFeedResponseBodyFeedItem extends $tea.Model {
    static names() {
        return {
            durationMillis: 'durationMillis',
            feedContentType: 'feedContentType',
            itemId: 'itemId',
            title: 'title',
            url: 'url',
        };
    }
    static types() {
        return {
            durationMillis: 'number',
            feedContentType: 'number',
            itemId: 'string',
            title: 'string',
            url: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFeedResponseBodyFeedItem = GetFeedResponseBodyFeedItem;
class ListItemUserDataResponseBodyStudyInfos extends $tea.Model {
    static names() {
        return {
            durationMillis: 'durationMillis',
            uid: 'uid',
        };
    }
    static types() {
        return {
            durationMillis: 'number',
            uid: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListItemUserDataResponseBodyStudyInfos = ListItemUserDataResponseBodyStudyInfos;
class PageFeedResponseBodyFeedList extends $tea.Model {
    static names() {
        return {
            feedCategory: 'feedCategory',
            feedId: 'feedId',
            feedType: 'feedType',
            name: 'name',
            thumbUrl: 'thumbUrl',
            url: 'url',
        };
    }
    static types() {
        return {
            feedCategory: 'string',
            feedId: 'string',
            feedType: 'number',
            name: 'string',
            thumbUrl: 'string',
            url: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageFeedResponseBodyFeedList = PageFeedResponseBodyFeedList;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 创建内容
     *
     * @param request CreateFeedRequest
     * @param headers CreateFeedHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateFeedResponse
     */
    async createFeedWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.courseInfo)) {
            body["courseInfo"] = request.courseInfo;
        }
        if (!tea_util_1.default.isUnset(request.createUserId)) {
            body["createUserId"] = request.createUserId;
        }
        if (!tea_util_1.default.isUnset(request.feedInfo)) {
            body["feedInfo"] = request.feedInfo;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateFeed",
            version: "content_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/content/feeds`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateFeedResponse({}));
    }
    /**
     * @summary 创建内容
     *
     * @param request CreateFeedRequest
     * @return CreateFeedResponse
     */
    async createFeed(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateFeedHeaders({});
        return await this.createFeedWithOptions(request, headers, runtime);
    }
    /**
     * @summary 获取feed的详细信息，包括子课程的信息
     *
     * @param request GetFeedRequest
     * @param headers GetFeedHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetFeedResponse
     */
    async getFeedWithOptions(feedId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.mcnId)) {
            query["mcnId"] = request.mcnId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetFeed",
            version: "content_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/content/feeds/${feedId}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetFeedResponse({}));
    }
    /**
     * @summary 获取feed的详细信息，包括子课程的信息
     *
     * @param request GetFeedRequest
     * @return GetFeedResponse
     */
    async getFeed(feedId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetFeedHeaders({});
        return await this.getFeedWithOptions(feedId, request, headers, runtime);
    }
    /**
     * @summary 获取oss上传凭证
     *
     * @param request GetMediaCerficateRequest
     * @param headers GetMediaCerficateHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetMediaCerficateResponse
     */
    async getMediaCerficateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.fileName)) {
            query["fileName"] = request.fileName;
        }
        if (!tea_util_1.default.isUnset(request.mcnId)) {
            query["mcnId"] = request.mcnId;
        }
        if (!tea_util_1.default.isUnset(request.mediaId)) {
            query["mediaId"] = request.mediaId;
        }
        if (!tea_util_1.default.isUnset(request.mediaIntroduction)) {
            query["mediaIntroduction"] = request.mediaIntroduction;
        }
        if (!tea_util_1.default.isUnset(request.mediaTitle)) {
            query["mediaTitle"] = request.mediaTitle;
        }
        if (!tea_util_1.default.isUnset(request.thumbUrl)) {
            query["thumbUrl"] = request.thumbUrl;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetMediaCerficate",
            version: "content_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/content/media/cerficates`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetMediaCerficateResponse({}));
    }
    /**
     * @summary 获取oss上传凭证
     *
     * @param request GetMediaCerficateRequest
     * @return GetMediaCerficateResponse
     */
    async getMediaCerficate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetMediaCerficateHeaders({});
        return await this.getMediaCerficateWithOptions(request, headers, runtime);
    }
    /**
     * @summary 展示机构内观看内容的统计信息
     *
     * @param request ListItemUserDataRequest
     * @param headers ListItemUserDataHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListItemUserDataResponse
     */
    async listItemUserDataWithOptions(itemId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: request.body,
        });
        let params = new $OpenApi.Params({
            action: "ListItemUserData",
            version: "content_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/content/feeds/items/${itemId}/userStatistics/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListItemUserDataResponse({}));
    }
    /**
     * @summary 展示机构内观看内容的统计信息
     *
     * @param request ListItemUserDataRequest
     * @return ListItemUserDataResponse
     */
    async listItemUserData(itemId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListItemUserDataHeaders({});
        return await this.listItemUserDataWithOptions(itemId, request, headers, runtime);
    }
    /**
     * @summary 获取机构下课程列表
     *
     * @param request PageFeedRequest
     * @param headers PageFeedHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return PageFeedResponse
     */
    async pageFeedWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.mcnId)) {
            query["mcnId"] = request.mcnId;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: request.body,
        });
        let params = new $OpenApi.Params({
            action: "PageFeed",
            version: "content_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/content/feeds/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PageFeedResponse({}));
    }
    /**
     * @summary 获取机构下课程列表
     *
     * @param request PageFeedRequest
     * @return PageFeedResponse
     */
    async pageFeed(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PageFeedHeaders({});
        return await this.pageFeedWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
