// @alicloud/dingtalk@2.1.16/dist/live_activities_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendLiveActivityRequestActivityEventOption = exports.SendLiveActivityRequestActivityEventData = exports.PushLiveActivityRequestActivityEventOption = exports.PushLiveActivityRequestActivityEventData = exports.SendLiveActivityResponse = exports.SendLiveActivityResponseBody = exports.SendLiveActivityRequest = exports.SendLiveActivityHeaders = exports.PushLiveActivityResponse = exports.PushLiveActivityResponseBody = exports.PushLiveActivityRequest = exports.PushLiveActivityHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class PushLiveActivityHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PushLiveActivityHeaders = PushLiveActivityHeaders;
class PushLiveActivityRequest extends $tea.Model {
    static names() {
        return {
            activityEventData: 'activityEventData',
            activityEventOption: 'activityEventOption',
            activityId: 'activityId',
        };
    }
    static types() {
        return {
            activityEventData: PushLiveActivityRequestActivityEventData,
            activityEventOption: PushLiveActivityRequestActivityEventOption,
            activityId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PushLiveActivityRequest = PushLiveActivityRequest;
class PushLiveActivityResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PushLiveActivityResponseBody = PushLiveActivityResponseBody;
class PushLiveActivityResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PushLiveActivityResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PushLiveActivityResponse = PushLiveActivityResponse;
class SendLiveActivityHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveActivityHeaders = SendLiveActivityHeaders;
class SendLiveActivityRequest extends $tea.Model {
    static names() {
        return {
            activityEventData: 'activityEventData',
            activityEventOption: 'activityEventOption',
            activityId: 'activityId',
        };
    }
    static types() {
        return {
            activityEventData: SendLiveActivityRequestActivityEventData,
            activityEventOption: SendLiveActivityRequestActivityEventOption,
            activityId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveActivityRequest = SendLiveActivityRequest;
class SendLiveActivityResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveActivityResponseBody = SendLiveActivityResponseBody;
class SendLiveActivityResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendLiveActivityResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveActivityResponse = SendLiveActivityResponse;
class PushLiveActivityRequestActivityEventData extends $tea.Model {
    static names() {
        return {
            i18nContentState: 'i18nContentState',
            templateId: 'templateId',
        };
    }
    static types() {
        return {
            i18nContentState: 'any',
            templateId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PushLiveActivityRequestActivityEventData = PushLiveActivityRequestActivityEventData;
class PushLiveActivityRequestActivityEventOption extends $tea.Model {
    static names() {
        return {
            dismissalDate: 'dismissalDate',
            pushType: 'pushType',
            sendDate: 'sendDate',
            staleDate: 'staleDate',
        };
    }
    static types() {
        return {
            dismissalDate: 'number',
            pushType: 'string',
            sendDate: 'number',
            staleDate: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PushLiveActivityRequestActivityEventOption = PushLiveActivityRequestActivityEventOption;
class SendLiveActivityRequestActivityEventData extends $tea.Model {
    static names() {
        return {
            i18nContentState: 'i18nContentState',
            templateId: 'templateId',
        };
    }
    static types() {
        return {
            i18nContentState: 'any',
            templateId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveActivityRequestActivityEventData = SendLiveActivityRequestActivityEventData;
class SendLiveActivityRequestActivityEventOption extends $tea.Model {
    static names() {
        return {
            dismissalDate: 'dismissalDate',
            pushType: 'pushType',
            sendDate: 'sendDate',
            staleDate: 'staleDate',
        };
    }
    static types() {
        return {
            dismissalDate: 'number',
            pushType: 'string',
            sendDate: 'number',
            staleDate: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveActivityRequestActivityEventOption = SendLiveActivityRequestActivityEventOption;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 实时活动发送接口
     *
     * @param request PushLiveActivityRequest
     * @param headers PushLiveActivityHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return PushLiveActivityResponse
     */
    async pushLiveActivityWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.activityEventData)) {
            body["activityEventData"] = request.activityEventData;
        }
        if (!tea_util_1.default.isUnset(request.activityEventOption)) {
            body["activityEventOption"] = request.activityEventOption;
        }
        if (!tea_util_1.default.isUnset(request.activityId)) {
            body["activityId"] = request.activityId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PushLiveActivity",
            version: "liveActivities_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/liveActivities/push`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PushLiveActivityResponse({}));
    }
    /**
     * @summary 实时活动发送接口
     *
     * @param request PushLiveActivityRequest
     * @return PushLiveActivityResponse
     */
    async pushLiveActivity(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PushLiveActivityHeaders({});
        return await this.pushLiveActivityWithOptions(request, headers, runtime);
    }
    /**
     * @summary 发送实时活动
     *
     * @param request SendLiveActivityRequest
     * @param headers SendLiveActivityHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SendLiveActivityResponse
     */
    async sendLiveActivityWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.activityEventData)) {
            body["activityEventData"] = request.activityEventData;
        }
        if (!tea_util_1.default.isUnset(request.activityEventOption)) {
            body["activityEventOption"] = request.activityEventOption;
        }
        if (!tea_util_1.default.isUnset(request.activityId)) {
            body["activityId"] = request.activityId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendLiveActivity",
            version: "liveActivities_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/liveActivities/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendLiveActivityResponse({}));
    }
    /**
     * @summary 发送实时活动
     *
     * @param request SendLiveActivityRequest
     * @return SendLiveActivityResponse
     */
    async sendLiveActivity(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendLiveActivityHeaders({});
        return await this.sendLiveActivityWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
