// @alicloud/dingtalk@2.1.16/dist/flashmsg_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendMessageTipResponseBodyResult = exports.SendMessageTipRequestPublicField = exports.SendMessageTipRequestDefaultView = exports.SendDingTipRequestLink = exports.QueryPluginRuleResponseBodyResult = exports.QueryPluginRuleResponseBodyResultData = exports.GetConversationResponseBodyResult = exports.GetBaseProfileListResponseBodyResult = exports.AddPluginRuleRequestRules = exports.SendMessageTipResponse = exports.SendMessageTipResponseBody = exports.SendMessageTipRequest = exports.SendMessageTipHeaders = exports.SendDingTipResponse = exports.SendDingTipResponseBody = exports.SendDingTipRequest = exports.SendDingTipHeaders = exports.QueryPluginRuleResponse = exports.QueryPluginRuleResponseBody = exports.QueryPluginRuleRequest = exports.QueryPluginRuleHeaders = exports.GetMemberListResponse = exports.GetMemberListResponseBody = exports.GetMemberListRequest = exports.GetMemberListHeaders = exports.GetConversationResponse = exports.GetConversationResponseBody = exports.GetConversationRequest = exports.GetConversationHeaders = exports.GetBaseProfileListResponse = exports.GetBaseProfileListResponseBody = exports.GetBaseProfileListRequest = exports.GetBaseProfileListHeaders = exports.DeletePlguinRuleResponse = exports.DeletePlguinRuleResponseBody = exports.DeletePlguinRuleRequest = exports.DeletePlguinRuleHeaders = exports.AddPluginRuleResponse = exports.AddPluginRuleResponseBody = exports.AddPluginRuleRequest = exports.AddPluginRuleHeaders = exports.PrivateFieldMapValue = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class PrivateFieldMapValue extends $tea.Model {
    static names() {
        return {
            tipTitle: 'tipTitle',
            isDingSend: 'isDingSend',
            isRead: 'isRead',
            buttonStatus: 'buttonStatus',
            extension: 'extension',
        };
    }
    static types() {
        return {
            tipTitle: 'string',
            isDingSend: 'boolean',
            isRead: 'boolean',
            buttonStatus: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PrivateFieldMapValue = PrivateFieldMapValue;
class AddPluginRuleHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddPluginRuleHeaders = AddPluginRuleHeaders;
class AddPluginRuleRequest extends $tea.Model {
    static names() {
        return {
            chatType: 'chatType',
            code: 'code',
            itemType: 'itemType',
            rules: 'rules',
            userId: 'userId',
        };
    }
    static types() {
        return {
            chatType: 'string',
            code: 'string',
            itemType: 'string',
            rules: { 'type': 'array', 'itemType': AddPluginRuleRequestRules },
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddPluginRuleRequest = AddPluginRuleRequest;
class AddPluginRuleResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddPluginRuleResponseBody = AddPluginRuleResponseBody;
class AddPluginRuleResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddPluginRuleResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddPluginRuleResponse = AddPluginRuleResponse;
class DeletePlguinRuleHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeletePlguinRuleHeaders = DeletePlguinRuleHeaders;
class DeletePlguinRuleRequest extends $tea.Model {
    static names() {
        return {
            bizIdList: 'bizIdList',
            userId: 'userId',
        };
    }
    static types() {
        return {
            bizIdList: { 'type': 'array', 'itemType': 'string' },
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeletePlguinRuleRequest = DeletePlguinRuleRequest;
class DeletePlguinRuleResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeletePlguinRuleResponseBody = DeletePlguinRuleResponseBody;
class DeletePlguinRuleResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeletePlguinRuleResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeletePlguinRuleResponse = DeletePlguinRuleResponse;
class GetBaseProfileListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetBaseProfileListHeaders = GetBaseProfileListHeaders;
class GetBaseProfileListRequest extends $tea.Model {
    static names() {
        return {
            body: 'body',
        };
    }
    static types() {
        return {
            body: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetBaseProfileListRequest = GetBaseProfileListRequest;
class GetBaseProfileListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': GetBaseProfileListResponseBodyResult },
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetBaseProfileListResponseBody = GetBaseProfileListResponseBody;
class GetBaseProfileListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetBaseProfileListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetBaseProfileListResponse = GetBaseProfileListResponse;
class GetConversationHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetConversationHeaders = GetConversationHeaders;
class GetConversationRequest extends $tea.Model {
    static names() {
        return {
            openConversationId: 'openConversationId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetConversationRequest = GetConversationRequest;
class GetConversationResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: GetConversationResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetConversationResponseBody = GetConversationResponseBody;
class GetConversationResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetConversationResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetConversationResponse = GetConversationResponse;
class GetMemberListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMemberListHeaders = GetMemberListHeaders;
class GetMemberListRequest extends $tea.Model {
    static names() {
        return {
            openConversationId: 'openConversationId',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            userId: 'userId',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
            pageNumber: 'number',
            pageSize: 'number',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMemberListRequest = GetMemberListRequest;
class GetMemberListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMemberListResponseBody = GetMemberListResponseBody;
class GetMemberListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetMemberListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMemberListResponse = GetMemberListResponse;
class QueryPluginRuleHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryPluginRuleHeaders = QueryPluginRuleHeaders;
class QueryPluginRuleRequest extends $tea.Model {
    static names() {
        return {
            chatType: 'chatType',
            code: 'code',
            itemId: 'itemId',
            itemType: 'itemType',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
        };
    }
    static types() {
        return {
            chatType: 'string',
            code: 'string',
            itemId: 'string',
            itemType: 'string',
            pageNumber: 'number',
            pageSize: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryPluginRuleRequest = QueryPluginRuleRequest;
class QueryPluginRuleResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: QueryPluginRuleResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryPluginRuleResponseBody = QueryPluginRuleResponseBody;
class QueryPluginRuleResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryPluginRuleResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryPluginRuleResponse = QueryPluginRuleResponse;
class SendDingTipHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendDingTipHeaders = SendDingTipHeaders;
class SendDingTipRequest extends $tea.Model {
    static names() {
        return {
            extension: 'extension',
            link: 'link',
            messageId: 'messageId',
            receiverUserId: 'receiverUserId',
            senderUserId: 'senderUserId',
            textContent: 'textContent',
        };
    }
    static types() {
        return {
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            link: SendDingTipRequestLink,
            messageId: 'string',
            receiverUserId: { 'type': 'array', 'itemType': 'string' },
            senderUserId: 'string',
            textContent: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendDingTipRequest = SendDingTipRequest;
class SendDingTipResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendDingTipResponseBody = SendDingTipResponseBody;
class SendDingTipResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendDingTipResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendDingTipResponse = SendDingTipResponse;
class SendMessageTipHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendMessageTipHeaders = SendMessageTipHeaders;
class SendMessageTipRequest extends $tea.Model {
    static names() {
        return {
            defaultView: 'defaultView',
            messageId: 'messageId',
            openConversationId: 'openConversationId',
            privateFieldMap: 'privateFieldMap',
            publicField: 'publicField',
            receiverUserId: 'receiverUserId',
            senderUserId: 'senderUserId',
        };
    }
    static types() {
        return {
            defaultView: SendMessageTipRequestDefaultView,
            messageId: 'string',
            openConversationId: 'string',
            privateFieldMap: { 'type': 'map', 'keyType': 'string', 'valueType': PrivateFieldMapValue },
            publicField: SendMessageTipRequestPublicField,
            receiverUserId: { 'type': 'array', 'itemType': 'string' },
            senderUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendMessageTipRequest = SendMessageTipRequest;
class SendMessageTipResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: SendMessageTipResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendMessageTipResponseBody = SendMessageTipResponseBody;
class SendMessageTipResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendMessageTipResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendMessageTipResponse = SendMessageTipResponse;
class AddPluginRuleRequestRules extends $tea.Model {
    static names() {
        return {
            itemId: 'itemId',
            itemName: 'itemName',
        };
    }
    static types() {
        return {
            itemId: 'string',
            itemName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddPluginRuleRequestRules = AddPluginRuleRequestRules;
class GetBaseProfileListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            avatarMediaId: 'avatarMediaId',
            nick: 'nick',
            nickPinyin: 'nickPinyin',
            userId: 'userId',
        };
    }
    static types() {
        return {
            avatarMediaId: 'string',
            nick: 'string',
            nickPinyin: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetBaseProfileListResponseBodyResult = GetBaseProfileListResponseBodyResult;
class GetConversationResponseBodyResult extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            memberCount: 'memberCount',
            title: 'title',
        };
    }
    static types() {
        return {
            corpId: 'string',
            memberCount: 'number',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetConversationResponseBodyResult = GetConversationResponseBodyResult;
class QueryPluginRuleResponseBodyResultData extends $tea.Model {
    static names() {
        return {
            bizId: 'bizId',
            chatType: 'chatType',
            code: 'code',
            gmtCreate: 'gmtCreate',
            itemId: 'itemId',
            itemName: 'itemName',
            itemType: 'itemType',
        };
    }
    static types() {
        return {
            bizId: 'string',
            chatType: 'string',
            code: 'string',
            gmtCreate: 'string',
            itemId: 'string',
            itemName: 'string',
            itemType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryPluginRuleResponseBodyResultData = QueryPluginRuleResponseBodyResultData;
class QueryPluginRuleResponseBodyResult extends $tea.Model {
    static names() {
        return {
            data: 'data',
            total: 'total',
        };
    }
    static types() {
        return {
            data: { 'type': 'array', 'itemType': QueryPluginRuleResponseBodyResultData },
            total: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryPluginRuleResponseBodyResult = QueryPluginRuleResponseBodyResult;
class SendDingTipRequestLink extends $tea.Model {
    static names() {
        return {
            extension: 'extension',
            linkUrl: 'linkUrl',
            picMediaId: 'picMediaId',
            text: 'text',
        };
    }
    static types() {
        return {
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            linkUrl: 'string',
            picMediaId: 'string',
            text: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendDingTipRequestLink = SendDingTipRequestLink;
class SendMessageTipRequestDefaultView extends $tea.Model {
    static names() {
        return {
            actionName: 'actionName',
            actionUrl: 'actionUrl',
            authCode: 'authCode',
            authMediaId: 'authMediaId',
            cardTitle: 'cardTitle',
            cardTitleColor: 'cardTitleColor',
            desc: 'desc',
            mediaId: 'mediaId',
            needShowUpdateTail: 'needShowUpdateTail',
            summary: 'summary',
            title: 'title',
        };
    }
    static types() {
        return {
            actionName: 'string',
            actionUrl: 'string',
            authCode: 'string',
            authMediaId: 'string',
            cardTitle: 'string',
            cardTitleColor: 'string',
            desc: 'string',
            mediaId: 'string',
            needShowUpdateTail: 'string',
            summary: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendMessageTipRequestDefaultView = SendMessageTipRequestDefaultView;
class SendMessageTipRequestPublicField extends $tea.Model {
    static names() {
        return {
            detailUrl: 'detailUrl',
            durationDesc: 'durationDesc',
            extension: 'extension',
            isExpired: 'isExpired',
            readActionUrl: 'readActionUrl',
            readProgressDesc: 'readProgressDesc',
        };
    }
    static types() {
        return {
            detailUrl: 'string',
            durationDesc: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            isExpired: 'boolean',
            readActionUrl: 'string',
            readProgressDesc: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendMessageTipRequestPublicField = SendMessageTipRequestPublicField;
class SendMessageTipResponseBodyResult extends $tea.Model {
    static names() {
        return {
            bizId: 'bizId',
            cardInstanceId: 'cardInstanceId',
        };
    }
    static types() {
        return {
            bizId: 'string',
            cardInstanceId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendMessageTipResponseBodyResult = SendMessageTipResponseBodyResult;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 添加插件规则
     *
     * @param request AddPluginRuleRequest
     * @param headers AddPluginRuleHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddPluginRuleResponse
     */
    async addPluginRuleWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.chatType)) {
            body["chatType"] = request.chatType;
        }
        if (!tea_util_1.default.isUnset(request.code)) {
            body["code"] = request.code;
        }
        if (!tea_util_1.default.isUnset(request.itemType)) {
            body["itemType"] = request.itemType;
        }
        if (!tea_util_1.default.isUnset(request.rules)) {
            body["rules"] = request.rules;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddPluginRule",
            version: "flashmsg_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmsg/plugins`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddPluginRuleResponse({}));
    }
    /**
     * @summary 添加插件规则
     *
     * @param request AddPluginRuleRequest
     * @return AddPluginRuleResponse
     */
    async addPluginRule(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddPluginRuleHeaders({});
        return await this.addPluginRuleWithOptions(request, headers, runtime);
    }
    /**
     * @summary 删除插件规则
     *
     * @param request DeletePlguinRuleRequest
     * @param headers DeletePlguinRuleHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeletePlguinRuleResponse
     */
    async deletePlguinRuleWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizIdList)) {
            body["bizIdList"] = request.bizIdList;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "DeletePlguinRule",
            version: "flashmsg_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmsg/plugins/remove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeletePlguinRuleResponse({}));
    }
    /**
     * @summary 删除插件规则
     *
     * @param request DeletePlguinRuleRequest
     * @return DeletePlguinRuleResponse
     */
    async deletePlguinRule(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeletePlguinRuleHeaders({});
        return await this.deletePlguinRuleWithOptions(request, headers, runtime);
    }
    /**
     * @summary 闪读用户基础信息查询
     *
     * @param request GetBaseProfileListRequest
     * @param headers GetBaseProfileListHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetBaseProfileListResponse
     */
    async getBaseProfileListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: request.body,
        });
        let params = new $OpenApi.Params({
            action: "GetBaseProfileList",
            version: "flashmsg_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmsg/users/baseInfos/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetBaseProfileListResponse({}));
    }
    /**
     * @summary 闪读用户基础信息查询
     *
     * @param request GetBaseProfileListRequest
     * @return GetBaseProfileListResponse
     */
    async getBaseProfileList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetBaseProfileListHeaders({});
        return await this.getBaseProfileListWithOptions(request, headers, runtime);
    }
    /**
     * @summary 获得闪读会话信息
     *
     * @param request GetConversationRequest
     * @param headers GetConversationHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetConversationResponse
     */
    async getConversationWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            query["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetConversation",
            version: "flashmsg_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmsg/conversations/infos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetConversationResponse({}));
    }
    /**
     * @summary 获得闪读会话信息
     *
     * @param request GetConversationRequest
     * @return GetConversationResponse
     */
    async getConversation(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetConversationHeaders({});
        return await this.getConversationWithOptions(request, headers, runtime);
    }
    /**
     * @summary 获得成员ID列表
     *
     * @param request GetMemberListRequest
     * @param headers GetMemberListHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetMemberListResponse
     */
    async getMemberListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            query["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            query["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            query["pageSize"] = request.pageSize;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetMemberList",
            version: "flashmsg_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmsg/conversations/memberIdLists`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetMemberListResponse({}));
    }
    /**
     * @summary 获得成员ID列表
     *
     * @param request GetMemberListRequest
     * @return GetMemberListResponse
     */
    async getMemberList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetMemberListHeaders({});
        return await this.getMemberListWithOptions(request, headers, runtime);
    }
    /**
     * @summary 查询插件规则
     *
     * @param request QueryPluginRuleRequest
     * @param headers QueryPluginRuleHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryPluginRuleResponse
     */
    async queryPluginRuleWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.chatType)) {
            query["chatType"] = request.chatType;
        }
        if (!tea_util_1.default.isUnset(request.code)) {
            query["code"] = request.code;
        }
        if (!tea_util_1.default.isUnset(request.itemId)) {
            query["itemId"] = request.itemId;
        }
        if (!tea_util_1.default.isUnset(request.itemType)) {
            query["itemType"] = request.itemType;
        }
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            query["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            query["pageSize"] = request.pageSize;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryPluginRule",
            version: "flashmsg_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmsg/plugins`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryPluginRuleResponse({}));
    }
    /**
     * @summary 查询插件规则
     *
     * @param request QueryPluginRuleRequest
     * @return QueryPluginRuleResponse
     */
    async queryPluginRule(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryPluginRuleHeaders({});
        return await this.queryPluginRuleWithOptions(request, headers, runtime);
    }
    /**
     * @summary 发送Ding提示消息
     *
     * @param request SendDingTipRequest
     * @param headers SendDingTipHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SendDingTipResponse
     */
    async sendDingTipWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.extension)) {
            body["extension"] = request.extension;
        }
        if (!tea_util_1.default.isUnset(request.link)) {
            body["link"] = request.link;
        }
        if (!tea_util_1.default.isUnset(request.messageId)) {
            body["messageId"] = request.messageId;
        }
        if (!tea_util_1.default.isUnset(request.receiverUserId)) {
            body["receiverUserId"] = request.receiverUserId;
        }
        if (!tea_util_1.default.isUnset(request.senderUserId)) {
            body["senderUserId"] = request.senderUserId;
        }
        if (!tea_util_1.default.isUnset(request.textContent)) {
            body["textContent"] = request.textContent;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendDingTip",
            version: "flashmsg_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmsg/ding/messages/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendDingTipResponse({}));
    }
    /**
     * @summary 发送Ding提示消息
     *
     * @param request SendDingTipRequest
     * @return SendDingTipResponse
     */
    async sendDingTip(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendDingTipHeaders({});
        return await this.sendDingTipWithOptions(request, headers, runtime);
    }
    /**
     * @summary 发送闪读消息提示
     *
     * @param request SendMessageTipRequest
     * @param headers SendMessageTipHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SendMessageTipResponse
     */
    async sendMessageTipWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.defaultView)) {
            body["defaultView"] = request.defaultView;
        }
        if (!tea_util_1.default.isUnset(request.messageId)) {
            body["messageId"] = request.messageId;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.privateFieldMap)) {
            body["privateFieldMap"] = request.privateFieldMap;
        }
        if (!tea_util_1.default.isUnset(request.publicField)) {
            body["publicField"] = request.publicField;
        }
        if (!tea_util_1.default.isUnset(request.receiverUserId)) {
            body["receiverUserId"] = request.receiverUserId;
        }
        if (!tea_util_1.default.isUnset(request.senderUserId)) {
            body["senderUserId"] = request.senderUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendMessageTip",
            version: "flashmsg_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmsg/messages/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendMessageTipResponse({}));
    }
    /**
     * @summary 发送闪读消息提示
     *
     * @param request SendMessageTipRequest
     * @return SendMessageTipResponse
     */
    async sendMessageTip(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendMessageTipHeaders({});
        return await this.sendMessageTipWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
