// @alicloud/dingtalk@2.1.16/dist/occupationauth_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckUserTasksStatusResponse = exports.CheckUserTasksStatusResponseBody = exports.CheckUserTasksStatusRequest = exports.CheckUserTasksStatusHeaders = exports.CheckUserTaskStatusResponse = exports.CheckUserTaskStatusResponseBody = exports.CheckUserTaskStatusRequest = exports.CheckUserTaskStatusHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CheckUserTaskStatusHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckUserTaskStatusHeaders = CheckUserTaskStatusHeaders;
class CheckUserTaskStatusRequest extends $tea.Model {
    static names() {
        return {
            provinceCode: 'provinceCode',
        };
    }
    static types() {
        return {
            provinceCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckUserTaskStatusRequest = CheckUserTaskStatusRequest;
class CheckUserTaskStatusResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckUserTaskStatusResponseBody = CheckUserTaskStatusResponseBody;
class CheckUserTaskStatusResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CheckUserTaskStatusResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckUserTaskStatusResponse = CheckUserTaskStatusResponse;
class CheckUserTasksStatusHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckUserTasksStatusHeaders = CheckUserTasksStatusHeaders;
class CheckUserTasksStatusRequest extends $tea.Model {
    static names() {
        return {
            provinceCode: 'provinceCode',
        };
    }
    static types() {
        return {
            provinceCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckUserTasksStatusRequest = CheckUserTasksStatusRequest;
class CheckUserTasksStatusResponseBody extends $tea.Model {
    static names() {
        return {
            status: 'status',
        };
    }
    static types() {
        return {
            status: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckUserTasksStatusResponseBody = CheckUserTasksStatusResponseBody;
class CheckUserTasksStatusResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CheckUserTasksStatusResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckUserTasksStatusResponse = CheckUserTasksStatusResponse;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 检查用户任务状态
     *
     * @param request CheckUserTaskStatusRequest
     * @param headers CheckUserTaskStatusHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CheckUserTaskStatusResponse
     */
    async checkUserTaskStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.provinceCode)) {
            body["provinceCode"] = request.provinceCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CheckUserTaskStatus",
            version: "occupationauth_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/occupationauth/auths/userTasks`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CheckUserTaskStatusResponse({}));
    }
    /**
     * @summary 检查用户任务状态
     *
     * @param request CheckUserTaskStatusRequest
     * @return CheckUserTaskStatusResponse
     */
    async checkUserTaskStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CheckUserTaskStatusHeaders({});
        return await this.checkUserTaskStatusWithOptions(request, headers, runtime);
    }
    /**
     * @summary 检查用户任务状态
     *
     * @param request CheckUserTasksStatusRequest
     * @param headers CheckUserTasksStatusHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CheckUserTasksStatusResponse
     */
    async checkUserTasksStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.provinceCode)) {
            query["provinceCode"] = request.provinceCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "CheckUserTasksStatus",
            version: "occupationauth_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/occupationauth/userTasks/check`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CheckUserTasksStatusResponse({}));
    }
    /**
     * @summary 检查用户任务状态
     *
     * @param request CheckUserTasksStatusRequest
     * @return CheckUserTasksStatusResponse
     */
    async checkUserTasksStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CheckUserTasksStatusHeaders({});
        return await this.checkUserTasksStatusWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
