// @alicloud/dingtalk@2.1.16/dist/watt_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RevertPointResponseBodyResult = exports.RevertPointRequestBody = exports.GetPointInfoResponseBodyResult = exports.ConsumePointResponseBodyResult = exports.ConsumePointRequestBody = exports.ConsumePointRequestBodyConsumeDetail = exports.ConsumePointRequestBodyConsumeDetailBenefit = exports.SendSearchShadeResponse = exports.SendSearchShadeResponseBody = exports.SendSearchShadeRequest = exports.SendSearchShadeHeaders = exports.SendPopupResponse = exports.SendPopupResponseBody = exports.SendPopupRequest = exports.SendPopupHeaders = exports.SendBannerResponse = exports.SendBannerResponseBody = exports.SendBannerRequest = exports.SendBannerHeaders = exports.RevertPointResponse = exports.RevertPointResponseBody = exports.RevertPointShrinkRequest = exports.RevertPointRequest = exports.RevertPointHeaders = exports.GetPointInfoResponse = exports.GetPointInfoResponseBody = exports.GetPointInfoRequest = exports.GetPointInfoHeaders = exports.CreateDeliveryPlanResponse = exports.CreateDeliveryPlanResponseBody = exports.CreateDeliveryPlanRequest = exports.CreateDeliveryPlanHeaders = exports.ConsumePointResponse = exports.ConsumePointResponseBody = exports.ConsumePointShrinkRequest = exports.ConsumePointRequest = exports.ConsumePointHeaders = exports.CheckInCrowdsByMobileResponse = exports.CheckInCrowdsByMobileResponseBody = exports.CheckInCrowdsByMobileRequest = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CheckInCrowdsByMobileRequest extends $tea.Model {
    static names() {
        return {
            crowdIds: 'crowdIds',
            mobile: 'mobile',
        };
    }
    static types() {
        return {
            crowdIds: 'Buffer',
            mobile: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckInCrowdsByMobileRequest = CheckInCrowdsByMobileRequest;
class CheckInCrowdsByMobileResponseBody extends $tea.Model {
    static names() {
        return {
            data: 'data',
            success: 'success',
            total: 'total',
        };
    }
    static types() {
        return {
            data: 'boolean',
            success: 'boolean',
            total: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckInCrowdsByMobileResponseBody = CheckInCrowdsByMobileResponseBody;
class CheckInCrowdsByMobileResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CheckInCrowdsByMobileResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CheckInCrowdsByMobileResponse = CheckInCrowdsByMobileResponse;
class ConsumePointHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConsumePointHeaders = ConsumePointHeaders;
class ConsumePointRequest extends $tea.Model {
    static names() {
        return {
            body: 'body',
        };
    }
    static types() {
        return {
            body: ConsumePointRequestBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConsumePointRequest = ConsumePointRequest;
class ConsumePointShrinkRequest extends $tea.Model {
    static names() {
        return {
            bodyShrink: 'body',
        };
    }
    static types() {
        return {
            bodyShrink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConsumePointShrinkRequest = ConsumePointShrinkRequest;
class ConsumePointResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: ConsumePointResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConsumePointResponseBody = ConsumePointResponseBody;
class ConsumePointResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ConsumePointResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConsumePointResponse = ConsumePointResponse;
class CreateDeliveryPlanHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateDeliveryPlanHeaders = CreateDeliveryPlanHeaders;
class CreateDeliveryPlanRequest extends $tea.Model {
    static names() {
        return {
            content: 'content',
            endTime: 'endTime',
            resId: 'resId',
            startTime: 'startTime',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            content: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            endTime: 'number',
            resId: 'string',
            startTime: 'number',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateDeliveryPlanRequest = CreateDeliveryPlanRequest;
class CreateDeliveryPlanResponseBody extends $tea.Model {
    static names() {
        return {
            arguments: 'arguments',
            success: 'success',
        };
    }
    static types() {
        return {
            arguments: { 'type': 'array', 'itemType': 'any' },
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateDeliveryPlanResponseBody = CreateDeliveryPlanResponseBody;
class CreateDeliveryPlanResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateDeliveryPlanResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateDeliveryPlanResponse = CreateDeliveryPlanResponse;
class GetPointInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPointInfoHeaders = GetPointInfoHeaders;
class GetPointInfoRequest extends $tea.Model {
    static names() {
        return {
            pointPoolCode: 'pointPoolCode',
        };
    }
    static types() {
        return {
            pointPoolCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPointInfoRequest = GetPointInfoRequest;
class GetPointInfoResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: GetPointInfoResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPointInfoResponseBody = GetPointInfoResponseBody;
class GetPointInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetPointInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPointInfoResponse = GetPointInfoResponse;
class RevertPointHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RevertPointHeaders = RevertPointHeaders;
class RevertPointRequest extends $tea.Model {
    static names() {
        return {
            body: 'body',
        };
    }
    static types() {
        return {
            body: RevertPointRequestBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RevertPointRequest = RevertPointRequest;
class RevertPointShrinkRequest extends $tea.Model {
    static names() {
        return {
            bodyShrink: 'body',
        };
    }
    static types() {
        return {
            bodyShrink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RevertPointShrinkRequest = RevertPointShrinkRequest;
class RevertPointResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: RevertPointResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RevertPointResponseBody = RevertPointResponseBody;
class RevertPointResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RevertPointResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RevertPointResponse = RevertPointResponse;
class SendBannerHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendBannerHeaders = SendBannerHeaders;
class SendBannerRequest extends $tea.Model {
    static names() {
        return {
            content: 'content',
            endTime: 'endTime',
            startTime: 'startTime',
            userId: 'userId',
        };
    }
    static types() {
        return {
            content: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            endTime: 'number',
            startTime: 'number',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendBannerRequest = SendBannerRequest;
class SendBannerResponseBody extends $tea.Model {
    static names() {
        return {
            arguments: 'arguments',
            success: 'success',
        };
    }
    static types() {
        return {
            arguments: { 'type': 'array', 'itemType': 'any' },
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendBannerResponseBody = SendBannerResponseBody;
class SendBannerResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendBannerResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendBannerResponse = SendBannerResponse;
class SendPopupHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendPopupHeaders = SendPopupHeaders;
class SendPopupRequest extends $tea.Model {
    static names() {
        return {
            content: 'content',
            endTime: 'endTime',
            startTime: 'startTime',
            userId: 'userId',
        };
    }
    static types() {
        return {
            content: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            endTime: 'number',
            startTime: 'number',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendPopupRequest = SendPopupRequest;
class SendPopupResponseBody extends $tea.Model {
    static names() {
        return {
            arguments: 'arguments',
            success: 'success',
        };
    }
    static types() {
        return {
            arguments: { 'type': 'array', 'itemType': 'any' },
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendPopupResponseBody = SendPopupResponseBody;
class SendPopupResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendPopupResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendPopupResponse = SendPopupResponse;
class SendSearchShadeHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendSearchShadeHeaders = SendSearchShadeHeaders;
class SendSearchShadeRequest extends $tea.Model {
    static names() {
        return {
            content: 'content',
            endTime: 'endTime',
            startTime: 'startTime',
            userId: 'userId',
        };
    }
    static types() {
        return {
            content: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            endTime: 'number',
            startTime: 'number',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendSearchShadeRequest = SendSearchShadeRequest;
class SendSearchShadeResponseBody extends $tea.Model {
    static names() {
        return {
            arguments: 'arguments',
            success: 'success',
        };
    }
    static types() {
        return {
            arguments: { 'type': 'array', 'itemType': 'any' },
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendSearchShadeResponseBody = SendSearchShadeResponseBody;
class SendSearchShadeResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendSearchShadeResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendSearchShadeResponse = SendSearchShadeResponse;
class ConsumePointRequestBodyConsumeDetailBenefit extends $tea.Model {
    static names() {
        return {
            benefitId: 'benefitId',
            name: 'name',
            supplierName: 'supplierName',
            useUrl: 'useUrl',
        };
    }
    static types() {
        return {
            benefitId: 'string',
            name: 'string',
            supplierName: 'string',
            useUrl: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConsumePointRequestBodyConsumeDetailBenefit = ConsumePointRequestBodyConsumeDetailBenefit;
class ConsumePointRequestBodyConsumeDetail extends $tea.Model {
    static names() {
        return {
            benefit: 'benefit',
            type: 'type',
        };
    }
    static types() {
        return {
            benefit: ConsumePointRequestBodyConsumeDetailBenefit,
            type: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConsumePointRequestBodyConsumeDetail = ConsumePointRequestBodyConsumeDetail;
class ConsumePointRequestBody extends $tea.Model {
    static names() {
        return {
            consumeDetail: 'consumeDetail',
            pointPoolCode: 'pointPoolCode',
            points: 'points',
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            consumeDetail: ConsumePointRequestBodyConsumeDetail,
            pointPoolCode: 'string',
            points: 'number',
            requestId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConsumePointRequestBody = ConsumePointRequestBody;
class ConsumePointResponseBodyResult extends $tea.Model {
    static names() {
        return {
            consumedPoints: 'consumedPoints',
        };
    }
    static types() {
        return {
            consumedPoints: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConsumePointResponseBodyResult = ConsumePointResponseBodyResult;
class GetPointInfoResponseBodyResult extends $tea.Model {
    static names() {
        return {
            userPoints: 'userPoints',
        };
    }
    static types() {
        return {
            userPoints: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPointInfoResponseBodyResult = GetPointInfoResponseBodyResult;
class RevertPointRequestBody extends $tea.Model {
    static names() {
        return {
            pointPoolCode: 'pointPoolCode',
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            pointPoolCode: 'string',
            requestId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RevertPointRequestBody = RevertPointRequestBody;
class RevertPointResponseBodyResult extends $tea.Model {
    static names() {
        return {
            revertedPoints: 'revertedPoints',
        };
    }
    static types() {
        return {
            revertedPoints: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RevertPointResponseBodyResult = RevertPointResponseBodyResult;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._signatureAlgorithm = "v2";
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 根据加密后的用户手机号检查该用户是否在某人群中
     *
     * @param request CheckInCrowdsByMobileRequest
     * @param headers map
     * @param runtime runtime options for this request RuntimeOptions
     * @return CheckInCrowdsByMobileResponse
     */
    async checkInCrowdsByMobileWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.crowdIds)) {
            query["crowdIds"] = request.crowdIds;
        }
        if (!tea_util_1.default.isUnset(request.mobile)) {
            query["mobile"] = request.mobile;
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "CheckInCrowdsByMobile",
            version: "watt_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/watt/crowdIdentifications/query`,
            method: "POST",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CheckInCrowdsByMobileResponse({}));
    }
    /**
     * @summary 根据加密后的用户手机号检查该用户是否在某人群中
     *
     * @param request CheckInCrowdsByMobileRequest
     * @return CheckInCrowdsByMobileResponse
     */
    async checkInCrowdsByMobile(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.checkInCrowdsByMobileWithOptions(request, headers, runtime);
    }
    /**
     * @summary 消耗用户积分
     *
     * @param tmpReq ConsumePointRequest
     * @param headers ConsumePointHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ConsumePointResponse
     */
    async consumePointWithOptions(tmpReq, headers, runtime) {
        tea_util_1.default.validateModel(tmpReq);
        let request = new ConsumePointShrinkRequest({});
        openapi_util_1.default.convert(tmpReq, request);
        if (!tea_util_1.default.isUnset(tmpReq.body)) {
            request.bodyShrink = openapi_util_1.default.arrayToStringWithSpecifiedStyle(tmpReq.body, "body", "json");
        }
        let query = {};
        if (!tea_util_1.default.isUnset(request.bodyShrink)) {
            query["body"] = request.bodyShrink;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ConsumePoint",
            version: "watt_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/watt/points/consume`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ConsumePointResponse({}));
    }
    /**
     * @summary 消耗用户积分
     *
     * @param request ConsumePointRequest
     * @return ConsumePointResponse
     */
    async consumePoint(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ConsumePointHeaders({});
        return await this.consumePointWithOptions(request, headers, runtime);
    }
    /**
     * @summary 发布钉钉投放任务（搜索穹顶、搜索发现、搜索关键字）
     *
     * @param request CreateDeliveryPlanRequest
     * @param headers CreateDeliveryPlanHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateDeliveryPlanResponse
     */
    async createDeliveryPlanWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.content)) {
            body["content"] = request.content;
        }
        if (!tea_util_1.default.isUnset(request.endTime)) {
            body["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.resId)) {
            body["resId"] = request.resId;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.userIdList)) {
            body["userIdList"] = request.userIdList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateDeliveryPlan",
            version: "watt_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/watt/deliveryPlans/publish`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateDeliveryPlanResponse({}));
    }
    /**
     * @summary 发布钉钉投放任务（搜索穹顶、搜索发现、搜索关键字）
     *
     * @param request CreateDeliveryPlanRequest
     * @return CreateDeliveryPlanResponse
     */
    async createDeliveryPlan(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateDeliveryPlanHeaders({});
        return await this.createDeliveryPlanWithOptions(request, headers, runtime);
    }
    /**
     * @summary 查询用户积分
     *
     * @param request GetPointInfoRequest
     * @param headers GetPointInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetPointInfoResponse
     */
    async getPointInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.pointPoolCode)) {
            query["pointPoolCode"] = request.pointPoolCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetPointInfo",
            version: "watt_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/watt/points`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetPointInfoResponse({}));
    }
    /**
     * @summary 查询用户积分
     *
     * @param request GetPointInfoRequest
     * @return GetPointInfoResponse
     */
    async getPointInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetPointInfoHeaders({});
        return await this.getPointInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 撤销用户单笔积分消耗
     *
     * @param tmpReq RevertPointRequest
     * @param headers RevertPointHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RevertPointResponse
     */
    async revertPointWithOptions(tmpReq, headers, runtime) {
        tea_util_1.default.validateModel(tmpReq);
        let request = new RevertPointShrinkRequest({});
        openapi_util_1.default.convert(tmpReq, request);
        if (!tea_util_1.default.isUnset(tmpReq.body)) {
            request.bodyShrink = openapi_util_1.default.arrayToStringWithSpecifiedStyle(tmpReq.body, "body", "json");
        }
        let query = {};
        if (!tea_util_1.default.isUnset(request.bodyShrink)) {
            query["body"] = request.bodyShrink;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "RevertPoint",
            version: "watt_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/watt/points/revert`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RevertPointResponse({}));
    }
    /**
     * @summary 撤销用户单笔积分消耗
     *
     * @param request RevertPointRequest
     * @return RevertPointResponse
     */
    async revertPoint(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RevertPointHeaders({});
        return await this.revertPointWithOptions(request, headers, runtime);
    }
    /**
     * @summary 发送钉钉统一引导Banner
     *
     * @param request SendBannerRequest
     * @param headers SendBannerHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SendBannerResponse
     */
    async sendBannerWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.content)) {
            body["content"] = request.content;
        }
        if (!tea_util_1.default.isUnset(request.endTime)) {
            body["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendBanner",
            version: "watt_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/watt/banners/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendBannerResponse({}));
    }
    /**
     * @summary 发送钉钉统一引导Banner
     *
     * @param request SendBannerRequest
     * @return SendBannerResponse
     */
    async sendBanner(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendBannerHeaders({});
        return await this.sendBannerWithOptions(request, headers, runtime);
    }
    /**
     * @summary 发送钉钉首页弹窗
     *
     * @param request SendPopupRequest
     * @param headers SendPopupHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SendPopupResponse
     */
    async sendPopupWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.content)) {
            body["content"] = request.content;
        }
        if (!tea_util_1.default.isUnset(request.endTime)) {
            body["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendPopup",
            version: "watt_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/watt/popups/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendPopupResponse({}));
    }
    /**
     * @summary 发送钉钉首页弹窗
     *
     * @param request SendPopupRequest
     * @return SendPopupResponse
     */
    async sendPopup(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendPopupHeaders({});
        return await this.sendPopupWithOptions(request, headers, runtime);
    }
    /**
     * @summary 发送钉钉搜索底纹
     *
     * @param request SendSearchShadeRequest
     * @param headers SendSearchShadeHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SendSearchShadeResponse
     */
    async sendSearchShadeWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.content)) {
            body["content"] = request.content;
        }
        if (!tea_util_1.default.isUnset(request.endTime)) {
            body["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendSearchShade",
            version: "watt_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/watt/searchShades/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendSearchShadeResponse({}));
    }
    /**
     * @summary 发送钉钉搜索底纹
     *
     * @param request SendSearchShadeRequest
     * @return SendSearchShadeResponse
     */
    async sendSearchShade(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendSearchShadeHeaders({});
        return await this.sendSearchShadeWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
