// @alicloud/dingtalk@2.1.16/dist/ats_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetJobAuthRequest = exports.GetJobAuthHeaders = exports.GetFlowIdByRelationEntityIdResponse = exports.GetFlowIdByRelationEntityIdResponseBody = exports.GetFlowIdByRelationEntityIdRequest = exports.GetFlowIdByRelationEntityIdHeaders = exports.GetFileUploadInfoResponse = exports.GetFileUploadInfoResponseBody = exports.GetFileUploadInfoRequest = exports.GetFileUploadInfoHeaders = exports.GetCandidateByPhoneNumberResponse = exports.GetCandidateByPhoneNumberResponseBody = exports.GetCandidateByPhoneNumberRequest = exports.GetCandidateByPhoneNumberHeaders = exports.GetApplicationRegFormByFlowIdResponse = exports.GetApplicationRegFormByFlowIdResponseBody = exports.GetApplicationRegFormByFlowIdRequest = exports.GetApplicationRegFormByFlowIdHeaders = exports.FinishBeginnerTaskResponse = exports.FinishBeginnerTaskResponseBody = exports.FinishBeginnerTaskRequest = exports.FinishBeginnerTaskHeaders = exports.ConfirmRightsResponse = exports.ConfirmRightsResponseBody = exports.ConfirmRightsRequest = exports.ConfirmRightsHeaders = exports.CollectResumeMailResponse = exports.CollectResumeMailResponseBody = exports.CollectResumeMailRequest = exports.CollectResumeMailHeaders = exports.CollectResumeDetailResponse = exports.CollectResumeDetailResponseBody = exports.CollectResumeDetailRequest = exports.CollectResumeDetailHeaders = exports.CollectRecruitJobDetailResponse = exports.CollectRecruitJobDetailResponseBody = exports.CollectRecruitJobDetailRequest = exports.CollectRecruitJobDetailHeaders = exports.AddUserAccountResponse = exports.AddUserAccountResponseBody = exports.AddUserAccountRequest = exports.AddUserAccountHeaders = exports.AddFileResponse = exports.AddFileResponseBody = exports.AddFileRequest = exports.AddFileHeaders = exports.AddApplicationRegFormTemplateResponse = exports.AddApplicationRegFormTemplateResponseBody = exports.AddApplicationRegFormTemplateRequest = exports.AddApplicationRegFormTemplateHeaders = void 0;
exports.UpdateApplicationRegFormRequestDingPanFile = exports.QueryInterviewsResponseBodyList = exports.QueryInterviewsResponseBodyListInterviewers = exports.QueryCandidatesResponseBodyList = exports.GetJobAuthResponseBodyJobOwners = exports.CollectResumeMailRequestResumeFile = exports.CollectResumeDetailRequestResumeFile = exports.CollectResumeDetailRequestResumeData = exports.CollectResumeDetailRequestResumeDataWorkExperiences = exports.CollectResumeDetailRequestResumeDataTrainingExperiences = exports.CollectResumeDetailRequestResumeDataLanguageSkill = exports.CollectResumeDetailRequestResumeDataJobExpect = exports.CollectResumeDetailRequestResumeDataEducationExperiences = exports.CollectResumeDetailRequestResumeDataCertificates = exports.CollectResumeDetailRequestResumeDataBaseInfo = exports.CollectRecruitJobDetailRequestRecruitUserInfo = exports.CollectRecruitJobDetailRequestJobInfo = exports.CollectRecruitJobDetailRequestJobInfoPartTimeInfo = exports.CollectRecruitJobDetailRequestJobInfoFullTimeInfo = exports.CollectRecruitJobDetailRequestJobInfoAddress = exports.UpdateJobDeliverResponse = exports.UpdateJobDeliverResponseBody = exports.UpdateJobDeliverRequest = exports.UpdateJobDeliverHeaders = exports.UpdateInterviewSignInInfoResponse = exports.UpdateInterviewSignInInfoRequest = exports.UpdateInterviewSignInInfoHeaders = exports.UpdateApplicationRegFormResponse = exports.UpdateApplicationRegFormResponseBody = exports.UpdateApplicationRegFormRequest = exports.UpdateApplicationRegFormHeaders = exports.SyncChannelMessageResponse = exports.SyncChannelMessageResponseBody = exports.SyncChannelMessageRequest = exports.SyncChannelMessageHeaders = exports.ReportMessageStatusResponse = exports.ReportMessageStatusResponseBody = exports.ReportMessageStatusRequest = exports.ReportMessageStatusHeaders = exports.QueryInterviewsResponse = exports.QueryInterviewsResponseBody = exports.QueryInterviewsRequest = exports.QueryInterviewsHeaders = exports.QueryCandidatesResponse = exports.QueryCandidatesResponseBody = exports.QueryCandidatesRequest = exports.QueryCandidatesHeaders = exports.GetJobAuthResponse = exports.GetJobAuthResponseBody = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AddApplicationRegFormTemplateHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddApplicationRegFormTemplateHeaders = AddApplicationRegFormTemplateHeaders;
class AddApplicationRegFormTemplateRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            content: 'content',
            name: 'name',
            outerId: 'outerId',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            content: 'string',
            name: 'string',
            outerId: 'string',
            opUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddApplicationRegFormTemplateRequest = AddApplicationRegFormTemplateRequest;
class AddApplicationRegFormTemplateResponseBody extends $tea.Model {
    static names() {
        return {
            templateId: 'templateId',
            version: 'version',
        };
    }
    static types() {
        return {
            templateId: 'string',
            version: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddApplicationRegFormTemplateResponseBody = AddApplicationRegFormTemplateResponseBody;
class AddApplicationRegFormTemplateResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddApplicationRegFormTemplateResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddApplicationRegFormTemplateResponse = AddApplicationRegFormTemplateResponse;
class AddFileHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddFileHeaders = AddFileHeaders;
class AddFileRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            fileName: 'fileName',
            mediaId: 'mediaId',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            fileName: 'string',
            mediaId: 'string',
            opUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddFileRequest = AddFileRequest;
class AddFileResponseBody extends $tea.Model {
    static names() {
        return {
            fileId: 'fileId',
            fileName: 'fileName',
            spaceId: 'spaceId',
        };
    }
    static types() {
        return {
            fileId: 'string',
            fileName: 'string',
            spaceId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddFileResponseBody = AddFileResponseBody;
class AddFileResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddFileResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddFileResponse = AddFileResponse;
class AddUserAccountHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddUserAccountHeaders = AddUserAccountHeaders;
class AddUserAccountRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            channelAccountName: 'channelAccountName',
            channelUserIdentify: 'channelUserIdentify',
            phoneNumber: 'phoneNumber',
            corpId: 'corpId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            channelAccountName: 'string',
            channelUserIdentify: 'string',
            phoneNumber: 'string',
            corpId: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddUserAccountRequest = AddUserAccountRequest;
class AddUserAccountResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddUserAccountResponseBody = AddUserAccountResponseBody;
class AddUserAccountResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddUserAccountResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddUserAccountResponse = AddUserAccountResponse;
class CollectRecruitJobDetailHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectRecruitJobDetailHeaders = CollectRecruitJobDetailHeaders;
class CollectRecruitJobDetailRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            channel: 'channel',
            jobInfo: 'jobInfo',
            outCorpId: 'outCorpId',
            outCorpName: 'outCorpName',
            recruitUserInfo: 'recruitUserInfo',
            source: 'source',
            updateTime: 'updateTime',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            channel: 'string',
            jobInfo: CollectRecruitJobDetailRequestJobInfo,
            outCorpId: 'string',
            outCorpName: 'string',
            recruitUserInfo: CollectRecruitJobDetailRequestRecruitUserInfo,
            source: 'string',
            updateTime: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectRecruitJobDetailRequest = CollectRecruitJobDetailRequest;
class CollectRecruitJobDetailResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectRecruitJobDetailResponseBody = CollectRecruitJobDetailResponseBody;
class CollectRecruitJobDetailResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CollectRecruitJobDetailResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectRecruitJobDetailResponse = CollectRecruitJobDetailResponse;
class CollectResumeDetailHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailHeaders = CollectResumeDetailHeaders;
class CollectResumeDetailRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            channelCode: 'channelCode',
            channelOuterId: 'channelOuterId',
            channelTalentId: 'channelTalentId',
            deliverJobId: 'deliverJobId',
            optUserId: 'optUserId',
            resumeChannelUrl: 'resumeChannelUrl',
            resumeData: 'resumeData',
            resumeFile: 'resumeFile',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            channelCode: 'string',
            channelOuterId: 'string',
            channelTalentId: 'string',
            deliverJobId: 'string',
            optUserId: 'string',
            resumeChannelUrl: 'string',
            resumeData: CollectResumeDetailRequestResumeData,
            resumeFile: CollectResumeDetailRequestResumeFile,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequest = CollectResumeDetailRequest;
class CollectResumeDetailResponseBody extends $tea.Model {
    static names() {
        return {
            resumeId: 'resumeId',
        };
    }
    static types() {
        return {
            resumeId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailResponseBody = CollectResumeDetailResponseBody;
class CollectResumeDetailResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CollectResumeDetailResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailResponse = CollectResumeDetailResponse;
class CollectResumeMailHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeMailHeaders = CollectResumeMailHeaders;
class CollectResumeMailRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            channelCode: 'channelCode',
            deliverJobId: 'deliverJobId',
            fromMailAddress: 'fromMailAddress',
            historyMailImport: 'historyMailImport',
            mailId: 'mailId',
            mailTitle: 'mailTitle',
            optUserId: 'optUserId',
            receiveMailAddress: 'receiveMailAddress',
            receiveMailType: 'receiveMailType',
            receivedTime: 'receivedTime',
            resumeChannelUrl: 'resumeChannelUrl',
            resumeFile: 'resumeFile',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            channelCode: 'string',
            deliverJobId: 'string',
            fromMailAddress: 'string',
            historyMailImport: 'boolean',
            mailId: 'string',
            mailTitle: 'string',
            optUserId: 'string',
            receiveMailAddress: 'string',
            receiveMailType: 'number',
            receivedTime: 'number',
            resumeChannelUrl: 'string',
            resumeFile: CollectResumeMailRequestResumeFile,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeMailRequest = CollectResumeMailRequest;
class CollectResumeMailResponseBody extends $tea.Model {
    static names() {
        return {
            resumeId: 'resumeId',
        };
    }
    static types() {
        return {
            resumeId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeMailResponseBody = CollectResumeMailResponseBody;
class CollectResumeMailResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CollectResumeMailResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeMailResponse = CollectResumeMailResponse;
class ConfirmRightsHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConfirmRightsHeaders = ConfirmRightsHeaders;
class ConfirmRightsRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
        };
    }
    static types() {
        return {
            bizCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConfirmRightsRequest = ConfirmRightsRequest;
class ConfirmRightsResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConfirmRightsResponseBody = ConfirmRightsResponseBody;
class ConfirmRightsResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ConfirmRightsResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ConfirmRightsResponse = ConfirmRightsResponse;
class FinishBeginnerTaskHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.FinishBeginnerTaskHeaders = FinishBeginnerTaskHeaders;
class FinishBeginnerTaskRequest extends $tea.Model {
    static names() {
        return {
            scope: 'scope',
            userId: 'userId',
        };
    }
    static types() {
        return {
            scope: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.FinishBeginnerTaskRequest = FinishBeginnerTaskRequest;
class FinishBeginnerTaskResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.FinishBeginnerTaskResponseBody = FinishBeginnerTaskResponseBody;
class FinishBeginnerTaskResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: FinishBeginnerTaskResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.FinishBeginnerTaskResponse = FinishBeginnerTaskResponse;
class GetApplicationRegFormByFlowIdHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetApplicationRegFormByFlowIdHeaders = GetApplicationRegFormByFlowIdHeaders;
class GetApplicationRegFormByFlowIdRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
        };
    }
    static types() {
        return {
            bizCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetApplicationRegFormByFlowIdRequest = GetApplicationRegFormByFlowIdRequest;
class GetApplicationRegFormByFlowIdResponseBody extends $tea.Model {
    static names() {
        return {
            candidateId: 'candidateId',
            creatorUserId: 'creatorUserId',
            flowId: 'flowId',
            formId: 'formId',
            gmtCreateMillis: 'gmtCreateMillis',
            gmtModifiedMillis: 'gmtModifiedMillis',
            jobId: 'jobId',
            status: 'status',
            templateId: 'templateId',
            templateVersion: 'templateVersion',
        };
    }
    static types() {
        return {
            candidateId: 'string',
            creatorUserId: 'string',
            flowId: 'string',
            formId: 'string',
            gmtCreateMillis: 'number',
            gmtModifiedMillis: 'number',
            jobId: 'string',
            status: 'number',
            templateId: 'string',
            templateVersion: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetApplicationRegFormByFlowIdResponseBody = GetApplicationRegFormByFlowIdResponseBody;
class GetApplicationRegFormByFlowIdResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetApplicationRegFormByFlowIdResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetApplicationRegFormByFlowIdResponse = GetApplicationRegFormByFlowIdResponse;
class GetCandidateByPhoneNumberHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetCandidateByPhoneNumberHeaders = GetCandidateByPhoneNumberHeaders;
class GetCandidateByPhoneNumberRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            phoneNumber: 'phoneNumber',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            phoneNumber: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetCandidateByPhoneNumberRequest = GetCandidateByPhoneNumberRequest;
class GetCandidateByPhoneNumberResponseBody extends $tea.Model {
    static names() {
        return {
            candidateId: 'candidateId',
            name: 'name',
        };
    }
    static types() {
        return {
            candidateId: 'string',
            name: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetCandidateByPhoneNumberResponseBody = GetCandidateByPhoneNumberResponseBody;
class GetCandidateByPhoneNumberResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetCandidateByPhoneNumberResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetCandidateByPhoneNumberResponse = GetCandidateByPhoneNumberResponse;
class GetFileUploadInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFileUploadInfoHeaders = GetFileUploadInfoHeaders;
class GetFileUploadInfoRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            fileName: 'fileName',
            fileSize: 'fileSize',
            md5: 'md5',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            fileName: 'string',
            fileSize: 'number',
            md5: 'string',
            opUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFileUploadInfoRequest = GetFileUploadInfoRequest;
class GetFileUploadInfoResponseBody extends $tea.Model {
    static names() {
        return {
            accessKeyId: 'accessKeyId',
            accessKeySecret: 'accessKeySecret',
            accessToken: 'accessToken',
            accessTokenExpirationMillis: 'accessTokenExpirationMillis',
            bucket: 'bucket',
            endPoint: 'endPoint',
            mediaId: 'mediaId',
        };
    }
    static types() {
        return {
            accessKeyId: 'string',
            accessKeySecret: 'string',
            accessToken: 'string',
            accessTokenExpirationMillis: 'number',
            bucket: 'string',
            endPoint: 'string',
            mediaId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFileUploadInfoResponseBody = GetFileUploadInfoResponseBody;
class GetFileUploadInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetFileUploadInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFileUploadInfoResponse = GetFileUploadInfoResponse;
class GetFlowIdByRelationEntityIdHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFlowIdByRelationEntityIdHeaders = GetFlowIdByRelationEntityIdHeaders;
class GetFlowIdByRelationEntityIdRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            relationEntity: 'relationEntity',
            relationEntityId: 'relationEntityId',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            relationEntity: 'string',
            relationEntityId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFlowIdByRelationEntityIdRequest = GetFlowIdByRelationEntityIdRequest;
class GetFlowIdByRelationEntityIdResponseBody extends $tea.Model {
    static names() {
        return {
            flowId: 'flowId',
        };
    }
    static types() {
        return {
            flowId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFlowIdByRelationEntityIdResponseBody = GetFlowIdByRelationEntityIdResponseBody;
class GetFlowIdByRelationEntityIdResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetFlowIdByRelationEntityIdResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetFlowIdByRelationEntityIdResponse = GetFlowIdByRelationEntityIdResponse;
class GetJobAuthHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobAuthHeaders = GetJobAuthHeaders;
class GetJobAuthRequest extends $tea.Model {
    static names() {
        return {
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            opUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobAuthRequest = GetJobAuthRequest;
class GetJobAuthResponseBody extends $tea.Model {
    static names() {
        return {
            jobId: 'jobId',
            jobOwners: 'jobOwners',
        };
    }
    static types() {
        return {
            jobId: 'string',
            jobOwners: { 'type': 'array', 'itemType': GetJobAuthResponseBodyJobOwners },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobAuthResponseBody = GetJobAuthResponseBody;
class GetJobAuthResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetJobAuthResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobAuthResponse = GetJobAuthResponse;
class QueryCandidatesHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryCandidatesHeaders = QueryCandidatesHeaders;
class QueryCandidatesRequest extends $tea.Model {
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            statId: 'statId',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'string',
            statId: 'string',
            opUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryCandidatesRequest = QueryCandidatesRequest;
class QueryCandidatesResponseBody extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            nextToken: 'nextToken',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': QueryCandidatesResponseBodyList },
            nextToken: 'number',
            totalCount: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryCandidatesResponseBody = QueryCandidatesResponseBody;
class QueryCandidatesResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryCandidatesResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryCandidatesResponse = QueryCandidatesResponse;
class QueryInterviewsHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryInterviewsHeaders = QueryInterviewsHeaders;
class QueryInterviewsRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            candidateId: 'candidateId',
            startTimeBeginMillis: 'startTimeBeginMillis',
            startTimeEndMillis: 'startTimeEndMillis',
            nextToken: 'nextToken',
            size: 'size',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            candidateId: 'string',
            startTimeBeginMillis: 'number',
            startTimeEndMillis: 'number',
            nextToken: 'string',
            size: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryInterviewsRequest = QueryInterviewsRequest;
class QueryInterviewsResponseBody extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            nextToken: 'nextToken',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': QueryInterviewsResponseBodyList },
            nextToken: 'string',
            totalCount: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryInterviewsResponseBody = QueryInterviewsResponseBody;
class QueryInterviewsResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryInterviewsResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryInterviewsResponse = QueryInterviewsResponse;
class ReportMessageStatusHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ReportMessageStatusHeaders = ReportMessageStatusHeaders;
class ReportMessageStatusRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            channel: 'channel',
            errorCode: 'errorCode',
            errorMsg: 'errorMsg',
            messageId: 'messageId',
            receiverUserId: 'receiverUserId',
            senderUserId: 'senderUserId',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            channel: 'string',
            errorCode: 'string',
            errorMsg: 'string',
            messageId: 'string',
            receiverUserId: 'string',
            senderUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ReportMessageStatusRequest = ReportMessageStatusRequest;
class ReportMessageStatusResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ReportMessageStatusResponseBody = ReportMessageStatusResponseBody;
class ReportMessageStatusResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ReportMessageStatusResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ReportMessageStatusResponse = ReportMessageStatusResponse;
class SyncChannelMessageHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncChannelMessageHeaders = SyncChannelMessageHeaders;
class SyncChannelMessageRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            channel: 'channel',
            content: 'content',
            createTime: 'createTime',
            receiverUserId: 'receiverUserId',
            senderUserId: 'senderUserId',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            channel: 'string',
            content: 'string',
            createTime: 'number',
            receiverUserId: 'string',
            senderUserId: 'string',
            uuid: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncChannelMessageRequest = SyncChannelMessageRequest;
class SyncChannelMessageResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncChannelMessageResponseBody = SyncChannelMessageResponseBody;
class SyncChannelMessageResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SyncChannelMessageResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncChannelMessageResponse = SyncChannelMessageResponse;
class UpdateApplicationRegFormHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateApplicationRegFormHeaders = UpdateApplicationRegFormHeaders;
class UpdateApplicationRegFormRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            content: 'content',
            dingPanFile: 'dingPanFile',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            content: 'string',
            dingPanFile: UpdateApplicationRegFormRequestDingPanFile,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateApplicationRegFormRequest = UpdateApplicationRegFormRequest;
class UpdateApplicationRegFormResponseBody extends $tea.Model {
    static names() {
        return {
            creatorUserId: 'creatorUserId',
            formId: 'formId',
            gmtCreateMillis: 'gmtCreateMillis',
            gmtModifiedMillis: 'gmtModifiedMillis',
            status: 'status',
            templateId: 'templateId',
            templateVersion: 'templateVersion',
        };
    }
    static types() {
        return {
            creatorUserId: 'string',
            formId: 'string',
            gmtCreateMillis: 'number',
            gmtModifiedMillis: 'number',
            status: 'number',
            templateId: 'string',
            templateVersion: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateApplicationRegFormResponseBody = UpdateApplicationRegFormResponseBody;
class UpdateApplicationRegFormResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateApplicationRegFormResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateApplicationRegFormResponse = UpdateApplicationRegFormResponse;
class UpdateInterviewSignInInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateInterviewSignInInfoHeaders = UpdateInterviewSignInInfoHeaders;
class UpdateInterviewSignInInfoRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            signInTimeMillis: 'signInTimeMillis',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            signInTimeMillis: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateInterviewSignInInfoRequest = UpdateInterviewSignInInfoRequest;
class UpdateInterviewSignInInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateInterviewSignInInfoResponse = UpdateInterviewSignInInfoResponse;
class UpdateJobDeliverHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateJobDeliverHeaders = UpdateJobDeliverHeaders;
class UpdateJobDeliverRequest extends $tea.Model {
    static names() {
        return {
            bizCode: 'bizCode',
            channelOuterId: 'channelOuterId',
            deliverUserId: 'deliverUserId',
            errorCode: 'errorCode',
            errorMsg: 'errorMsg',
            opTime: 'opTime',
            opUserId: 'opUserId',
            status: 'status',
            jobId: 'jobId',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            channelOuterId: 'string',
            deliverUserId: 'string',
            errorCode: 'string',
            errorMsg: 'string',
            opTime: 'number',
            opUserId: 'string',
            status: 'number',
            jobId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateJobDeliverRequest = UpdateJobDeliverRequest;
class UpdateJobDeliverResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateJobDeliverResponseBody = UpdateJobDeliverResponseBody;
class UpdateJobDeliverResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateJobDeliverResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateJobDeliverResponse = UpdateJobDeliverResponse;
class CollectRecruitJobDetailRequestJobInfoAddress extends $tea.Model {
    static names() {
        return {
            cityCode: 'cityCode',
            detail: 'detail',
            districtCode: 'districtCode',
            latitude: 'latitude',
            longitude: 'longitude',
            name: 'name',
            provinceCode: 'provinceCode',
        };
    }
    static types() {
        return {
            cityCode: 'string',
            detail: 'string',
            districtCode: 'string',
            latitude: 'string',
            longitude: 'string',
            name: 'string',
            provinceCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectRecruitJobDetailRequestJobInfoAddress = CollectRecruitJobDetailRequestJobInfoAddress;
class CollectRecruitJobDetailRequestJobInfoFullTimeInfo extends $tea.Model {
    static names() {
        return {
            maxJobExperience: 'maxJobExperience',
            minJobExperience: 'minJobExperience',
            salaryMonth: 'salaryMonth',
        };
    }
    static types() {
        return {
            maxJobExperience: 'string',
            minJobExperience: 'string',
            salaryMonth: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectRecruitJobDetailRequestJobInfoFullTimeInfo = CollectRecruitJobDetailRequestJobInfoFullTimeInfo;
class CollectRecruitJobDetailRequestJobInfoPartTimeInfo extends $tea.Model {
    static names() {
        return {
            contactNumber: 'contactNumber',
            salaryPeriod: 'salaryPeriod',
            settleType: 'settleType',
            specifyWorkDate: 'specifyWorkDate',
            specifyWorkTime: 'specifyWorkTime',
            workBeginTimeMin: 'workBeginTimeMin',
            workDateType: 'workDateType',
            workEndDate: 'workEndDate',
            workEndTimeMin: 'workEndTimeMin',
            workStartDate: 'workStartDate',
        };
    }
    static types() {
        return {
            contactNumber: 'string',
            salaryPeriod: 'string',
            settleType: 'string',
            specifyWorkDate: 'string',
            specifyWorkTime: 'string',
            workBeginTimeMin: 'string',
            workDateType: 'string',
            workEndDate: 'string',
            workEndTimeMin: 'string',
            workStartDate: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectRecruitJobDetailRequestJobInfoPartTimeInfo = CollectRecruitJobDetailRequestJobInfoPartTimeInfo;
class CollectRecruitJobDetailRequestJobInfo extends $tea.Model {
    static names() {
        return {
            address: 'address',
            category: 'category',
            description: 'description',
            extInfo: 'extInfo',
            fullTimeInfo: 'fullTimeInfo',
            headCount: 'headCount',
            jobNature: 'jobNature',
            jobTags: 'jobTags',
            maxSalary: 'maxSalary',
            minSalary: 'minSalary',
            name: 'name',
            outJobId: 'outJobId',
            partTimeInfo: 'partTimeInfo',
            requiredEdu: 'requiredEdu',
        };
    }
    static types() {
        return {
            address: CollectRecruitJobDetailRequestJobInfoAddress,
            category: 'string',
            description: 'string',
            extInfo: 'string',
            fullTimeInfo: CollectRecruitJobDetailRequestJobInfoFullTimeInfo,
            headCount: 'string',
            jobNature: 'string',
            jobTags: { 'type': 'array', 'itemType': 'string' },
            maxSalary: 'string',
            minSalary: 'string',
            name: 'string',
            outJobId: 'string',
            partTimeInfo: CollectRecruitJobDetailRequestJobInfoPartTimeInfo,
            requiredEdu: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectRecruitJobDetailRequestJobInfo = CollectRecruitJobDetailRequestJobInfo;
class CollectRecruitJobDetailRequestRecruitUserInfo extends $tea.Model {
    static names() {
        return {
            extInfo: 'extInfo',
            outUserId: 'outUserId',
            userMobile: 'userMobile',
            userName: 'userName',
        };
    }
    static types() {
        return {
            extInfo: 'string',
            outUserId: 'string',
            userMobile: 'string',
            userName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectRecruitJobDetailRequestRecruitUserInfo = CollectRecruitJobDetailRequestRecruitUserInfo;
class CollectResumeDetailRequestResumeDataBaseInfo extends $tea.Model {
    static names() {
        return {
            age: 'age',
            avatar: 'avatar',
            beginWorkTime: 'beginWorkTime',
            birthday: 'birthday',
            email: 'email',
            englishName: 'englishName',
            graduateTime: 'graduateTime',
            highestEducation: 'highestEducation',
            jobTitle: 'jobTitle',
            lastSchoolName: 'lastSchoolName',
            married: 'married',
            name: 'name',
            nativePlace: 'nativePlace',
            nowLocation: 'nowLocation',
            personalHonor: 'personalHonor',
            phoneNum: 'phoneNum',
            politicalStatus: 'politicalStatus',
            selfEvaluation: 'selfEvaluation',
            sex: 'sex',
            virtualPhoneNum: 'virtualPhoneNum',
            workingYears: 'workingYears',
        };
    }
    static types() {
        return {
            age: 'number',
            avatar: 'string',
            beginWorkTime: 'string',
            birthday: 'string',
            email: 'string',
            englishName: 'string',
            graduateTime: 'string',
            highestEducation: 'number',
            jobTitle: 'string',
            lastSchoolName: 'string',
            married: 'number',
            name: 'string',
            nativePlace: 'string',
            nowLocation: 'string',
            personalHonor: 'string',
            phoneNum: 'string',
            politicalStatus: 'number',
            selfEvaluation: 'string',
            sex: 'number',
            virtualPhoneNum: 'string',
            workingYears: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequestResumeDataBaseInfo = CollectResumeDetailRequestResumeDataBaseInfo;
class CollectResumeDetailRequestResumeDataCertificates extends $tea.Model {
    static names() {
        return {
            certificateName: 'certificateName',
            grantTime: 'grantTime',
        };
    }
    static types() {
        return {
            certificateName: 'string',
            grantTime: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequestResumeDataCertificates = CollectResumeDetailRequestResumeDataCertificates;
class CollectResumeDetailRequestResumeDataEducationExperiences extends $tea.Model {
    static names() {
        return {
            degree: 'degree',
            department: 'department',
            description: 'description',
            endDate: 'endDate',
            major: 'major',
            schoolName: 'schoolName',
            startDate: 'startDate',
        };
    }
    static types() {
        return {
            degree: 'number',
            department: 'string',
            description: 'string',
            endDate: 'string',
            major: 'string',
            schoolName: 'string',
            startDate: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequestResumeDataEducationExperiences = CollectResumeDetailRequestResumeDataEducationExperiences;
class CollectResumeDetailRequestResumeDataJobExpect extends $tea.Model {
    static names() {
        return {
            jobName: 'jobName',
            locations: 'locations',
            maxSalary: 'maxSalary',
            minSalary: 'minSalary',
            onboardTime: 'onboardTime',
        };
    }
    static types() {
        return {
            jobName: 'string',
            locations: { 'type': 'array', 'itemType': 'string' },
            maxSalary: 'string',
            minSalary: 'string',
            onboardTime: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequestResumeDataJobExpect = CollectResumeDetailRequestResumeDataJobExpect;
class CollectResumeDetailRequestResumeDataLanguageSkill extends $tea.Model {
    static names() {
        return {
            certificateName: 'certificateName',
            languageName: 'languageName',
        };
    }
    static types() {
        return {
            certificateName: 'string',
            languageName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequestResumeDataLanguageSkill = CollectResumeDetailRequestResumeDataLanguageSkill;
class CollectResumeDetailRequestResumeDataTrainingExperiences extends $tea.Model {
    static names() {
        return {
            description: 'description',
            endDate: 'endDate',
            institutionName: 'institutionName',
            location: 'location',
            name: 'name',
            startDate: 'startDate',
        };
    }
    static types() {
        return {
            description: 'string',
            endDate: 'string',
            institutionName: 'string',
            location: 'string',
            name: 'string',
            startDate: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequestResumeDataTrainingExperiences = CollectResumeDetailRequestResumeDataTrainingExperiences;
class CollectResumeDetailRequestResumeDataWorkExperiences extends $tea.Model {
    static names() {
        return {
            companyName: 'companyName',
            department: 'department',
            description: 'description',
            endDate: 'endDate',
            jobTitle: 'jobTitle',
            location: 'location',
            responsibility: 'responsibility',
            startDate: 'startDate',
        };
    }
    static types() {
        return {
            companyName: 'string',
            department: 'string',
            description: 'string',
            endDate: 'string',
            jobTitle: 'string',
            location: 'string',
            responsibility: 'string',
            startDate: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequestResumeDataWorkExperiences = CollectResumeDetailRequestResumeDataWorkExperiences;
class CollectResumeDetailRequestResumeData extends $tea.Model {
    static names() {
        return {
            baseInfo: 'baseInfo',
            certificates: 'certificates',
            educationExperiences: 'educationExperiences',
            jobExpect: 'jobExpect',
            languageSkill: 'languageSkill',
            trainingExperiences: 'trainingExperiences',
            workExperiences: 'workExperiences',
        };
    }
    static types() {
        return {
            baseInfo: CollectResumeDetailRequestResumeDataBaseInfo,
            certificates: { 'type': 'array', 'itemType': CollectResumeDetailRequestResumeDataCertificates },
            educationExperiences: { 'type': 'array', 'itemType': CollectResumeDetailRequestResumeDataEducationExperiences },
            jobExpect: CollectResumeDetailRequestResumeDataJobExpect,
            languageSkill: { 'type': 'array', 'itemType': CollectResumeDetailRequestResumeDataLanguageSkill },
            trainingExperiences: { 'type': 'array', 'itemType': CollectResumeDetailRequestResumeDataTrainingExperiences },
            workExperiences: { 'type': 'array', 'itemType': CollectResumeDetailRequestResumeDataWorkExperiences },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequestResumeData = CollectResumeDetailRequestResumeData;
class CollectResumeDetailRequestResumeFile extends $tea.Model {
    static names() {
        return {
            downloadUrl: 'downloadUrl',
            fileName: 'fileName',
            fileType: 'fileType',
        };
    }
    static types() {
        return {
            downloadUrl: 'string',
            fileName: 'string',
            fileType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeDetailRequestResumeFile = CollectResumeDetailRequestResumeFile;
class CollectResumeMailRequestResumeFile extends $tea.Model {
    static names() {
        return {
            downloadUrl: 'downloadUrl',
            fileName: 'fileName',
            fileType: 'fileType',
        };
    }
    static types() {
        return {
            downloadUrl: 'string',
            fileName: 'string',
            fileType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CollectResumeMailRequestResumeFile = CollectResumeMailRequestResumeFile;
class GetJobAuthResponseBodyJobOwners extends $tea.Model {
    static names() {
        return {
            name: 'name',
            userId: 'userId',
        };
    }
    static types() {
        return {
            name: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobAuthResponseBodyJobOwners = GetJobAuthResponseBodyJobOwners;
class QueryCandidatesResponseBodyList extends $tea.Model {
    static names() {
        return {
            candidateId: 'candidateId',
            corpId: 'corpId',
            entryDate: 'entryDate',
            userId: 'userId',
        };
    }
    static types() {
        return {
            candidateId: 'string',
            corpId: 'string',
            entryDate: 'number',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryCandidatesResponseBodyList = QueryCandidatesResponseBodyList;
class QueryInterviewsResponseBodyListInterviewers extends $tea.Model {
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryInterviewsResponseBodyListInterviewers = QueryInterviewsResponseBodyListInterviewers;
class QueryInterviewsResponseBodyList extends $tea.Model {
    static names() {
        return {
            cancelled: 'cancelled',
            creatorUserId: 'creatorUserId',
            endTimeMillis: 'endTimeMillis',
            interviewId: 'interviewId',
            interviewers: 'interviewers',
            jobId: 'jobId',
            startTimeMillis: 'startTimeMillis',
        };
    }
    static types() {
        return {
            cancelled: 'boolean',
            creatorUserId: 'string',
            endTimeMillis: 'number',
            interviewId: 'string',
            interviewers: { 'type': 'array', 'itemType': QueryInterviewsResponseBodyListInterviewers },
            jobId: 'string',
            startTimeMillis: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryInterviewsResponseBodyList = QueryInterviewsResponseBodyList;
class UpdateApplicationRegFormRequestDingPanFile extends $tea.Model {
    static names() {
        return {
            fileId: 'fileId',
            fileName: 'fileName',
            fileSize: 'fileSize',
            fileType: 'fileType',
            spaceId: 'spaceId',
        };
    }
    static types() {
        return {
            fileId: 'string',
            fileName: 'string',
            fileSize: 'number',
            fileType: 'string',
            spaceId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateApplicationRegFormRequestDingPanFile = UpdateApplicationRegFormRequestDingPanFile;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 
     *
     * @param request AddApplicationRegFormTemplateRequest
     * @param headers AddApplicationRegFormTemplateHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddApplicationRegFormTemplateResponse
     */
    async addApplicationRegFormTemplateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.content)) {
            body["content"] = request.content;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.outerId)) {
            body["outerId"] = request.outerId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddApplicationRegFormTemplate",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/flows/applicationRegForms/templates`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddApplicationRegFormTemplateResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddApplicationRegFormTemplateRequest
     * @return AddApplicationRegFormTemplateResponse
     */
    async addApplicationRegFormTemplate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddApplicationRegFormTemplateHeaders({});
        return await this.addApplicationRegFormTemplateWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request AddFileRequest
     * @param headers AddFileHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddFileResponse
     */
    async addFileWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.fileName)) {
            body["fileName"] = request.fileName;
        }
        if (!tea_util_1.default.isUnset(request.mediaId)) {
            body["mediaId"] = request.mediaId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddFile",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/files`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddFileResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddFileRequest
     * @return AddFileResponse
     */
    async addFile(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddFileHeaders({});
        return await this.addFileWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request AddUserAccountRequest
     * @param headers AddUserAccountHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddUserAccountResponse
     */
    async addUserAccountWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.corpId)) {
            query["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.channelAccountName)) {
            body["channelAccountName"] = request.channelAccountName;
        }
        if (!tea_util_1.default.isUnset(request.channelUserIdentify)) {
            body["channelUserIdentify"] = request.channelUserIdentify;
        }
        if (!tea_util_1.default.isUnset(request.phoneNumber)) {
            body["phoneNumber"] = request.phoneNumber;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddUserAccount",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/channels/users/accounts`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddUserAccountResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddUserAccountRequest
     * @return AddUserAccountResponse
     */
    async addUserAccount(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddUserAccountHeaders({});
        return await this.addUserAccountWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CollectRecruitJobDetailRequest
     * @param headers CollectRecruitJobDetailHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CollectRecruitJobDetailResponse
     */
    async collectRecruitJobDetailWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.channel)) {
            body["channel"] = request.channel;
        }
        if (!tea_util_1.default.isUnset(request.jobInfo)) {
            body["jobInfo"] = request.jobInfo;
        }
        if (!tea_util_1.default.isUnset(request.outCorpId)) {
            body["outCorpId"] = request.outCorpId;
        }
        if (!tea_util_1.default.isUnset(request.outCorpName)) {
            body["outCorpName"] = request.outCorpName;
        }
        if (!tea_util_1.default.isUnset(request.recruitUserInfo)) {
            body["recruitUserInfo"] = request.recruitUserInfo;
        }
        if (!tea_util_1.default.isUnset(request.source)) {
            body["source"] = request.source;
        }
        if (!tea_util_1.default.isUnset(request.updateTime)) {
            body["updateTime"] = request.updateTime;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CollectRecruitJobDetail",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/channels/jobs/import`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CollectRecruitJobDetailResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CollectRecruitJobDetailRequest
     * @return CollectRecruitJobDetailResponse
     */
    async collectRecruitJobDetail(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CollectRecruitJobDetailHeaders({});
        return await this.collectRecruitJobDetailWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CollectResumeDetailRequest
     * @param headers CollectResumeDetailHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CollectResumeDetailResponse
     */
    async collectResumeDetailWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.channelCode)) {
            body["channelCode"] = request.channelCode;
        }
        if (!tea_util_1.default.isUnset(request.channelOuterId)) {
            body["channelOuterId"] = request.channelOuterId;
        }
        if (!tea_util_1.default.isUnset(request.channelTalentId)) {
            body["channelTalentId"] = request.channelTalentId;
        }
        if (!tea_util_1.default.isUnset(request.deliverJobId)) {
            body["deliverJobId"] = request.deliverJobId;
        }
        if (!tea_util_1.default.isUnset(request.optUserId)) {
            body["optUserId"] = request.optUserId;
        }
        if (!tea_util_1.default.isUnset(request.resumeChannelUrl)) {
            body["resumeChannelUrl"] = request.resumeChannelUrl;
        }
        if (!tea_util_1.default.isUnset(request.resumeData)) {
            body["resumeData"] = request.resumeData;
        }
        if (!tea_util_1.default.isUnset(request.resumeFile)) {
            body["resumeFile"] = request.resumeFile;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CollectResumeDetail",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/resumes/details`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CollectResumeDetailResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CollectResumeDetailRequest
     * @return CollectResumeDetailResponse
     */
    async collectResumeDetail(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CollectResumeDetailHeaders({});
        return await this.collectResumeDetailWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CollectResumeMailRequest
     * @param headers CollectResumeMailHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CollectResumeMailResponse
     */
    async collectResumeMailWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.channelCode)) {
            body["channelCode"] = request.channelCode;
        }
        if (!tea_util_1.default.isUnset(request.deliverJobId)) {
            body["deliverJobId"] = request.deliverJobId;
        }
        if (!tea_util_1.default.isUnset(request.fromMailAddress)) {
            body["fromMailAddress"] = request.fromMailAddress;
        }
        if (!tea_util_1.default.isUnset(request.historyMailImport)) {
            body["historyMailImport"] = request.historyMailImport;
        }
        if (!tea_util_1.default.isUnset(request.mailId)) {
            body["mailId"] = request.mailId;
        }
        if (!tea_util_1.default.isUnset(request.mailTitle)) {
            body["mailTitle"] = request.mailTitle;
        }
        if (!tea_util_1.default.isUnset(request.optUserId)) {
            body["optUserId"] = request.optUserId;
        }
        if (!tea_util_1.default.isUnset(request.receiveMailAddress)) {
            body["receiveMailAddress"] = request.receiveMailAddress;
        }
        if (!tea_util_1.default.isUnset(request.receiveMailType)) {
            body["receiveMailType"] = request.receiveMailType;
        }
        if (!tea_util_1.default.isUnset(request.receivedTime)) {
            body["receivedTime"] = request.receivedTime;
        }
        if (!tea_util_1.default.isUnset(request.resumeChannelUrl)) {
            body["resumeChannelUrl"] = request.resumeChannelUrl;
        }
        if (!tea_util_1.default.isUnset(request.resumeFile)) {
            body["resumeFile"] = request.resumeFile;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CollectResumeMail",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/resumes/mails`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CollectResumeMailResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CollectResumeMailRequest
     * @return CollectResumeMailResponse
     */
    async collectResumeMail(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CollectResumeMailHeaders({});
        return await this.collectResumeMailWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request ConfirmRightsRequest
     * @param headers ConfirmRightsHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ConfirmRightsResponse
     */
    async confirmRightsWithOptions(rightsCode, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ConfirmRights",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/rights/${rightsCode}/confirm`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ConfirmRightsResponse({}));
    }
    /**
     * @summary 
     *
     * @param request ConfirmRightsRequest
     * @return ConfirmRightsResponse
     */
    async confirmRights(rightsCode, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ConfirmRightsHeaders({});
        return await this.confirmRightsWithOptions(rightsCode, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request FinishBeginnerTaskRequest
     * @param headers FinishBeginnerTaskHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return FinishBeginnerTaskResponse
     */
    async finishBeginnerTaskWithOptions(taskCode, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.scope)) {
            query["scope"] = request.scope;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "FinishBeginnerTask",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/beginnerTasks/${taskCode}/finish`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new FinishBeginnerTaskResponse({}));
    }
    /**
     * @summary 
     *
     * @param request FinishBeginnerTaskRequest
     * @return FinishBeginnerTaskResponse
     */
    async finishBeginnerTask(taskCode, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new FinishBeginnerTaskHeaders({});
        return await this.finishBeginnerTaskWithOptions(taskCode, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetApplicationRegFormByFlowIdRequest
     * @param headers GetApplicationRegFormByFlowIdHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetApplicationRegFormByFlowIdResponse
     */
    async getApplicationRegFormByFlowIdWithOptions(flowId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetApplicationRegFormByFlowId",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/flows/${flowId}/applicationRegForms`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetApplicationRegFormByFlowIdResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetApplicationRegFormByFlowIdRequest
     * @return GetApplicationRegFormByFlowIdResponse
     */
    async getApplicationRegFormByFlowId(flowId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetApplicationRegFormByFlowIdHeaders({});
        return await this.getApplicationRegFormByFlowIdWithOptions(flowId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetCandidateByPhoneNumberRequest
     * @param headers GetCandidateByPhoneNumberHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetCandidateByPhoneNumberResponse
     */
    async getCandidateByPhoneNumberWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.phoneNumber)) {
            query["phoneNumber"] = request.phoneNumber;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetCandidateByPhoneNumber",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/candidates`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetCandidateByPhoneNumberResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetCandidateByPhoneNumberRequest
     * @return GetCandidateByPhoneNumberResponse
     */
    async getCandidateByPhoneNumber(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetCandidateByPhoneNumberHeaders({});
        return await this.getCandidateByPhoneNumberWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetFileUploadInfoRequest
     * @param headers GetFileUploadInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetFileUploadInfoResponse
     */
    async getFileUploadInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.fileName)) {
            query["fileName"] = request.fileName;
        }
        if (!tea_util_1.default.isUnset(request.fileSize)) {
            query["fileSize"] = request.fileSize;
        }
        if (!tea_util_1.default.isUnset(request.md5)) {
            query["md5"] = request.md5;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetFileUploadInfo",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/files/uploadInfos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetFileUploadInfoResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetFileUploadInfoRequest
     * @return GetFileUploadInfoResponse
     */
    async getFileUploadInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetFileUploadInfoHeaders({});
        return await this.getFileUploadInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetFlowIdByRelationEntityIdRequest
     * @param headers GetFlowIdByRelationEntityIdHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetFlowIdByRelationEntityIdResponse
     */
    async getFlowIdByRelationEntityIdWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.relationEntity)) {
            query["relationEntity"] = request.relationEntity;
        }
        if (!tea_util_1.default.isUnset(request.relationEntityId)) {
            query["relationEntityId"] = request.relationEntityId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetFlowIdByRelationEntityId",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/flows/ids`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetFlowIdByRelationEntityIdResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetFlowIdByRelationEntityIdRequest
     * @return GetFlowIdByRelationEntityIdResponse
     */
    async getFlowIdByRelationEntityId(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetFlowIdByRelationEntityIdHeaders({});
        return await this.getFlowIdByRelationEntityIdWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetJobAuthRequest
     * @param headers GetJobAuthHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetJobAuthResponse
     */
    async getJobAuthWithOptions(jobId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetJobAuth",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/auths/jobs/${jobId}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetJobAuthResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetJobAuthRequest
     * @return GetJobAuthResponse
     */
    async getJobAuth(jobId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetJobAuthHeaders({});
        return await this.getJobAuthWithOptions(jobId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request QueryCandidatesRequest
     * @param headers QueryCandidatesHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryCandidatesResponse
     */
    async queryCandidatesWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            body["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            body["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.statId)) {
            body["statId"] = request.statId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "QueryCandidates",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/candidates/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryCandidatesResponse({}));
    }
    /**
     * @summary 
     *
     * @param request QueryCandidatesRequest
     * @return QueryCandidatesResponse
     */
    async queryCandidates(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryCandidatesHeaders({});
        return await this.queryCandidatesWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request QueryInterviewsRequest
     * @param headers QueryInterviewsHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryInterviewsResponse
     */
    async queryInterviewsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.size)) {
            query["size"] = request.size;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.candidateId)) {
            body["candidateId"] = request.candidateId;
        }
        if (!tea_util_1.default.isUnset(request.startTimeBeginMillis)) {
            body["startTimeBeginMillis"] = request.startTimeBeginMillis;
        }
        if (!tea_util_1.default.isUnset(request.startTimeEndMillis)) {
            body["startTimeEndMillis"] = request.startTimeEndMillis;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "QueryInterviews",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/interviews/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryInterviewsResponse({}));
    }
    /**
     * @summary 
     *
     * @param request QueryInterviewsRequest
     * @return QueryInterviewsResponse
     */
    async queryInterviews(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryInterviewsHeaders({});
        return await this.queryInterviewsWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request ReportMessageStatusRequest
     * @param headers ReportMessageStatusHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ReportMessageStatusResponse
     */
    async reportMessageStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.channel)) {
            body["channel"] = request.channel;
        }
        if (!tea_util_1.default.isUnset(request.errorCode)) {
            body["errorCode"] = request.errorCode;
        }
        if (!tea_util_1.default.isUnset(request.errorMsg)) {
            body["errorMsg"] = request.errorMsg;
        }
        if (!tea_util_1.default.isUnset(request.messageId)) {
            body["messageId"] = request.messageId;
        }
        if (!tea_util_1.default.isUnset(request.receiverUserId)) {
            body["receiverUserId"] = request.receiverUserId;
        }
        if (!tea_util_1.default.isUnset(request.senderUserId)) {
            body["senderUserId"] = request.senderUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ReportMessageStatus",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/channels/messages/statuses/report`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ReportMessageStatusResponse({}));
    }
    /**
     * @summary 
     *
     * @param request ReportMessageStatusRequest
     * @return ReportMessageStatusResponse
     */
    async reportMessageStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ReportMessageStatusHeaders({});
        return await this.reportMessageStatusWithOptions(request, headers, runtime);
    }
    /**
     * @summary IM
     *
     * @param request SyncChannelMessageRequest
     * @param headers SyncChannelMessageHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SyncChannelMessageResponse
     */
    async syncChannelMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.channel)) {
            body["channel"] = request.channel;
        }
        if (!tea_util_1.default.isUnset(request.content)) {
            body["content"] = request.content;
        }
        if (!tea_util_1.default.isUnset(request.createTime)) {
            body["createTime"] = request.createTime;
        }
        if (!tea_util_1.default.isUnset(request.receiverUserId)) {
            body["receiverUserId"] = request.receiverUserId;
        }
        if (!tea_util_1.default.isUnset(request.senderUserId)) {
            body["senderUserId"] = request.senderUserId;
        }
        if (!tea_util_1.default.isUnset(request.uuid)) {
            body["uuid"] = request.uuid;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SyncChannelMessage",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/channels/messages/sync`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SyncChannelMessageResponse({}));
    }
    /**
     * @summary IM
     *
     * @param request SyncChannelMessageRequest
     * @return SyncChannelMessageResponse
     */
    async syncChannelMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SyncChannelMessageHeaders({});
        return await this.syncChannelMessageWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateApplicationRegFormRequest
     * @param headers UpdateApplicationRegFormHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateApplicationRegFormResponse
     */
    async updateApplicationRegFormWithOptions(flowId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.content)) {
            body["content"] = request.content;
        }
        if (!tea_util_1.default.isUnset(request.dingPanFile)) {
            body["dingPanFile"] = request.dingPanFile;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateApplicationRegForm",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/flows/${flowId}/applicationRegForms`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateApplicationRegFormResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateApplicationRegFormRequest
     * @return UpdateApplicationRegFormResponse
     */
    async updateApplicationRegForm(flowId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateApplicationRegFormHeaders({});
        return await this.updateApplicationRegFormWithOptions(flowId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateInterviewSignInInfoRequest
     * @param headers UpdateInterviewSignInInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateInterviewSignInInfoResponse
     */
    async updateInterviewSignInInfoWithOptions(interviewId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.signInTimeMillis)) {
            body["signInTimeMillis"] = request.signInTimeMillis;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateInterviewSignInInfo",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/interviews/${interviewId}/signInInfos`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateInterviewSignInInfoResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateInterviewSignInInfoRequest
     * @return UpdateInterviewSignInInfoResponse
     */
    async updateInterviewSignInInfo(interviewId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateInterviewSignInInfoHeaders({});
        return await this.updateInterviewSignInInfoWithOptions(interviewId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateJobDeliverRequest
     * @param headers UpdateJobDeliverHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateJobDeliverResponse
     */
    async updateJobDeliverWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.jobId)) {
            query["jobId"] = request.jobId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.channelOuterId)) {
            body["channelOuterId"] = request.channelOuterId;
        }
        if (!tea_util_1.default.isUnset(request.deliverUserId)) {
            body["deliverUserId"] = request.deliverUserId;
        }
        if (!tea_util_1.default.isUnset(request.errorCode)) {
            body["errorCode"] = request.errorCode;
        }
        if (!tea_util_1.default.isUnset(request.errorMsg)) {
            body["errorMsg"] = request.errorMsg;
        }
        if (!tea_util_1.default.isUnset(request.opTime)) {
            body["opTime"] = request.opTime;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.status)) {
            body["status"] = request.status;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateJobDeliver",
            version: "ats_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/ats/jobs/deliveryStatus`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateJobDeliverResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateJobDeliverRequest
     * @return UpdateJobDeliverResponse
     */
    async updateJobDeliver(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateJobDeliverHeaders({});
        return await this.updateJobDeliverWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
