// @alicloud/dingtalk@2.1.16/dist/resident_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetResidentInfoResponse = exports.GetResidentInfoResponseBody = exports.GetResidentInfoRequest = exports.GetResidentInfoHeaders = exports.GetPropertyInfoResponse = exports.GetPropertyInfoResponseBody = exports.GetPropertyInfoRequest = exports.GetPropertyInfoHeaders = exports.GetIndustryTypeResponse = exports.GetIndustryTypeResponseBody = exports.GetIndustryTypeHeaders = exports.GetConversationIdResponse = exports.GetConversationIdResponseBody = exports.GetConversationIdRequest = exports.GetConversationIdHeaders = exports.DeleteSpaceResponse = exports.DeleteSpaceResponseBody = exports.DeleteSpaceRequest = exports.DeleteSpaceHeaders = exports.DeleteResidentDepartmentResponse = exports.DeleteResidentDepartmentResponseBody = exports.DeleteResidentDepartmentRequest = exports.DeleteResidentDepartmentHeaders = exports.DeleteResidentBlackBoardResponse = exports.DeleteResidentBlackBoardResponseBody = exports.DeleteResidentBlackBoardRequest = exports.DeleteResidentBlackBoardHeaders = exports.CreateSpaceResponse = exports.CreateSpaceResponseBody = exports.CreateSpaceRequest = exports.CreateSpaceHeaders = exports.CreateResidentBlackBoardResponse = exports.CreateResidentBlackBoardResponseBody = exports.CreateResidentBlackBoardRequest = exports.CreateResidentBlackBoardHeaders = exports.AddResidentUsersResponse = exports.AddResidentUsersResponseBody = exports.AddResidentUsersRequest = exports.AddResidentUsersHeaders = exports.AddResidentMemberResponse = exports.AddResidentMemberResponseBody = exports.AddResidentMemberRequest = exports.AddResidentMemberHeaders = exports.AddResidentDepartmentResponse = exports.AddResidentDepartmentResponseBody = exports.AddResidentDepartmentRequest = exports.AddResidentDepartmentHeaders = exports.AddPointResponse = exports.AddPointRequest = exports.AddPointHeaders = void 0;
exports.UpdateResideceGroupRequest = exports.UpdateResideceGroupHeaders = exports.SearchResidentResponse = exports.SearchResidentResponseBody = exports.SearchResidentRequest = exports.SearchResidentHeaders = exports.RemoveResidentUserResponse = exports.RemoveResidentUserResponseBody = exports.RemoveResidentUserRequest = exports.RemoveResidentUserHeaders = exports.RemoveResidentMemberResponse = exports.RemoveResidentMemberResponseBody = exports.RemoveResidentMemberRequest = exports.RemoveResidentMemberHeaders = exports.PagePointHistoryResponse = exports.PagePointHistoryResponseBody = exports.PagePointHistoryRequest = exports.PagePointHistoryHeaders = exports.ListUserIndustryRolesResponse = exports.ListUserIndustryRolesResponseBody = exports.ListUserIndustryRolesRequest = exports.ListUserIndustryRolesHeaders = exports.ListUncheckUsersResponse = exports.ListUncheckUsersResponseBody = exports.ListUncheckUsersRequest = exports.ListUncheckUsersHeaders = exports.ListSubSpaceResponse = exports.ListSubSpaceResponseBody = exports.ListSubSpaceRequest = exports.ListSubSpaceHeaders = exports.ListPointRulesResponse = exports.ListPointRulesResponseBody = exports.ListPointRulesRequest = exports.ListPointRulesHeaders = exports.ListIndustryRoleUsersResponse = exports.ListIndustryRoleUsersResponseBody = exports.ListIndustryRoleUsersRequest = exports.ListIndustryRoleUsersHeaders = exports.GetSpacesInfoResponse = exports.GetSpacesInfoResponseBody = exports.GetSpacesInfoRequest = exports.GetSpacesInfoHeaders = exports.GetSpaceIdByTypeResponse = exports.GetSpaceIdByTypeResponseBody = exports.GetSpaceIdByTypeRequest = exports.GetSpaceIdByTypeHeaders = exports.GetResidentMembersInfoResponse = exports.GetResidentMembersInfoResponseBody = exports.GetResidentMembersInfoRequest = exports.GetResidentMembersInfoHeaders = void 0;
exports.UpdateSpaceRequestSpaceInfoVOList = exports.UpdateResidentUserRequestExtField = exports.UpdateResidentMemberRequestResidentUpdateInfo = exports.SearchResidentResponseBodyResidenceList = exports.PagePointHistoryResponseBodyPointRecordList = exports.ListUserIndustryRolesResponseBodyRoleList = exports.ListUncheckUsersResponseBodyValues = exports.ListSubSpaceResponseBodySpaceList = exports.ListPointRulesResponseBodyPointRuleList = exports.GetSpacesInfoResponseBodySpaceList = exports.GetResidentMembersInfoResponseBodyResidenceList = exports.GetResidentInfoResponseBodyProjectManager = exports.DeleteSpaceResponseBodyDelFailedDept = exports.AddResidentUsersRequestExtField = exports.AddResidentMemberRequestResidentAddInfo = exports.UpdateSpaceResponse = exports.UpdateSpaceResponseBody = exports.UpdateSpaceRequest = exports.UpdateSpaceHeaders = exports.UpdateResidentUserResponse = exports.UpdateResidentUserResponseBody = exports.UpdateResidentUserRequest = exports.UpdateResidentUserHeaders = exports.UpdateResidentMemberResponse = exports.UpdateResidentMemberResponseBody = exports.UpdateResidentMemberRequest = exports.UpdateResidentMemberHeaders = exports.UpdateResidentInfoResponse = exports.UpdateResidentInfoResponseBody = exports.UpdateResidentInfoRequest = exports.UpdateResidentInfoHeaders = exports.UpdateResidentBlackBoardResponse = exports.UpdateResidentBlackBoardResponseBody = exports.UpdateResidentBlackBoardRequest = exports.UpdateResidentBlackBoardHeaders = exports.UpdateResidenceResponse = exports.UpdateResidenceResponseBody = exports.UpdateResidenceRequest = exports.UpdateResidenceHeaders = exports.UpdateResideceGroupResponse = exports.UpdateResideceGroupResponseBody = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AddPointHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddPointHeaders = AddPointHeaders;
class AddPointRequest extends $tea.Model {
    static names() {
        return {
            actionTime: 'actionTime',
            isCircle: 'isCircle',
            ruleCode: 'ruleCode',
            ruleName: 'ruleName',
            score: 'score',
            userId: 'userId',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            actionTime: 'number',
            isCircle: 'boolean',
            ruleCode: 'string',
            ruleName: 'string',
            score: 'number',
            userId: 'string',
            uuid: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddPointRequest = AddPointRequest;
class AddPointResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddPointResponse = AddPointResponse;
class AddResidentDepartmentHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentDepartmentHeaders = AddResidentDepartmentHeaders;
class AddResidentDepartmentRequest extends $tea.Model {
    static names() {
        return {
            departmentName: 'departmentName',
            isResidenceGroup: 'isResidenceGroup',
            parentDepartmentId: 'parentDepartmentId',
        };
    }
    static types() {
        return {
            departmentName: 'string',
            isResidenceGroup: 'boolean',
            parentDepartmentId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentDepartmentRequest = AddResidentDepartmentRequest;
class AddResidentDepartmentResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentDepartmentResponseBody = AddResidentDepartmentResponseBody;
class AddResidentDepartmentResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddResidentDepartmentResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentDepartmentResponse = AddResidentDepartmentResponse;
class AddResidentMemberHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentMemberHeaders = AddResidentMemberHeaders;
class AddResidentMemberRequest extends $tea.Model {
    static names() {
        return {
            residentAddInfo: 'residentAddInfo',
        };
    }
    static types() {
        return {
            residentAddInfo: AddResidentMemberRequestResidentAddInfo,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentMemberRequest = AddResidentMemberRequest;
class AddResidentMemberResponseBody extends $tea.Model {
    static names() {
        return {
            status: 'status',
            unionId: 'unionId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            status: 'number',
            unionId: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentMemberResponseBody = AddResidentMemberResponseBody;
class AddResidentMemberResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddResidentMemberResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentMemberResponse = AddResidentMemberResponse;
class AddResidentUsersHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentUsersHeaders = AddResidentUsersHeaders;
class AddResidentUsersRequest extends $tea.Model {
    static names() {
        return {
            address: 'address',
            departmentId: 'departmentId',
            extField: 'extField',
            isLeaseholder: 'isLeaseholder',
            mobile: 'mobile',
            relateType: 'relateType',
            userName: 'userName',
        };
    }
    static types() {
        return {
            address: 'string',
            departmentId: 'number',
            extField: { 'type': 'array', 'itemType': AddResidentUsersRequestExtField },
            isLeaseholder: 'boolean',
            mobile: 'string',
            relateType: 'string',
            userName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentUsersRequest = AddResidentUsersRequest;
class AddResidentUsersResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentUsersResponseBody = AddResidentUsersResponseBody;
class AddResidentUsersResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddResidentUsersResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentUsersResponse = AddResidentUsersResponse;
class CreateResidentBlackBoardHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateResidentBlackBoardHeaders = CreateResidentBlackBoardHeaders;
class CreateResidentBlackBoardRequest extends $tea.Model {
    static names() {
        return {
            context: 'context',
            mediaId: 'mediaId',
            sendTime: 'sendTime',
            title: 'title',
        };
    }
    static types() {
        return {
            context: 'string',
            mediaId: 'string',
            sendTime: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateResidentBlackBoardRequest = CreateResidentBlackBoardRequest;
class CreateResidentBlackBoardResponseBody extends $tea.Model {
    static names() {
        return {
            blackBoardId: 'blackBoardId',
        };
    }
    static types() {
        return {
            blackBoardId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateResidentBlackBoardResponseBody = CreateResidentBlackBoardResponseBody;
class CreateResidentBlackBoardResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateResidentBlackBoardResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateResidentBlackBoardResponse = CreateResidentBlackBoardResponse;
class CreateSpaceHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateSpaceHeaders = CreateSpaceHeaders;
class CreateSpaceRequest extends $tea.Model {
    static names() {
        return {
            billingArea: 'billingArea',
            buildingArea: 'buildingArea',
            floor: 'floor',
            houseState: 'houseState',
            name: 'name',
            parentDeptId: 'parentDeptId',
            tagCode: 'tagCode',
            type: 'type',
        };
    }
    static types() {
        return {
            billingArea: 'number',
            buildingArea: 'number',
            floor: 'string',
            houseState: 'number',
            name: 'string',
            parentDeptId: 'string',
            tagCode: 'string',
            type: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateSpaceRequest = CreateSpaceRequest;
class CreateSpaceResponseBody extends $tea.Model {
    static names() {
        return {
            deptId: 'deptId',
        };
    }
    static types() {
        return {
            deptId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateSpaceResponseBody = CreateSpaceResponseBody;
class CreateSpaceResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateSpaceResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateSpaceResponse = CreateSpaceResponse;
class DeleteResidentBlackBoardHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteResidentBlackBoardHeaders = DeleteResidentBlackBoardHeaders;
class DeleteResidentBlackBoardRequest extends $tea.Model {
    static names() {
        return {
            blackboardId: 'blackboardId',
        };
    }
    static types() {
        return {
            blackboardId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteResidentBlackBoardRequest = DeleteResidentBlackBoardRequest;
class DeleteResidentBlackBoardResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteResidentBlackBoardResponseBody = DeleteResidentBlackBoardResponseBody;
class DeleteResidentBlackBoardResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteResidentBlackBoardResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteResidentBlackBoardResponse = DeleteResidentBlackBoardResponse;
class DeleteResidentDepartmentHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteResidentDepartmentHeaders = DeleteResidentDepartmentHeaders;
class DeleteResidentDepartmentRequest extends $tea.Model {
    static names() {
        return {
            departmentId: 'departmentId',
        };
    }
    static types() {
        return {
            departmentId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteResidentDepartmentRequest = DeleteResidentDepartmentRequest;
class DeleteResidentDepartmentResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteResidentDepartmentResponseBody = DeleteResidentDepartmentResponseBody;
class DeleteResidentDepartmentResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteResidentDepartmentResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteResidentDepartmentResponse = DeleteResidentDepartmentResponse;
class DeleteSpaceHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteSpaceHeaders = DeleteSpaceHeaders;
class DeleteSpaceRequest extends $tea.Model {
    static names() {
        return {
            deptIds: 'deptIds',
        };
    }
    static types() {
        return {
            deptIds: { 'type': 'array', 'itemType': 'number' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteSpaceRequest = DeleteSpaceRequest;
class DeleteSpaceResponseBody extends $tea.Model {
    static names() {
        return {
            delFailedDept: 'delFailedDept',
            delSuccessCount: 'delSuccessCount',
        };
    }
    static types() {
        return {
            delFailedDept: { 'type': 'array', 'itemType': DeleteSpaceResponseBodyDelFailedDept },
            delSuccessCount: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteSpaceResponseBody = DeleteSpaceResponseBody;
class DeleteSpaceResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteSpaceResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteSpaceResponse = DeleteSpaceResponse;
class GetConversationIdHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetConversationIdHeaders = GetConversationIdHeaders;
class GetConversationIdRequest extends $tea.Model {
    static names() {
        return {
            chatId: 'chatId',
        };
    }
    static types() {
        return {
            chatId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetConversationIdRequest = GetConversationIdRequest;
class GetConversationIdResponseBody extends $tea.Model {
    static names() {
        return {
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetConversationIdResponseBody = GetConversationIdResponseBody;
class GetConversationIdResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetConversationIdResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetConversationIdResponse = GetConversationIdResponse;
class GetIndustryTypeHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetIndustryTypeHeaders = GetIndustryTypeHeaders;
class GetIndustryTypeResponseBody extends $tea.Model {
    static names() {
        return {
            industryType: 'industryType',
        };
    }
    static types() {
        return {
            industryType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetIndustryTypeResponseBody = GetIndustryTypeResponseBody;
class GetIndustryTypeResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetIndustryTypeResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetIndustryTypeResponse = GetIndustryTypeResponse;
class GetPropertyInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPropertyInfoHeaders = GetPropertyInfoHeaders;
class GetPropertyInfoRequest extends $tea.Model {
    static names() {
        return {
            propertyCorpId: 'propertyCorpId',
        };
    }
    static types() {
        return {
            propertyCorpId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPropertyInfoRequest = GetPropertyInfoRequest;
class GetPropertyInfoResponseBody extends $tea.Model {
    static names() {
        return {
            adminName: 'adminName',
            adminUserId: 'adminUserId',
            name: 'name',
            orgId: 'orgId',
            unifiedSocialCredit: 'unifiedSocialCredit',
        };
    }
    static types() {
        return {
            adminName: 'string',
            adminUserId: 'string',
            name: 'string',
            orgId: 'number',
            unifiedSocialCredit: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPropertyInfoResponseBody = GetPropertyInfoResponseBody;
class GetPropertyInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetPropertyInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPropertyInfoResponse = GetPropertyInfoResponse;
class GetResidentInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentInfoHeaders = GetResidentInfoHeaders;
class GetResidentInfoRequest extends $tea.Model {
    static names() {
        return {
            residentCorpId: 'residentCorpId',
        };
    }
    static types() {
        return {
            residentCorpId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentInfoRequest = GetResidentInfoRequest;
class GetResidentInfoResponseBody extends $tea.Model {
    static names() {
        return {
            address: 'address',
            allUserGroupOpenConversationId: 'allUserGroupOpenConversationId',
            allUserGroupOwnerUserId: 'allUserGroupOwnerUserId',
            buildingArea: 'buildingArea',
            cityId: 'cityId',
            contactMode: 'contactMode',
            countyId: 'countyId',
            deliveryTime: 'deliveryTime',
            location: 'location',
            name: 'name',
            projectManager: 'projectManager',
            propertyDeptGroupOpenConversationId: 'propertyDeptGroupOpenConversationId',
            propertyDeptGroupOwnerUserId: 'propertyDeptGroupOwnerUserId',
            provId: 'provId',
            scopeEast: 'scopeEast',
            scopeNorth: 'scopeNorth',
            scopeSouth: 'scopeSouth',
            scopeWest: 'scopeWest',
            telephone: 'telephone',
            townId: 'townId',
            type: 'type',
        };
    }
    static types() {
        return {
            address: 'string',
            allUserGroupOpenConversationId: 'string',
            allUserGroupOwnerUserId: 'string',
            buildingArea: 'number',
            cityId: 'number',
            contactMode: 'number',
            countyId: 'number',
            deliveryTime: 'number',
            location: 'string',
            name: 'string',
            projectManager: GetResidentInfoResponseBodyProjectManager,
            propertyDeptGroupOpenConversationId: 'string',
            propertyDeptGroupOwnerUserId: 'string',
            provId: 'number',
            scopeEast: 'string',
            scopeNorth: 'string',
            scopeSouth: 'string',
            scopeWest: 'string',
            telephone: 'string',
            townId: 'number',
            type: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentInfoResponseBody = GetResidentInfoResponseBody;
class GetResidentInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetResidentInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentInfoResponse = GetResidentInfoResponse;
class GetResidentMembersInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentMembersInfoHeaders = GetResidentMembersInfoHeaders;
class GetResidentMembersInfoRequest extends $tea.Model {
    static names() {
        return {
            residentCropId: 'residentCropId',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            residentCropId: 'string',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentMembersInfoRequest = GetResidentMembersInfoRequest;
class GetResidentMembersInfoResponseBody extends $tea.Model {
    static names() {
        return {
            residenceList: 'residenceList',
        };
    }
    static types() {
        return {
            residenceList: { 'type': 'array', 'itemType': GetResidentMembersInfoResponseBodyResidenceList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentMembersInfoResponseBody = GetResidentMembersInfoResponseBody;
class GetResidentMembersInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetResidentMembersInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentMembersInfoResponse = GetResidentMembersInfoResponse;
class GetSpaceIdByTypeHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSpaceIdByTypeHeaders = GetSpaceIdByTypeHeaders;
class GetSpaceIdByTypeRequest extends $tea.Model {
    static names() {
        return {
            departmentType: 'departmentType',
        };
    }
    static types() {
        return {
            departmentType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSpaceIdByTypeRequest = GetSpaceIdByTypeRequest;
class GetSpaceIdByTypeResponseBody extends $tea.Model {
    static names() {
        return {
            referId: 'referId',
        };
    }
    static types() {
        return {
            referId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSpaceIdByTypeResponseBody = GetSpaceIdByTypeResponseBody;
class GetSpaceIdByTypeResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetSpaceIdByTypeResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSpaceIdByTypeResponse = GetSpaceIdByTypeResponse;
class GetSpacesInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSpacesInfoHeaders = GetSpacesInfoHeaders;
class GetSpacesInfoRequest extends $tea.Model {
    static names() {
        return {
            referIds: 'referIds',
            residentCorpId: 'residentCorpId',
        };
    }
    static types() {
        return {
            referIds: { 'type': 'array', 'itemType': 'number' },
            residentCorpId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSpacesInfoRequest = GetSpacesInfoRequest;
class GetSpacesInfoResponseBody extends $tea.Model {
    static names() {
        return {
            spaceList: 'spaceList',
        };
    }
    static types() {
        return {
            spaceList: { 'type': 'array', 'itemType': GetSpacesInfoResponseBodySpaceList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSpacesInfoResponseBody = GetSpacesInfoResponseBody;
class GetSpacesInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetSpacesInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSpacesInfoResponse = GetSpacesInfoResponse;
class ListIndustryRoleUsersHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListIndustryRoleUsersHeaders = ListIndustryRoleUsersHeaders;
class ListIndustryRoleUsersRequest extends $tea.Model {
    static names() {
        return {
            tagCode: 'tagCode',
        };
    }
    static types() {
        return {
            tagCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListIndustryRoleUsersRequest = ListIndustryRoleUsersRequest;
class ListIndustryRoleUsersResponseBody extends $tea.Model {
    static names() {
        return {
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListIndustryRoleUsersResponseBody = ListIndustryRoleUsersResponseBody;
class ListIndustryRoleUsersResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListIndustryRoleUsersResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListIndustryRoleUsersResponse = ListIndustryRoleUsersResponse;
class ListPointRulesHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListPointRulesHeaders = ListPointRulesHeaders;
class ListPointRulesRequest extends $tea.Model {
    static names() {
        return {
            isCircle: 'isCircle',
        };
    }
    static types() {
        return {
            isCircle: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListPointRulesRequest = ListPointRulesRequest;
class ListPointRulesResponseBody extends $tea.Model {
    static names() {
        return {
            pointRuleList: 'pointRuleList',
        };
    }
    static types() {
        return {
            pointRuleList: { 'type': 'array', 'itemType': ListPointRulesResponseBodyPointRuleList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListPointRulesResponseBody = ListPointRulesResponseBody;
class ListPointRulesResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListPointRulesResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListPointRulesResponse = ListPointRulesResponse;
class ListSubSpaceHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListSubSpaceHeaders = ListSubSpaceHeaders;
class ListSubSpaceRequest extends $tea.Model {
    static names() {
        return {
            referId: 'referId',
            residentCorpId: 'residentCorpId',
        };
    }
    static types() {
        return {
            referId: 'number',
            residentCorpId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListSubSpaceRequest = ListSubSpaceRequest;
class ListSubSpaceResponseBody extends $tea.Model {
    static names() {
        return {
            spaceList: 'spaceList',
        };
    }
    static types() {
        return {
            spaceList: { 'type': 'array', 'itemType': ListSubSpaceResponseBodySpaceList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListSubSpaceResponseBody = ListSubSpaceResponseBody;
class ListSubSpaceResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListSubSpaceResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListSubSpaceResponse = ListSubSpaceResponse;
class ListUncheckUsersHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUncheckUsersHeaders = ListUncheckUsersHeaders;
class ListUncheckUsersRequest extends $tea.Model {
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            startTime: 'startTime',
            status: 'status',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'number',
            startTime: 'number',
            status: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUncheckUsersRequest = ListUncheckUsersRequest;
class ListUncheckUsersResponseBody extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            nextToken: 'nextToken',
            values: 'values',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            nextToken: 'number',
            values: { 'type': 'array', 'itemType': ListUncheckUsersResponseBodyValues },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUncheckUsersResponseBody = ListUncheckUsersResponseBody;
class ListUncheckUsersResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListUncheckUsersResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUncheckUsersResponse = ListUncheckUsersResponse;
class ListUserIndustryRolesHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUserIndustryRolesHeaders = ListUserIndustryRolesHeaders;
class ListUserIndustryRolesRequest extends $tea.Model {
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUserIndustryRolesRequest = ListUserIndustryRolesRequest;
class ListUserIndustryRolesResponseBody extends $tea.Model {
    static names() {
        return {
            roleList: 'roleList',
        };
    }
    static types() {
        return {
            roleList: { 'type': 'array', 'itemType': ListUserIndustryRolesResponseBodyRoleList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUserIndustryRolesResponseBody = ListUserIndustryRolesResponseBody;
class ListUserIndustryRolesResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListUserIndustryRolesResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUserIndustryRolesResponse = ListUserIndustryRolesResponse;
class PagePointHistoryHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PagePointHistoryHeaders = PagePointHistoryHeaders;
class PagePointHistoryRequest extends $tea.Model {
    static names() {
        return {
            endTime: 'endTime',
            isCircle: 'isCircle',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            startTime: 'startTime',
            userId: 'userId',
        };
    }
    static types() {
        return {
            endTime: 'number',
            isCircle: 'boolean',
            maxResults: 'number',
            nextToken: 'number',
            startTime: 'number',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PagePointHistoryRequest = PagePointHistoryRequest;
class PagePointHistoryResponseBody extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            nextToken: 'nextToken',
            pointRecordList: 'pointRecordList',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            nextToken: 'number',
            pointRecordList: { 'type': 'array', 'itemType': PagePointHistoryResponseBodyPointRecordList },
            totalCount: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PagePointHistoryResponseBody = PagePointHistoryResponseBody;
class PagePointHistoryResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PagePointHistoryResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PagePointHistoryResponse = PagePointHistoryResponse;
class RemoveResidentMemberHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveResidentMemberHeaders = RemoveResidentMemberHeaders;
class RemoveResidentMemberRequest extends $tea.Model {
    static names() {
        return {
            deptId: 'deptId',
            unionId: 'unionId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            deptId: 'number',
            unionId: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveResidentMemberRequest = RemoveResidentMemberRequest;
class RemoveResidentMemberResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveResidentMemberResponseBody = RemoveResidentMemberResponseBody;
class RemoveResidentMemberResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RemoveResidentMemberResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveResidentMemberResponse = RemoveResidentMemberResponse;
class RemoveResidentUserHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveResidentUserHeaders = RemoveResidentUserHeaders;
class RemoveResidentUserRequest extends $tea.Model {
    static names() {
        return {
            departmentId: 'departmentId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            departmentId: 'number',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveResidentUserRequest = RemoveResidentUserRequest;
class RemoveResidentUserResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveResidentUserResponseBody = RemoveResidentUserResponseBody;
class RemoveResidentUserResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RemoveResidentUserResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveResidentUserResponse = RemoveResidentUserResponse;
class SearchResidentHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchResidentHeaders = SearchResidentHeaders;
class SearchResidentRequest extends $tea.Model {
    static names() {
        return {
            residentCropId: 'residentCropId',
            searchWord: 'searchWord',
        };
    }
    static types() {
        return {
            residentCropId: 'string',
            searchWord: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchResidentRequest = SearchResidentRequest;
class SearchResidentResponseBody extends $tea.Model {
    static names() {
        return {
            residenceList: 'residenceList',
        };
    }
    static types() {
        return {
            residenceList: { 'type': 'array', 'itemType': SearchResidentResponseBodyResidenceList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchResidentResponseBody = SearchResidentResponseBody;
class SearchResidentResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SearchResidentResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchResidentResponse = SearchResidentResponse;
class UpdateResideceGroupHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResideceGroupHeaders = UpdateResideceGroupHeaders;
class UpdateResideceGroupRequest extends $tea.Model {
    static names() {
        return {
            departmentId: 'departmentId',
            departmentName: 'departmentName',
            managerUserId: 'managerUserId',
        };
    }
    static types() {
        return {
            departmentId: 'number',
            departmentName: 'string',
            managerUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResideceGroupRequest = UpdateResideceGroupRequest;
class UpdateResideceGroupResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResideceGroupResponseBody = UpdateResideceGroupResponseBody;
class UpdateResideceGroupResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateResideceGroupResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResideceGroupResponse = UpdateResideceGroupResponse;
class UpdateResidenceHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidenceHeaders = UpdateResidenceHeaders;
class UpdateResidenceRequest extends $tea.Model {
    static names() {
        return {
            departmentId: 'departmentId',
            departmentName: 'departmentName',
            destitute: 'destitute',
            grid: 'grid',
            homeTel: 'homeTel',
            managerUserId: 'managerUserId',
            parentDepartmentId: 'parentDepartmentId',
        };
    }
    static types() {
        return {
            departmentId: 'number',
            departmentName: 'string',
            destitute: 'boolean',
            grid: 'string',
            homeTel: 'string',
            managerUserId: 'string',
            parentDepartmentId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidenceRequest = UpdateResidenceRequest;
class UpdateResidenceResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidenceResponseBody = UpdateResidenceResponseBody;
class UpdateResidenceResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateResidenceResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidenceResponse = UpdateResidenceResponse;
class UpdateResidentBlackBoardHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentBlackBoardHeaders = UpdateResidentBlackBoardHeaders;
class UpdateResidentBlackBoardRequest extends $tea.Model {
    static names() {
        return {
            blackboardId: 'blackboardId',
            context: 'context',
            mediaId: 'mediaId',
            title: 'title',
        };
    }
    static types() {
        return {
            blackboardId: 'string',
            context: 'string',
            mediaId: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentBlackBoardRequest = UpdateResidentBlackBoardRequest;
class UpdateResidentBlackBoardResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentBlackBoardResponseBody = UpdateResidentBlackBoardResponseBody;
class UpdateResidentBlackBoardResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateResidentBlackBoardResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentBlackBoardResponse = UpdateResidentBlackBoardResponse;
class UpdateResidentInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentInfoHeaders = UpdateResidentInfoHeaders;
class UpdateResidentInfoRequest extends $tea.Model {
    static names() {
        return {
            address: 'address',
            buildingArea: 'buildingArea',
            cityName: 'cityName',
            communityType: 'communityType',
            countyName: 'countyName',
            location: 'location',
            name: 'name',
            provName: 'provName',
            state: 'state',
            telephone: 'telephone',
        };
    }
    static types() {
        return {
            address: 'string',
            buildingArea: 'number',
            cityName: 'string',
            communityType: 'number',
            countyName: 'string',
            location: 'string',
            name: 'string',
            provName: 'string',
            state: 'number',
            telephone: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentInfoRequest = UpdateResidentInfoRequest;
class UpdateResidentInfoResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentInfoResponseBody = UpdateResidentInfoResponseBody;
class UpdateResidentInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateResidentInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentInfoResponse = UpdateResidentInfoResponse;
class UpdateResidentMemberHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentMemberHeaders = UpdateResidentMemberHeaders;
class UpdateResidentMemberRequest extends $tea.Model {
    static names() {
        return {
            residentUpdateInfo: 'residentUpdateInfo',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            residentUpdateInfo: UpdateResidentMemberRequestResidentUpdateInfo,
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentMemberRequest = UpdateResidentMemberRequest;
class UpdateResidentMemberResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentMemberResponseBody = UpdateResidentMemberResponseBody;
class UpdateResidentMemberResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateResidentMemberResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentMemberResponse = UpdateResidentMemberResponse;
class UpdateResidentUserHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentUserHeaders = UpdateResidentUserHeaders;
class UpdateResidentUserRequest extends $tea.Model {
    static names() {
        return {
            address: 'address',
            departmentId: 'departmentId',
            extField: 'extField',
            isRetainOldDept: 'isRetainOldDept',
            mobile: 'mobile',
            oldDepartmentId: 'oldDepartmentId',
            relateType: 'relateType',
            userId: 'userId',
            userName: 'userName',
        };
    }
    static types() {
        return {
            address: 'string',
            departmentId: 'number',
            extField: { 'type': 'array', 'itemType': UpdateResidentUserRequestExtField },
            isRetainOldDept: 'boolean',
            mobile: 'string',
            oldDepartmentId: 'number',
            relateType: 'string',
            userId: 'string',
            userName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentUserRequest = UpdateResidentUserRequest;
class UpdateResidentUserResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentUserResponseBody = UpdateResidentUserResponseBody;
class UpdateResidentUserResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateResidentUserResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentUserResponse = UpdateResidentUserResponse;
class UpdateSpaceHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateSpaceHeaders = UpdateSpaceHeaders;
class UpdateSpaceRequest extends $tea.Model {
    static names() {
        return {
            spaceInfoVOList: 'spaceInfoVOList',
        };
    }
    static types() {
        return {
            spaceInfoVOList: { 'type': 'array', 'itemType': UpdateSpaceRequestSpaceInfoVOList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateSpaceRequest = UpdateSpaceRequest;
class UpdateSpaceResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateSpaceResponseBody = UpdateSpaceResponseBody;
class UpdateSpaceResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateSpaceResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateSpaceResponse = UpdateSpaceResponse;
class AddResidentMemberRequestResidentAddInfo extends $tea.Model {
    static names() {
        return {
            deptId: 'deptId',
            isPropertyOwner: 'isPropertyOwner',
            memberDeptExtension: 'memberDeptExtension',
            mobile: 'mobile',
            name: 'name',
            relateType: 'relateType',
        };
    }
    static types() {
        return {
            deptId: 'number',
            isPropertyOwner: 'boolean',
            memberDeptExtension: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            mobile: 'string',
            name: 'string',
            relateType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentMemberRequestResidentAddInfo = AddResidentMemberRequestResidentAddInfo;
class AddResidentUsersRequestExtField extends $tea.Model {
    static names() {
        return {
            itemName: 'itemName',
            itemValue: 'itemValue',
        };
    }
    static types() {
        return {
            itemName: 'string',
            itemValue: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddResidentUsersRequestExtField = AddResidentUsersRequestExtField;
class DeleteSpaceResponseBodyDelFailedDept extends $tea.Model {
    static names() {
        return {
            deptId: 'deptId',
        };
    }
    static types() {
        return {
            deptId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteSpaceResponseBodyDelFailedDept = DeleteSpaceResponseBodyDelFailedDept;
class GetResidentInfoResponseBodyProjectManager extends $tea.Model {
    static names() {
        return {
            avatar: 'avatar',
            userId: 'userId',
            userName: 'userName',
        };
    }
    static types() {
        return {
            avatar: 'string',
            userId: 'string',
            userName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentInfoResponseBodyProjectManager = GetResidentInfoResponseBodyProjectManager;
class GetResidentMembersInfoResponseBodyResidenceList extends $tea.Model {
    static names() {
        return {
            active: 'active',
            extField: 'extField',
            isPropertyOwner: 'isPropertyOwner',
            name: 'name',
            relateType: 'relateType',
        };
    }
    static types() {
        return {
            active: 'boolean',
            extField: 'string',
            isPropertyOwner: 'boolean',
            name: 'string',
            relateType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetResidentMembersInfoResponseBodyResidenceList = GetResidentMembersInfoResponseBodyResidenceList;
class GetSpacesInfoResponseBodySpaceList extends $tea.Model {
    static names() {
        return {
            billingArea: 'billingArea',
            buildingArea: 'buildingArea',
            floor: 'floor',
            houseState: 'houseState',
            isVirtual: 'isVirtual',
            parentReferId: 'parentReferId',
            referId: 'referId',
            spaceName: 'spaceName',
            tagCode: 'tagCode',
            type: 'type',
        };
    }
    static types() {
        return {
            billingArea: 'number',
            buildingArea: 'number',
            floor: 'string',
            houseState: 'number',
            isVirtual: 'number',
            parentReferId: 'number',
            referId: 'number',
            spaceName: 'string',
            tagCode: 'string',
            type: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSpacesInfoResponseBodySpaceList = GetSpacesInfoResponseBodySpaceList;
class ListPointRulesResponseBodyPointRuleList extends $tea.Model {
    static names() {
        return {
            dayLimitTimes: 'dayLimitTimes',
            extension: 'extension',
            groupId: 'groupId',
            orderId: 'orderId',
            ruleCode: 'ruleCode',
            ruleName: 'ruleName',
            score: 'score',
            status: 'status',
        };
    }
    static types() {
        return {
            dayLimitTimes: 'number',
            extension: 'string',
            groupId: 'number',
            orderId: 'number',
            ruleCode: 'string',
            ruleName: 'string',
            score: 'number',
            status: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListPointRulesResponseBodyPointRuleList = ListPointRulesResponseBodyPointRuleList;
class ListSubSpaceResponseBodySpaceList extends $tea.Model {
    static names() {
        return {
            billingArea: 'billingArea',
            buildingArea: 'buildingArea',
            floor: 'floor',
            houseState: 'houseState',
            isVirtual: 'isVirtual',
            parentReferId: 'parentReferId',
            referId: 'referId',
            spaceName: 'spaceName',
            tagCode: 'tagCode',
            type: 'type',
        };
    }
    static types() {
        return {
            billingArea: 'number',
            buildingArea: 'number',
            floor: 'string',
            houseState: 'number',
            isVirtual: 'number',
            parentReferId: 'number',
            referId: 'number',
            spaceName: 'string',
            tagCode: 'string',
            type: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListSubSpaceResponseBodySpaceList = ListSubSpaceResponseBodySpaceList;
class ListUncheckUsersResponseBodyValues extends $tea.Model {
    static names() {
        return {
            deptId: 'deptId',
            extension: 'extension',
            gmtCreate: 'gmtCreate',
            gmtModified: 'gmtModified',
            isPropertyOwner: 'isPropertyOwner',
            name: 'name',
            status: 'status',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            deptId: 'number',
            extension: 'string',
            gmtCreate: 'number',
            gmtModified: 'number',
            isPropertyOwner: 'boolean',
            name: 'string',
            status: 'number',
            unionId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUncheckUsersResponseBodyValues = ListUncheckUsersResponseBodyValues;
class ListUserIndustryRolesResponseBodyRoleList extends $tea.Model {
    static names() {
        return {
            roleId: 'roleId',
            roleName: 'roleName',
            tagCode: 'tagCode',
        };
    }
    static types() {
        return {
            roleId: 'number',
            roleName: 'string',
            tagCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUserIndustryRolesResponseBodyRoleList = ListUserIndustryRolesResponseBodyRoleList;
class PagePointHistoryResponseBodyPointRecordList extends $tea.Model {
    static names() {
        return {
            createAt: 'createAt',
            ruleCode: 'ruleCode',
            ruleName: 'ruleName',
            score: 'score',
            userId: 'userId',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            createAt: 'number',
            ruleCode: 'string',
            ruleName: 'string',
            score: 'number',
            userId: 'string',
            uuid: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PagePointHistoryResponseBodyPointRecordList = PagePointHistoryResponseBodyPointRecordList;
class SearchResidentResponseBodyResidenceList extends $tea.Model {
    static names() {
        return {
            active: 'active',
            extField: 'extField',
            isPropertyOwner: 'isPropertyOwner',
            name: 'name',
            relateType: 'relateType',
        };
    }
    static types() {
        return {
            active: 'boolean',
            extField: 'string',
            isPropertyOwner: 'boolean',
            name: 'string',
            relateType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchResidentResponseBodyResidenceList = SearchResidentResponseBodyResidenceList;
class UpdateResidentMemberRequestResidentUpdateInfo extends $tea.Model {
    static names() {
        return {
            deptId: 'deptId',
            isPropertyOwner: 'isPropertyOwner',
            memberDeptExtension: 'memberDeptExtension',
            name: 'name',
            oldDeptId: 'oldDeptId',
            relateType: 'relateType',
            userId: 'userId',
        };
    }
    static types() {
        return {
            deptId: 'number',
            isPropertyOwner: 'boolean',
            memberDeptExtension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            name: 'string',
            oldDeptId: 'number',
            relateType: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentMemberRequestResidentUpdateInfo = UpdateResidentMemberRequestResidentUpdateInfo;
class UpdateResidentUserRequestExtField extends $tea.Model {
    static names() {
        return {
            itemName: 'itemName',
            itemValue: 'itemValue',
        };
    }
    static types() {
        return {
            itemName: 'string',
            itemValue: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateResidentUserRequestExtField = UpdateResidentUserRequestExtField;
class UpdateSpaceRequestSpaceInfoVOList extends $tea.Model {
    static names() {
        return {
            billingArea: 'billingArea',
            buildingArea: 'buildingArea',
            buildingType: 'buildingType',
            deptId: 'deptId',
            floor: 'floor',
            houseState: 'houseState',
            houseType: 'houseType',
            name: 'name',
            parentDeptId: 'parentDeptId',
            tagCode: 'tagCode',
        };
    }
    static types() {
        return {
            billingArea: 'number',
            buildingArea: 'number',
            buildingType: 'number',
            deptId: 'number',
            floor: 'string',
            houseState: 'number',
            houseType: 'number',
            name: 'string',
            parentDeptId: 'number',
            tagCode: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateSpaceRequestSpaceInfoVOList = UpdateSpaceRequestSpaceInfoVOList;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 
     *
     * @param request AddPointRequest
     * @param headers AddPointHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddPointResponse
     */
    async addPointWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.actionTime)) {
            query["actionTime"] = request.actionTime;
        }
        if (!tea_util_1.default.isUnset(request.isCircle)) {
            query["isCircle"] = request.isCircle;
        }
        if (!tea_util_1.default.isUnset(request.ruleCode)) {
            query["ruleCode"] = request.ruleCode;
        }
        if (!tea_util_1.default.isUnset(request.ruleName)) {
            query["ruleName"] = request.ruleName;
        }
        if (!tea_util_1.default.isUnset(request.score)) {
            query["score"] = request.score;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        if (!tea_util_1.default.isUnset(request.uuid)) {
            query["uuid"] = request.uuid;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "AddPoint",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/points`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddPointResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddPointRequest
     * @return AddPointResponse
     */
    async addPoint(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddPointHeaders({});
        return await this.addPointWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request AddResidentDepartmentRequest
     * @param headers AddResidentDepartmentHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddResidentDepartmentResponse
     */
    async addResidentDepartmentWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.departmentName)) {
            query["departmentName"] = request.departmentName;
        }
        if (!tea_util_1.default.isUnset(request.isResidenceGroup)) {
            query["isResidenceGroup"] = request.isResidenceGroup;
        }
        if (!tea_util_1.default.isUnset(request.parentDepartmentId)) {
            query["parentDepartmentId"] = request.parentDepartmentId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "AddResidentDepartment",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/departments`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddResidentDepartmentResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddResidentDepartmentRequest
     * @return AddResidentDepartmentResponse
     */
    async addResidentDepartment(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddResidentDepartmentHeaders({});
        return await this.addResidentDepartmentWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request AddResidentMemberRequest
     * @param headers AddResidentMemberHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddResidentMemberResponse
     */
    async addResidentMemberWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.residentAddInfo)) {
            body["residentAddInfo"] = request.residentAddInfo;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddResidentMember",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/members`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddResidentMemberResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddResidentMemberRequest
     * @return AddResidentMemberResponse
     */
    async addResidentMember(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddResidentMemberHeaders({});
        return await this.addResidentMemberWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request AddResidentUsersRequest
     * @param headers AddResidentUsersHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddResidentUsersResponse
     */
    async addResidentUsersWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.address)) {
            query["address"] = request.address;
        }
        if (!tea_util_1.default.isUnset(request.departmentId)) {
            query["departmentId"] = request.departmentId;
        }
        if (!tea_util_1.default.isUnset(request.extField)) {
            query["extField"] = request.extField;
        }
        if (!tea_util_1.default.isUnset(request.isLeaseholder)) {
            query["isLeaseholder"] = request.isLeaseholder;
        }
        if (!tea_util_1.default.isUnset(request.mobile)) {
            query["mobile"] = request.mobile;
        }
        if (!tea_util_1.default.isUnset(request.relateType)) {
            query["relateType"] = request.relateType;
        }
        if (!tea_util_1.default.isUnset(request.userName)) {
            query["userName"] = request.userName;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "AddResidentUsers",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/users`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddResidentUsersResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddResidentUsersRequest
     * @return AddResidentUsersResponse
     */
    async addResidentUsers(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddResidentUsersHeaders({});
        return await this.addResidentUsersWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CreateResidentBlackBoardRequest
     * @param headers CreateResidentBlackBoardHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateResidentBlackBoardResponse
     */
    async createResidentBlackBoardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.context)) {
            body["context"] = request.context;
        }
        if (!tea_util_1.default.isUnset(request.mediaId)) {
            body["mediaId"] = request.mediaId;
        }
        if (!tea_util_1.default.isUnset(request.sendTime)) {
            body["sendTime"] = request.sendTime;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateResidentBlackBoard",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/blackboards`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateResidentBlackBoardResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CreateResidentBlackBoardRequest
     * @return CreateResidentBlackBoardResponse
     */
    async createResidentBlackBoard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateResidentBlackBoardHeaders({});
        return await this.createResidentBlackBoardWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CreateSpaceRequest
     * @param headers CreateSpaceHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateSpaceResponse
     */
    async createSpaceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.billingArea)) {
            body["billingArea"] = request.billingArea;
        }
        if (!tea_util_1.default.isUnset(request.buildingArea)) {
            body["buildingArea"] = request.buildingArea;
        }
        if (!tea_util_1.default.isUnset(request.floor)) {
            body["floor"] = request.floor;
        }
        if (!tea_util_1.default.isUnset(request.houseState)) {
            body["houseState"] = request.houseState;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.parentDeptId)) {
            body["parentDeptId"] = request.parentDeptId;
        }
        if (!tea_util_1.default.isUnset(request.tagCode)) {
            body["tagCode"] = request.tagCode;
        }
        if (!tea_util_1.default.isUnset(request.type)) {
            body["type"] = request.type;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateSpace",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/spaces`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateSpaceResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CreateSpaceRequest
     * @return CreateSpaceResponse
     */
    async createSpace(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateSpaceHeaders({});
        return await this.createSpaceWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request DeleteResidentBlackBoardRequest
     * @param headers DeleteResidentBlackBoardHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeleteResidentBlackBoardResponse
     */
    async deleteResidentBlackBoardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.blackboardId)) {
            query["blackboardId"] = request.blackboardId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DeleteResidentBlackBoard",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/blackboards`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteResidentBlackBoardResponse({}));
    }
    /**
     * @summary 
     *
     * @param request DeleteResidentBlackBoardRequest
     * @return DeleteResidentBlackBoardResponse
     */
    async deleteResidentBlackBoard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteResidentBlackBoardHeaders({});
        return await this.deleteResidentBlackBoardWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request DeleteResidentDepartmentRequest
     * @param headers DeleteResidentDepartmentHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeleteResidentDepartmentResponse
     */
    async deleteResidentDepartmentWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.departmentId)) {
            query["departmentId"] = request.departmentId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DeleteResidentDepartment",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/departments`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteResidentDepartmentResponse({}));
    }
    /**
     * @summary 
     *
     * @param request DeleteResidentDepartmentRequest
     * @return DeleteResidentDepartmentResponse
     */
    async deleteResidentDepartment(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteResidentDepartmentHeaders({});
        return await this.deleteResidentDepartmentWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request DeleteSpaceRequest
     * @param headers DeleteSpaceHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeleteSpaceResponse
     */
    async deleteSpaceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.deptIds)) {
            body["deptIds"] = request.deptIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "DeleteSpace",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/spaces/remove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteSpaceResponse({}));
    }
    /**
     * @summary 
     *
     * @param request DeleteSpaceRequest
     * @return DeleteSpaceResponse
     */
    async deleteSpace(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteSpaceHeaders({});
        return await this.deleteSpaceWithOptions(request, headers, runtime);
    }
    /**
     * @summary openConversationId
     *
     * @param request GetConversationIdRequest
     * @param headers GetConversationIdHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetConversationIdResponse
     */
    async getConversationIdWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.chatId)) {
            query["chatId"] = request.chatId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetConversationId",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/conversations`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetConversationIdResponse({}));
    }
    /**
     * @summary openConversationId
     *
     * @param request GetConversationIdRequest
     * @return GetConversationIdResponse
     */
    async getConversationId(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetConversationIdHeaders({});
        return await this.getConversationIdWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers GetIndustryTypeHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetIndustryTypeResponse
     */
    async getIndustryTypeWithOptions(headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "GetIndustryType",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/organizations/industryTypes`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetIndustryTypeResponse({}));
    }
    /**
     * @summary 
     *
     * @return GetIndustryTypeResponse
     */
    async getIndustryType() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetIndustryTypeHeaders({});
        return await this.getIndustryTypeWithOptions(headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetPropertyInfoRequest
     * @param headers GetPropertyInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetPropertyInfoResponse
     */
    async getPropertyInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.propertyCorpId)) {
            query["propertyCorpId"] = request.propertyCorpId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetPropertyInfo",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/propertyInfos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetPropertyInfoResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetPropertyInfoRequest
     * @return GetPropertyInfoResponse
     */
    async getPropertyInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetPropertyInfoHeaders({});
        return await this.getPropertyInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetResidentInfoRequest
     * @param headers GetResidentInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetResidentInfoResponse
     */
    async getResidentInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.residentCorpId)) {
            query["residentCorpId"] = request.residentCorpId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetResidentInfo",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/residentInfos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetResidentInfoResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetResidentInfoRequest
     * @return GetResidentInfoResponse
     */
    async getResidentInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetResidentInfoHeaders({});
        return await this.getResidentInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetResidentMembersInfoRequest
     * @param headers GetResidentMembersInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetResidentMembersInfoResponse
     */
    async getResidentMembersInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.residentCropId)) {
            body["residentCropId"] = request.residentCropId;
        }
        if (!tea_util_1.default.isUnset(request.userIdList)) {
            body["userIdList"] = request.userIdList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "GetResidentMembersInfo",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/residences/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetResidentMembersInfoResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetResidentMembersInfoRequest
     * @return GetResidentMembersInfoResponse
     */
    async getResidentMembersInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetResidentMembersInfoHeaders({});
        return await this.getResidentMembersInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary id
     *
     * @param request GetSpaceIdByTypeRequest
     * @param headers GetSpaceIdByTypeHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetSpaceIdByTypeResponse
     */
    async getSpaceIdByTypeWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.departmentType)) {
            query["departmentType"] = request.departmentType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetSpaceIdByType",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/spaces/types`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetSpaceIdByTypeResponse({}));
    }
    /**
     * @summary id
     *
     * @param request GetSpaceIdByTypeRequest
     * @return GetSpaceIdByTypeResponse
     */
    async getSpaceIdByType(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetSpaceIdByTypeHeaders({});
        return await this.getSpaceIdByTypeWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetSpacesInfoRequest
     * @param headers GetSpacesInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetSpacesInfoResponse
     */
    async getSpacesInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.referIds)) {
            body["referIds"] = request.referIds;
        }
        if (!tea_util_1.default.isUnset(request.residentCorpId)) {
            body["residentCorpId"] = request.residentCorpId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "GetSpacesInfo",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/spaces/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetSpacesInfoResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetSpacesInfoRequest
     * @return GetSpacesInfoResponse
     */
    async getSpacesInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetSpacesInfoHeaders({});
        return await this.getSpacesInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request ListIndustryRoleUsersRequest
     * @param headers ListIndustryRoleUsersHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListIndustryRoleUsersResponse
     */
    async listIndustryRoleUsersWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.tagCode)) {
            query["tagCode"] = request.tagCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListIndustryRoleUsers",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/industryRoles/users`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListIndustryRoleUsersResponse({}));
    }
    /**
     * @summary 
     *
     * @param request ListIndustryRoleUsersRequest
     * @return ListIndustryRoleUsersResponse
     */
    async listIndustryRoleUsers(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListIndustryRoleUsersHeaders({});
        return await this.listIndustryRoleUsersWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request ListPointRulesRequest
     * @param headers ListPointRulesHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListPointRulesResponse
     */
    async listPointRulesWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.isCircle)) {
            query["isCircle"] = request.isCircle;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListPointRules",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/points/rules`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListPointRulesResponse({}));
    }
    /**
     * @summary 
     *
     * @param request ListPointRulesRequest
     * @return ListPointRulesResponse
     */
    async listPointRules(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListPointRulesHeaders({});
        return await this.listPointRulesWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request ListSubSpaceRequest
     * @param headers ListSubSpaceHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListSubSpaceResponse
     */
    async listSubSpaceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.referId)) {
            query["referId"] = request.referId;
        }
        if (!tea_util_1.default.isUnset(request.residentCorpId)) {
            query["residentCorpId"] = request.residentCorpId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListSubSpace",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/spaces/subSpaces`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListSubSpaceResponse({}));
    }
    /**
     * @summary 
     *
     * @param request ListSubSpaceRequest
     * @return ListSubSpaceResponse
     */
    async listSubSpace(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListSubSpaceHeaders({});
        return await this.listSubSpaceWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request ListUncheckUsersRequest
     * @param headers ListUncheckUsersHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListUncheckUsersResponse
     */
    async listUncheckUsersWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            query["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.status)) {
            query["status"] = request.status;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListUncheckUsers",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/organizations/noJoinUsers`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListUncheckUsersResponse({}));
    }
    /**
     * @summary 
     *
     * @param request ListUncheckUsersRequest
     * @return ListUncheckUsersResponse
     */
    async listUncheckUsers(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListUncheckUsersHeaders({});
        return await this.listUncheckUsersWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request ListUserIndustryRolesRequest
     * @param headers ListUserIndustryRolesHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListUserIndustryRolesResponse
     */
    async listUserIndustryRolesWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListUserIndustryRoles",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/users/industryRoles`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListUserIndustryRolesResponse({}));
    }
    /**
     * @summary 
     *
     * @param request ListUserIndustryRolesRequest
     * @return ListUserIndustryRolesResponse
     */
    async listUserIndustryRoles(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListUserIndustryRolesHeaders({});
        return await this.listUserIndustryRolesWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request PagePointHistoryRequest
     * @param headers PagePointHistoryHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return PagePointHistoryResponse
     */
    async pagePointHistoryWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.endTime)) {
            query["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.isCircle)) {
            query["isCircle"] = request.isCircle;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            query["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "PagePointHistory",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/points/records`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PagePointHistoryResponse({}));
    }
    /**
     * @summary 
     *
     * @param request PagePointHistoryRequest
     * @return PagePointHistoryResponse
     */
    async pagePointHistory(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PagePointHistoryHeaders({});
        return await this.pagePointHistoryWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request RemoveResidentMemberRequest
     * @param headers RemoveResidentMemberHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RemoveResidentMemberResponse
     */
    async removeResidentMemberWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.deptId)) {
            body["deptId"] = request.deptId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "RemoveResidentMember",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/members/remove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RemoveResidentMemberResponse({}));
    }
    /**
     * @summary 
     *
     * @param request RemoveResidentMemberRequest
     * @return RemoveResidentMemberResponse
     */
    async removeResidentMember(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RemoveResidentMemberHeaders({});
        return await this.removeResidentMemberWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request RemoveResidentUserRequest
     * @param headers RemoveResidentUserHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RemoveResidentUserResponse
     */
    async removeResidentUserWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.departmentId)) {
            query["departmentId"] = request.departmentId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "RemoveResidentUser",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/users/remove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RemoveResidentUserResponse({}));
    }
    /**
     * @summary 
     *
     * @param request RemoveResidentUserRequest
     * @return RemoveResidentUserResponse
     */
    async removeResidentUser(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RemoveResidentUserHeaders({});
        return await this.removeResidentUserWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request SearchResidentRequest
     * @param headers SearchResidentHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SearchResidentResponse
     */
    async searchResidentWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.residentCropId)) {
            query["residentCropId"] = request.residentCropId;
        }
        if (!tea_util_1.default.isUnset(request.searchWord)) {
            query["searchWord"] = request.searchWord;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "SearchResident",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/residences`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SearchResidentResponse({}));
    }
    /**
     * @summary 
     *
     * @param request SearchResidentRequest
     * @return SearchResidentResponse
     */
    async searchResident(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SearchResidentHeaders({});
        return await this.searchResidentWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateResideceGroupRequest
     * @param headers UpdateResideceGroupHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateResideceGroupResponse
     */
    async updateResideceGroupWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.departmentId)) {
            query["departmentId"] = request.departmentId;
        }
        if (!tea_util_1.default.isUnset(request.departmentName)) {
            query["departmentName"] = request.departmentName;
        }
        if (!tea_util_1.default.isUnset(request.managerUserId)) {
            query["managerUserId"] = request.managerUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "UpdateResideceGroup",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/departments/updateResideceGroup`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateResideceGroupResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateResideceGroupRequest
     * @return UpdateResideceGroupResponse
     */
    async updateResideceGroup(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateResideceGroupHeaders({});
        return await this.updateResideceGroupWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateResidenceRequest
     * @param headers UpdateResidenceHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateResidenceResponse
     */
    async updateResidenceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.departmentId)) {
            query["departmentId"] = request.departmentId;
        }
        if (!tea_util_1.default.isUnset(request.departmentName)) {
            query["departmentName"] = request.departmentName;
        }
        if (!tea_util_1.default.isUnset(request.destitute)) {
            query["destitute"] = request.destitute;
        }
        if (!tea_util_1.default.isUnset(request.grid)) {
            query["grid"] = request.grid;
        }
        if (!tea_util_1.default.isUnset(request.homeTel)) {
            query["homeTel"] = request.homeTel;
        }
        if (!tea_util_1.default.isUnset(request.managerUserId)) {
            query["managerUserId"] = request.managerUserId;
        }
        if (!tea_util_1.default.isUnset(request.parentDepartmentId)) {
            query["parentDepartmentId"] = request.parentDepartmentId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "UpdateResidence",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/departments/updateResidece`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateResidenceResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateResidenceRequest
     * @return UpdateResidenceResponse
     */
    async updateResidence(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateResidenceHeaders({});
        return await this.updateResidenceWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateResidentBlackBoardRequest
     * @param headers UpdateResidentBlackBoardHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateResidentBlackBoardResponse
     */
    async updateResidentBlackBoardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.blackboardId)) {
            body["blackboardId"] = request.blackboardId;
        }
        if (!tea_util_1.default.isUnset(request.context)) {
            body["context"] = request.context;
        }
        if (!tea_util_1.default.isUnset(request.mediaId)) {
            body["mediaId"] = request.mediaId;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateResidentBlackBoard",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/blackboards`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateResidentBlackBoardResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateResidentBlackBoardRequest
     * @return UpdateResidentBlackBoardResponse
     */
    async updateResidentBlackBoard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateResidentBlackBoardHeaders({});
        return await this.updateResidentBlackBoardWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateResidentInfoRequest
     * @param headers UpdateResidentInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateResidentInfoResponse
     */
    async updateResidentInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.address)) {
            body["address"] = request.address;
        }
        if (!tea_util_1.default.isUnset(request.buildingArea)) {
            body["buildingArea"] = request.buildingArea;
        }
        if (!tea_util_1.default.isUnset(request.cityName)) {
            body["cityName"] = request.cityName;
        }
        if (!tea_util_1.default.isUnset(request.communityType)) {
            body["communityType"] = request.communityType;
        }
        if (!tea_util_1.default.isUnset(request.countyName)) {
            body["countyName"] = request.countyName;
        }
        if (!tea_util_1.default.isUnset(request.location)) {
            body["location"] = request.location;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.provName)) {
            body["provName"] = request.provName;
        }
        if (!tea_util_1.default.isUnset(request.state)) {
            body["state"] = request.state;
        }
        if (!tea_util_1.default.isUnset(request.telephone)) {
            body["telephone"] = request.telephone;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateResidentInfo",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/residences`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateResidentInfoResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateResidentInfoRequest
     * @return UpdateResidentInfoResponse
     */
    async updateResidentInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateResidentInfoHeaders({});
        return await this.updateResidentInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateResidentMemberRequest
     * @param headers UpdateResidentMemberHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateResidentMemberResponse
     */
    async updateResidentMemberWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.residentUpdateInfo)) {
            body["residentUpdateInfo"] = request.residentUpdateInfo;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateResidentMember",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/members`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateResidentMemberResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateResidentMemberRequest
     * @return UpdateResidentMemberResponse
     */
    async updateResidentMember(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateResidentMemberHeaders({});
        return await this.updateResidentMemberWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateResidentUserRequest
     * @param headers UpdateResidentUserHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateResidentUserResponse
     */
    async updateResidentUserWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.address)) {
            query["address"] = request.address;
        }
        if (!tea_util_1.default.isUnset(request.departmentId)) {
            query["departmentId"] = request.departmentId;
        }
        if (!tea_util_1.default.isUnset(request.extField)) {
            query["extField"] = request.extField;
        }
        if (!tea_util_1.default.isUnset(request.isRetainOldDept)) {
            query["isRetainOldDept"] = request.isRetainOldDept;
        }
        if (!tea_util_1.default.isUnset(request.mobile)) {
            query["mobile"] = request.mobile;
        }
        if (!tea_util_1.default.isUnset(request.oldDepartmentId)) {
            query["oldDepartmentId"] = request.oldDepartmentId;
        }
        if (!tea_util_1.default.isUnset(request.relateType)) {
            query["relateType"] = request.relateType;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        if (!tea_util_1.default.isUnset(request.userName)) {
            query["userName"] = request.userName;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "UpdateResidentUser",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/users`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateResidentUserResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateResidentUserRequest
     * @return UpdateResidentUserResponse
     */
    async updateResidentUser(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateResidentUserHeaders({});
        return await this.updateResidentUserWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateSpaceRequest
     * @param headers UpdateSpaceHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateSpaceResponse
     */
    async updateSpaceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.spaceInfoVOList)) {
            body["spaceInfoVOList"] = request.spaceInfoVOList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateSpace",
            version: "resident_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/resident/spaces`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateSpaceResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateSpaceRequest
     * @return UpdateSpaceResponse
     */
    async updateSpace(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateSpaceHeaders({});
        return await this.updateSpaceWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
