// @alicloud/dingtalk@2.1.16/dist/contract_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendContractCardRequestSender = exports.SendContractCardRequestReceivers = exports.SendContractCardRequestContractInfo = exports.QueryAdvancedContractVersionResponseBodyResult = exports.EsignSyncEventResponseBodyResult = exports.EsignQueryIdentityByTicketResponseBodyResult = exports.EsignQueryGrantInfoResponseBodyResult = exports.EsignQueryApprovalInfoResponseBodyResult = exports.ContractBenefitConsumeResponseBodyResult = exports.SendContractCardResponse = exports.SendContractCardResponseBody = exports.SendContractCardRequest = exports.SendContractCardHeaders = exports.QueryAdvancedContractVersionResponse = exports.QueryAdvancedContractVersionResponseBody = exports.QueryAdvancedContractVersionRequest = exports.QueryAdvancedContractVersionHeaders = exports.EsignSyncEventResponse = exports.EsignSyncEventResponseBody = exports.EsignSyncEventRequest = exports.EsignSyncEventHeaders = exports.EsignQueryIdentityByTicketResponse = exports.EsignQueryIdentityByTicketResponseBody = exports.EsignQueryIdentityByTicketRequest = exports.EsignQueryIdentityByTicketHeaders = exports.EsignQueryGrantInfoResponse = exports.EsignQueryGrantInfoResponseBody = exports.EsignQueryGrantInfoRequest = exports.EsignQueryGrantInfoHeaders = exports.EsignQueryApprovalInfoResponse = exports.EsignQueryApprovalInfoResponseBody = exports.EsignQueryApprovalInfoRequest = exports.EsignQueryApprovalInfoHeaders = exports.ContractBenefitConsumeResponse = exports.ContractBenefitConsumeResponseBody = exports.ContractBenefitConsumeRequest = exports.ContractBenefitConsumeHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class ContractBenefitConsumeHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ContractBenefitConsumeHeaders = ContractBenefitConsumeHeaders;
class ContractBenefitConsumeRequest extends $tea.Model {
    static names() {
        return {
            benefitPoint: 'benefitPoint',
            bizRequestId: 'bizRequestId',
            consumeQuota: 'consumeQuota',
            corpId: 'corpId',
            extParams: 'extParams',
            isvCorpId: 'isvCorpId',
            optUnionId: 'optUnionId',
        };
    }
    static types() {
        return {
            benefitPoint: 'string',
            bizRequestId: 'string',
            consumeQuota: 'number',
            corpId: 'string',
            extParams: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            isvCorpId: 'string',
            optUnionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ContractBenefitConsumeRequest = ContractBenefitConsumeRequest;
class ContractBenefitConsumeResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: ContractBenefitConsumeResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ContractBenefitConsumeResponseBody = ContractBenefitConsumeResponseBody;
class ContractBenefitConsumeResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ContractBenefitConsumeResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ContractBenefitConsumeResponse = ContractBenefitConsumeResponse;
class EsignQueryApprovalInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryApprovalInfoHeaders = EsignQueryApprovalInfoHeaders;
class EsignQueryApprovalInfoRequest extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            esignFlowId: 'esignFlowId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            corpId: 'string',
            esignFlowId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryApprovalInfoRequest = EsignQueryApprovalInfoRequest;
class EsignQueryApprovalInfoResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: EsignQueryApprovalInfoResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryApprovalInfoResponseBody = EsignQueryApprovalInfoResponseBody;
class EsignQueryApprovalInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: EsignQueryApprovalInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryApprovalInfoResponse = EsignQueryApprovalInfoResponse;
class EsignQueryGrantInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryGrantInfoHeaders = EsignQueryGrantInfoHeaders;
class EsignQueryGrantInfoRequest extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            extension: 'extension',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            corpId: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryGrantInfoRequest = EsignQueryGrantInfoRequest;
class EsignQueryGrantInfoResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: EsignQueryGrantInfoResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryGrantInfoResponseBody = EsignQueryGrantInfoResponseBody;
class EsignQueryGrantInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: EsignQueryGrantInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryGrantInfoResponse = EsignQueryGrantInfoResponse;
class EsignQueryIdentityByTicketHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryIdentityByTicketHeaders = EsignQueryIdentityByTicketHeaders;
class EsignQueryIdentityByTicketRequest extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            extension: 'extension',
            ticket: 'ticket',
        };
    }
    static types() {
        return {
            corpId: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            ticket: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryIdentityByTicketRequest = EsignQueryIdentityByTicketRequest;
class EsignQueryIdentityByTicketResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: EsignQueryIdentityByTicketResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryIdentityByTicketResponseBody = EsignQueryIdentityByTicketResponseBody;
class EsignQueryIdentityByTicketResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: EsignQueryIdentityByTicketResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryIdentityByTicketResponse = EsignQueryIdentityByTicketResponse;
class EsignSyncEventHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignSyncEventHeaders = EsignSyncEventHeaders;
class EsignSyncEventRequest extends $tea.Model {
    static names() {
        return {
            action: 'action',
            corpId: 'corpId',
            esignData: 'esignData',
            extension: 'extension',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            action: 'string',
            corpId: 'string',
            esignData: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignSyncEventRequest = EsignSyncEventRequest;
class EsignSyncEventResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: EsignSyncEventResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignSyncEventResponseBody = EsignSyncEventResponseBody;
class EsignSyncEventResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: EsignSyncEventResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignSyncEventResponse = EsignSyncEventResponse;
class QueryAdvancedContractVersionHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryAdvancedContractVersionHeaders = QueryAdvancedContractVersionHeaders;
class QueryAdvancedContractVersionRequest extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            extension: 'extension',
        };
    }
    static types() {
        return {
            corpId: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryAdvancedContractVersionRequest = QueryAdvancedContractVersionRequest;
class QueryAdvancedContractVersionResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: QueryAdvancedContractVersionResponseBodyResult,
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryAdvancedContractVersionResponseBody = QueryAdvancedContractVersionResponseBody;
class QueryAdvancedContractVersionResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryAdvancedContractVersionResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryAdvancedContractVersionResponse = QueryAdvancedContractVersionResponse;
class SendContractCardHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendContractCardHeaders = SendContractCardHeaders;
class SendContractCardRequest extends $tea.Model {
    static names() {
        return {
            cardType: 'cardType',
            contractInfo: 'contractInfo',
            corpId: 'corpId',
            extension: 'extension',
            processInstanceId: 'processInstanceId',
            receiveGroups: 'receiveGroups',
            receivers: 'receivers',
            sender: 'sender',
            syncSingleChat: 'syncSingleChat',
        };
    }
    static types() {
        return {
            cardType: 'string',
            contractInfo: SendContractCardRequestContractInfo,
            corpId: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            processInstanceId: 'string',
            receiveGroups: { 'type': 'array', 'itemType': 'string' },
            receivers: { 'type': 'array', 'itemType': SendContractCardRequestReceivers },
            sender: SendContractCardRequestSender,
            syncSingleChat: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendContractCardRequest = SendContractCardRequest;
class SendContractCardResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendContractCardResponseBody = SendContractCardResponseBody;
class SendContractCardResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendContractCardResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendContractCardResponse = SendContractCardResponse;
class ContractBenefitConsumeResponseBodyResult extends $tea.Model {
    static names() {
        return {
            consumeResult: 'consumeResult',
        };
    }
    static types() {
        return {
            consumeResult: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ContractBenefitConsumeResponseBodyResult = ContractBenefitConsumeResponseBodyResult;
class EsignQueryApprovalInfoResponseBodyResult extends $tea.Model {
    static names() {
        return {
            bpmsProcessBusinessId: 'bpmsProcessBusinessId',
            bpmsProcessInstanceId: 'bpmsProcessInstanceId',
            bpmsProcessInstanceUrl: 'bpmsProcessInstanceUrl',
        };
    }
    static types() {
        return {
            bpmsProcessBusinessId: 'string',
            bpmsProcessInstanceId: 'string',
            bpmsProcessInstanceUrl: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryApprovalInfoResponseBodyResult = EsignQueryApprovalInfoResponseBodyResult;
class EsignQueryGrantInfoResponseBodyResult extends $tea.Model {
    static names() {
        return {
            legalPerson: 'legalPerson',
            mobilePhoneNumber: 'mobilePhoneNumber',
            orgName: 'orgName',
            stateCode: 'stateCode',
            unifiedSocialCredit: 'unifiedSocialCredit',
            userName: 'userName',
        };
    }
    static types() {
        return {
            legalPerson: 'string',
            mobilePhoneNumber: 'string',
            orgName: 'string',
            stateCode: 'string',
            unifiedSocialCredit: 'string',
            userName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryGrantInfoResponseBodyResult = EsignQueryGrantInfoResponseBodyResult;
class EsignQueryIdentityByTicketResponseBodyResult extends $tea.Model {
    static names() {
        return {
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignQueryIdentityByTicketResponseBodyResult = EsignQueryIdentityByTicketResponseBodyResult;
class EsignSyncEventResponseBodyResult extends $tea.Model {
    static names() {
        return {
            message: 'message',
        };
    }
    static types() {
        return {
            message: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EsignSyncEventResponseBodyResult = EsignSyncEventResponseBodyResult;
class QueryAdvancedContractVersionResponseBodyResult extends $tea.Model {
    static names() {
        return {
            enableEsignAttachmentSign: 'enableEsignAttachmentSign',
            extension: 'extension',
            version: 'version',
        };
    }
    static types() {
        return {
            enableEsignAttachmentSign: 'boolean',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            version: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryAdvancedContractVersionResponseBodyResult = QueryAdvancedContractVersionResponseBodyResult;
class SendContractCardRequestContractInfo extends $tea.Model {
    static names() {
        return {
            contractCode: 'contractCode',
            contractName: 'contractName',
            createTime: 'createTime',
            signUserName: 'signUserName',
        };
    }
    static types() {
        return {
            contractCode: 'string',
            contractName: 'string',
            createTime: 'number',
            signUserName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendContractCardRequestContractInfo = SendContractCardRequestContractInfo;
class SendContractCardRequestReceivers extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            userId: 'userId',
            userType: 'userType',
        };
    }
    static types() {
        return {
            corpId: 'string',
            userId: 'string',
            userType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendContractCardRequestReceivers = SendContractCardRequestReceivers;
class SendContractCardRequestSender extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            userId: 'userId',
            userType: 'userType',
        };
    }
    static types() {
        return {
            corpId: 'string',
            userId: 'string',
            userType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendContractCardRequestSender = SendContractCardRequestSender;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 合同权益核销
     *
     * @param request ContractBenefitConsumeRequest
     * @param headers ContractBenefitConsumeHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ContractBenefitConsumeResponse
     */
    async contractBenefitConsumeWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.benefitPoint)) {
            body["benefitPoint"] = request.benefitPoint;
        }
        if (!tea_util_1.default.isUnset(request.bizRequestId)) {
            body["bizRequestId"] = request.bizRequestId;
        }
        if (!tea_util_1.default.isUnset(request.consumeQuota)) {
            body["consumeQuota"] = request.consumeQuota;
        }
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.extParams)) {
            body["extParams"] = request.extParams;
        }
        if (!tea_util_1.default.isUnset(request.isvCorpId)) {
            body["isvCorpId"] = request.isvCorpId;
        }
        if (!tea_util_1.default.isUnset(request.optUnionId)) {
            body["optUnionId"] = request.optUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ContractBenefitConsume",
            version: "contract_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/contract/benefits/consume`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ContractBenefitConsumeResponse({}));
    }
    /**
     * @summary 合同权益核销
     *
     * @param request ContractBenefitConsumeRequest
     * @return ContractBenefitConsumeResponse
     */
    async contractBenefitConsume(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ContractBenefitConsumeHeaders({});
        return await this.contractBenefitConsumeWithOptions(request, headers, runtime);
    }
    /**
     * @summary 天谷侧查询审批单
     *
     * @param request EsignQueryApprovalInfoRequest
     * @param headers EsignQueryApprovalInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return EsignQueryApprovalInfoResponse
     */
    async esignQueryApprovalInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.esignFlowId)) {
            body["esignFlowId"] = request.esignFlowId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "EsignQueryApprovalInfo",
            version: "contract_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/contract/esign/approvalInfos/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new EsignQueryApprovalInfoResponse({}));
    }
    /**
     * @summary 天谷侧查询审批单
     *
     * @param request EsignQueryApprovalInfoRequest
     * @return EsignQueryApprovalInfoResponse
     */
    async esignQueryApprovalInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new EsignQueryApprovalInfoHeaders({});
        return await this.esignQueryApprovalInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 天谷侧查询授权信息接口
     *
     * @param request EsignQueryGrantInfoRequest
     * @param headers EsignQueryGrantInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return EsignQueryGrantInfoResponse
     */
    async esignQueryGrantInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.extension)) {
            body["extension"] = request.extension;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "EsignQueryGrantInfo",
            version: "contract_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/contract/esign/anthInfos/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new EsignQueryGrantInfoResponse({}));
    }
    /**
     * @summary 天谷侧查询授权信息接口
     *
     * @param request EsignQueryGrantInfoRequest
     * @return EsignQueryGrantInfoResponse
     */
    async esignQueryGrantInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new EsignQueryGrantInfoHeaders({});
        return await this.esignQueryGrantInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 天谷侧查询获取免登信息
     *
     * @param request EsignQueryIdentityByTicketRequest
     * @param headers EsignQueryIdentityByTicketHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return EsignQueryIdentityByTicketResponse
     */
    async esignQueryIdentityByTicketWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.extension)) {
            body["extension"] = request.extension;
        }
        if (!tea_util_1.default.isUnset(request.ticket)) {
            body["ticket"] = request.ticket;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "EsignQueryIdentityByTicket",
            version: "contract_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/contract/esign/tickets/identities/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new EsignQueryIdentityByTicketResponse({}));
    }
    /**
     * @summary 天谷侧查询获取免登信息
     *
     * @param request EsignQueryIdentityByTicketRequest
     * @return EsignQueryIdentityByTicketResponse
     */
    async esignQueryIdentityByTicket(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new EsignQueryIdentityByTicketHeaders({});
        return await this.esignQueryIdentityByTicketWithOptions(request, headers, runtime);
    }
    /**
     * @summary e签宝电子签事件同步回传接口
     *
     * @param request EsignSyncEventRequest
     * @param headers EsignSyncEventHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return EsignSyncEventResponse
     */
    async esignSyncEventWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.action)) {
            body["action"] = request.action;
        }
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.esignData)) {
            body["esignData"] = request.esignData;
        }
        if (!tea_util_1.default.isUnset(request.extension)) {
            body["extension"] = request.extension;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "EsignSyncEvent",
            version: "contract_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/contract/esign/events/sync`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new EsignSyncEventResponse({}));
    }
    /**
     * @summary e签宝电子签事件同步回传接口
     *
     * @param request EsignSyncEventRequest
     * @return EsignSyncEventResponse
     */
    async esignSyncEvent(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new EsignSyncEventHeaders({});
        return await this.esignSyncEventWithOptions(request, headers, runtime);
    }
    /**
     * @summary e签宝查询智能合同版本接口
     *
     * @param request QueryAdvancedContractVersionRequest
     * @param headers QueryAdvancedContractVersionHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryAdvancedContractVersionResponse
     */
    async queryAdvancedContractVersionWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.extension)) {
            body["extension"] = request.extension;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "QueryAdvancedContractVersion",
            version: "contract_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/contract/versions/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryAdvancedContractVersionResponse({}));
    }
    /**
     * @summary e签宝查询智能合同版本接口
     *
     * @param request QueryAdvancedContractVersionRequest
     * @return QueryAdvancedContractVersionResponse
     */
    async queryAdvancedContractVersion(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryAdvancedContractVersionHeaders({});
        return await this.queryAdvancedContractVersionWithOptions(request, headers, runtime);
    }
    /**
     * @summary 发送合同相关卡片
     *
     * @param request SendContractCardRequest
     * @param headers SendContractCardHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SendContractCardResponse
     */
    async sendContractCardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.cardType)) {
            body["cardType"] = request.cardType;
        }
        if (!tea_util_1.default.isUnset(request.contractInfo)) {
            body["contractInfo"] = request.contractInfo;
        }
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.extension)) {
            body["extension"] = request.extension;
        }
        if (!tea_util_1.default.isUnset(request.processInstanceId)) {
            body["processInstanceId"] = request.processInstanceId;
        }
        if (!tea_util_1.default.isUnset(request.receiveGroups)) {
            body["receiveGroups"] = request.receiveGroups;
        }
        if (!tea_util_1.default.isUnset(request.receivers)) {
            body["receivers"] = request.receivers;
        }
        if (!tea_util_1.default.isUnset(request.sender)) {
            body["sender"] = request.sender;
        }
        if (!tea_util_1.default.isUnset(request.syncSingleChat)) {
            body["syncSingleChat"] = request.syncSingleChat;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendContractCard",
            version: "contract_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/contract/cards/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendContractCardResponse({}));
    }
    /**
     * @summary 发送合同相关卡片
     *
     * @param request SendContractCardRequest
     * @return SendContractCardResponse
     */
    async sendContractCard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendContractCardHeaders({});
        return await this.sendContractCardWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
