// @alicloud/dingtalk@2.1.16/dist/yun_shu_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SaveOpenExternalLogResponse = exports.SaveOpenExternalLogResponseBody = exports.SaveOpenExternalLogRequest = exports.SaveOpenExternalLogHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class SaveOpenExternalLogHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SaveOpenExternalLogHeaders = SaveOpenExternalLogHeaders;
class SaveOpenExternalLogRequest extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            logSource: 'logSource',
            logType: 'logType',
            openExt: 'openExt',
        };
    }
    static types() {
        return {
            corpId: 'string',
            logSource: 'string',
            logType: 'string',
            openExt: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SaveOpenExternalLogRequest = SaveOpenExternalLogRequest;
class SaveOpenExternalLogResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SaveOpenExternalLogResponseBody = SaveOpenExternalLogResponseBody;
class SaveOpenExternalLogResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SaveOpenExternalLogResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SaveOpenExternalLogResponse = SaveOpenExternalLogResponse;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 生态日志数据互通
     *
     * @param request SaveOpenExternalLogRequest
     * @param headers SaveOpenExternalLogHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SaveOpenExternalLogResponse
     */
    async saveOpenExternalLogWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.logSource)) {
            body["logSource"] = request.logSource;
        }
        if (!tea_util_1.default.isUnset(request.logType)) {
            body["logType"] = request.logType;
        }
        if (!tea_util_1.default.isUnset(request.openExt)) {
            body["openExt"] = request.openExt;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SaveOpenExternalLog",
            version: "yunShu_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/yunShu/externalLogs/save`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SaveOpenExternalLogResponse({}));
    }
    /**
     * @summary 生态日志数据互通
     *
     * @param request SaveOpenExternalLogRequest
     * @return SaveOpenExternalLogResponse
     */
    async saveOpenExternalLog(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SaveOpenExternalLogHeaders({});
        return await this.saveOpenExternalLogWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
