// @alicloud/dingtalk@2.1.16/dist/workbench_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkBenchGroupResponseBodyGroupList = exports.GetDingPortalDetailResponseBodyPages = exports.AddRecentUserAppListRequestUsedAppDetailList = exports.UpdateDingPortalPageScopeResponse = exports.UpdateDingPortalPageScopeRequest = exports.UpdateDingPortalPageScopeHeaders = exports.UndoDeletionResponse = exports.UndoDeletionResponseBody = exports.UndoDeletionRequest = exports.UndoDeletionHeaders = exports.QueryShortcutScopesResponse = exports.QueryShortcutScopesResponseBody = exports.QueryShortcutScopesHeaders = exports.QueryComponentScopesResponse = exports.QueryComponentScopesResponseBody = exports.QueryComponentScopesHeaders = exports.ModifyWorkbenchBadgeResponse = exports.ModifyWorkbenchBadgeResponseBody = exports.ModifyWorkbenchBadgeRequest = exports.ModifyWorkbenchBadgeHeaders = exports.ListWorkBenchGroupResponse = exports.ListWorkBenchGroupResponseBody = exports.ListWorkBenchGroupRequest = exports.ListWorkBenchGroupHeaders = exports.GetPluginRuleCheckInfoResponse = exports.GetPluginRuleCheckInfoResponseBody = exports.GetPluginRuleCheckInfoRequest = exports.GetPluginRuleCheckInfoHeaders = exports.GetPluginPermissionPointResponse = exports.GetPluginPermissionPointResponseBody = exports.GetPluginPermissionPointRequest = exports.GetPluginPermissionPointHeaders = exports.GetDingPortalDetailResponse = exports.GetDingPortalDetailResponseBody = exports.GetDingPortalDetailHeaders = exports.AddRecentUserAppListResponse = exports.AddRecentUserAppListResponseBody = exports.AddRecentUserAppListRequest = exports.AddRecentUserAppListHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AddRecentUserAppListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddRecentUserAppListHeaders = AddRecentUserAppListHeaders;
class AddRecentUserAppListRequest extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            usedAppDetailList: 'usedAppDetailList',
            userId: 'userId',
        };
    }
    static types() {
        return {
            corpId: 'string',
            usedAppDetailList: { 'type': 'array', 'itemType': AddRecentUserAppListRequestUsedAppDetailList },
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddRecentUserAppListRequest = AddRecentUserAppListRequest;
class AddRecentUserAppListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddRecentUserAppListResponseBody = AddRecentUserAppListResponseBody;
class AddRecentUserAppListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddRecentUserAppListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddRecentUserAppListResponse = AddRecentUserAppListResponse;
class GetDingPortalDetailHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetDingPortalDetailHeaders = GetDingPortalDetailHeaders;
class GetDingPortalDetailResponseBody extends $tea.Model {
    static names() {
        return {
            appUuid: 'appUuid',
            dingPortalName: 'dingPortalName',
            pages: 'pages',
        };
    }
    static types() {
        return {
            appUuid: 'string',
            dingPortalName: 'string',
            pages: { 'type': 'array', 'itemType': GetDingPortalDetailResponseBodyPages },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetDingPortalDetailResponseBody = GetDingPortalDetailResponseBody;
class GetDingPortalDetailResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetDingPortalDetailResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetDingPortalDetailResponse = GetDingPortalDetailResponse;
class GetPluginPermissionPointHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPluginPermissionPointHeaders = GetPluginPermissionPointHeaders;
class GetPluginPermissionPointRequest extends $tea.Model {
    static names() {
        return {
            miniAppId: 'miniAppId',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPluginPermissionPointRequest = GetPluginPermissionPointRequest;
class GetPluginPermissionPointResponseBody extends $tea.Model {
    static names() {
        return {
            permissionPointList: 'permissionPointList',
        };
    }
    static types() {
        return {
            permissionPointList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPluginPermissionPointResponseBody = GetPluginPermissionPointResponseBody;
class GetPluginPermissionPointResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetPluginPermissionPointResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPluginPermissionPointResponse = GetPluginPermissionPointResponse;
class GetPluginRuleCheckInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPluginRuleCheckInfoHeaders = GetPluginRuleCheckInfoHeaders;
class GetPluginRuleCheckInfoRequest extends $tea.Model {
    static names() {
        return {
            miniAppId: 'miniAppId',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPluginRuleCheckInfoRequest = GetPluginRuleCheckInfoRequest;
class GetPluginRuleCheckInfoResponseBody extends $tea.Model {
    static names() {
        return {
            packCode: 'packCode',
            pluginRuleCheckDetail: 'pluginRuleCheckDetail',
        };
    }
    static types() {
        return {
            packCode: 'string',
            pluginRuleCheckDetail: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPluginRuleCheckInfoResponseBody = GetPluginRuleCheckInfoResponseBody;
class GetPluginRuleCheckInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetPluginRuleCheckInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPluginRuleCheckInfoResponse = GetPluginRuleCheckInfoResponse;
class ListWorkBenchGroupHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListWorkBenchGroupHeaders = ListWorkBenchGroupHeaders;
class ListWorkBenchGroupRequest extends $tea.Model {
    static names() {
        return {
            ecologicalCorpId: 'ecologicalCorpId',
            groupType: 'groupType',
            opUnionId: 'opUnionId',
        };
    }
    static types() {
        return {
            ecologicalCorpId: 'string',
            groupType: 'string',
            opUnionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListWorkBenchGroupRequest = ListWorkBenchGroupRequest;
class ListWorkBenchGroupResponseBody extends $tea.Model {
    static names() {
        return {
            groupList: 'groupList',
        };
    }
    static types() {
        return {
            groupList: { 'type': 'array', 'itemType': ListWorkBenchGroupResponseBodyGroupList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListWorkBenchGroupResponseBody = ListWorkBenchGroupResponseBody;
class ListWorkBenchGroupResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListWorkBenchGroupResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListWorkBenchGroupResponse = ListWorkBenchGroupResponse;
class ModifyWorkbenchBadgeHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ModifyWorkbenchBadgeHeaders = ModifyWorkbenchBadgeHeaders;
class ModifyWorkbenchBadgeRequest extends $tea.Model {
    static names() {
        return {
            bizIdList: 'bizIdList',
            isAdded: 'isAdded',
            modifyMode: 'modifyMode',
            redDotRelationId: 'redDotRelationId',
            redDotType: 'redDotType',
            userId: 'userId',
        };
    }
    static types() {
        return {
            bizIdList: { 'type': 'array', 'itemType': 'string' },
            isAdded: 'boolean',
            modifyMode: 'string',
            redDotRelationId: 'string',
            redDotType: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ModifyWorkbenchBadgeRequest = ModifyWorkbenchBadgeRequest;
class ModifyWorkbenchBadgeResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ModifyWorkbenchBadgeResponseBody = ModifyWorkbenchBadgeResponseBody;
class ModifyWorkbenchBadgeResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ModifyWorkbenchBadgeResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ModifyWorkbenchBadgeResponse = ModifyWorkbenchBadgeResponse;
class QueryComponentScopesHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryComponentScopesHeaders = QueryComponentScopesHeaders;
class QueryComponentScopesResponseBody extends $tea.Model {
    static names() {
        return {
            deptVisibleScopes: 'deptVisibleScopes',
            userVisibleScopes: 'userVisibleScopes',
        };
    }
    static types() {
        return {
            deptVisibleScopes: { 'type': 'array', 'itemType': 'number' },
            userVisibleScopes: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryComponentScopesResponseBody = QueryComponentScopesResponseBody;
class QueryComponentScopesResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryComponentScopesResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryComponentScopesResponse = QueryComponentScopesResponse;
class QueryShortcutScopesHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryShortcutScopesHeaders = QueryShortcutScopesHeaders;
class QueryShortcutScopesResponseBody extends $tea.Model {
    static names() {
        return {
            deptVisibleScopes: 'deptVisibleScopes',
            userVisibleScopes: 'userVisibleScopes',
        };
    }
    static types() {
        return {
            deptVisibleScopes: { 'type': 'array', 'itemType': 'number' },
            userVisibleScopes: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryShortcutScopesResponseBody = QueryShortcutScopesResponseBody;
class QueryShortcutScopesResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryShortcutScopesResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryShortcutScopesResponse = QueryShortcutScopesResponse;
class UndoDeletionHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UndoDeletionHeaders = UndoDeletionHeaders;
class UndoDeletionRequest extends $tea.Model {
    static names() {
        return {
            bizIdList: 'bizIdList',
            redDotRelationId: 'redDotRelationId',
            redDotType: 'redDotType',
            userId: 'userId',
        };
    }
    static types() {
        return {
            bizIdList: { 'type': 'array', 'itemType': 'string' },
            redDotRelationId: 'string',
            redDotType: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UndoDeletionRequest = UndoDeletionRequest;
class UndoDeletionResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UndoDeletionResponseBody = UndoDeletionResponseBody;
class UndoDeletionResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UndoDeletionResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UndoDeletionResponse = UndoDeletionResponse;
class UpdateDingPortalPageScopeHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateDingPortalPageScopeHeaders = UpdateDingPortalPageScopeHeaders;
class UpdateDingPortalPageScopeRequest extends $tea.Model {
    static names() {
        return {
            allVisible: 'allVisible',
            deptIds: 'deptIds',
            roleIds: 'roleIds',
            userids: 'userids',
        };
    }
    static types() {
        return {
            allVisible: 'boolean',
            deptIds: { 'type': 'array', 'itemType': 'number' },
            roleIds: { 'type': 'array', 'itemType': 'number' },
            userids: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateDingPortalPageScopeRequest = UpdateDingPortalPageScopeRequest;
class UpdateDingPortalPageScopeResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateDingPortalPageScopeResponse = UpdateDingPortalPageScopeResponse;
class AddRecentUserAppListRequestUsedAppDetailList extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
        };
    }
    static types() {
        return {
            agentId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddRecentUserAppListRequestUsedAppDetailList = AddRecentUserAppListRequestUsedAppDetailList;
class GetDingPortalDetailResponseBodyPages extends $tea.Model {
    static names() {
        return {
            allVisible: 'allVisible',
            deptIds: 'deptIds',
            pageName: 'pageName',
            pageUuid: 'pageUuid',
            roleIds: 'roleIds',
            userids: 'userids',
        };
    }
    static types() {
        return {
            allVisible: 'boolean',
            deptIds: { 'type': 'array', 'itemType': 'number' },
            pageName: 'string',
            pageUuid: 'string',
            roleIds: { 'type': 'array', 'itemType': 'number' },
            userids: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetDingPortalDetailResponseBodyPages = GetDingPortalDetailResponseBodyPages;
class ListWorkBenchGroupResponseBodyGroupList extends $tea.Model {
    static names() {
        return {
            componentId: 'componentId',
            name: 'name',
        };
    }
    static types() {
        return {
            componentId: 'string',
            name: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListWorkBenchGroupResponseBodyGroupList = ListWorkBenchGroupResponseBodyGroupList;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 批量添加最近使用记录
     *
     * @param request AddRecentUserAppListRequest
     * @param headers AddRecentUserAppListHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddRecentUserAppListResponse
     */
    async addRecentUserAppListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.usedAppDetailList)) {
            body["usedAppDetailList"] = request.usedAppDetailList;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddRecentUserAppList",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/components/recentUsed/batch`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddRecentUserAppListResponse({}));
    }
    /**
     * @summary 批量添加最近使用记录
     *
     * @param request AddRecentUserAppListRequest
     * @return AddRecentUserAppListResponse
     */
    async addRecentUserAppList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddRecentUserAppListHeaders({});
        return await this.addRecentUserAppListWithOptions(request, headers, runtime);
    }
    /**
     * @summary 查询自定义工作台
     *
     * @param headers GetDingPortalDetailHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetDingPortalDetailResponse
     */
    async getDingPortalDetailWithOptions(appUuid, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "GetDingPortalDetail",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/dingPortals/${appUuid}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetDingPortalDetailResponse({}));
    }
    /**
     * @summary 查询自定义工作台
     *
     * @return GetDingPortalDetailResponse
     */
    async getDingPortalDetail(appUuid) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetDingPortalDetailHeaders({});
        return await this.getDingPortalDetailWithOptions(appUuid, headers, runtime);
    }
    /**
     * @summary 获取工作台插件的权限点
     *
     * @param request GetPluginPermissionPointRequest
     * @param headers GetPluginPermissionPointHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetPluginPermissionPointResponse
     */
    async getPluginPermissionPointWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetPluginPermissionPoint",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/plugins/permissions`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetPluginPermissionPointResponse({}));
    }
    /**
     * @summary 获取工作台插件的权限点
     *
     * @param request GetPluginPermissionPointRequest
     * @return GetPluginPermissionPointResponse
     */
    async getPluginPermissionPoint(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetPluginPermissionPointHeaders({});
        return await this.getPluginPermissionPointWithOptions(request, headers, runtime);
    }
    /**
     * @summary 获取插件的校验规则
     *
     * @param request GetPluginRuleCheckInfoRequest
     * @param headers GetPluginRuleCheckInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetPluginRuleCheckInfoResponse
     */
    async getPluginRuleCheckInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetPluginRuleCheckInfo",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/plugins/validationRules`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetPluginRuleCheckInfoResponse({}));
    }
    /**
     * @summary 获取插件的校验规则
     *
     * @param request GetPluginRuleCheckInfoRequest
     * @return GetPluginRuleCheckInfoResponse
     */
    async getPluginRuleCheckInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetPluginRuleCheckInfoHeaders({});
        return await this.getPluginRuleCheckInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 获取工作台分组列表
     *
     * @param request ListWorkBenchGroupRequest
     * @param headers ListWorkBenchGroupHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListWorkBenchGroupResponse
     */
    async listWorkBenchGroupWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.ecologicalCorpId)) {
            query["ecologicalCorpId"] = request.ecologicalCorpId;
        }
        if (!tea_util_1.default.isUnset(request.groupType)) {
            query["groupType"] = request.groupType;
        }
        if (!tea_util_1.default.isUnset(request.opUnionId)) {
            query["opUnionId"] = request.opUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListWorkBenchGroup",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/groups`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListWorkBenchGroupResponse({}));
    }
    /**
     * @summary 获取工作台分组列表
     *
     * @param request ListWorkBenchGroupRequest
     * @return ListWorkBenchGroupResponse
     */
    async listWorkBenchGroup(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListWorkBenchGroupHeaders({});
        return await this.listWorkBenchGroupWithOptions(request, headers, runtime);
    }
    /**
     * @summary 工作台支持数字红点
     *
     * @param request ModifyWorkbenchBadgeRequest
     * @param headers ModifyWorkbenchBadgeHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ModifyWorkbenchBadgeResponse
     */
    async modifyWorkbenchBadgeWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizIdList)) {
            body["bizIdList"] = request.bizIdList;
        }
        if (!tea_util_1.default.isUnset(request.isAdded)) {
            body["isAdded"] = request.isAdded;
        }
        if (!tea_util_1.default.isUnset(request.modifyMode)) {
            body["modifyMode"] = request.modifyMode;
        }
        if (!tea_util_1.default.isUnset(request.redDotRelationId)) {
            body["redDotRelationId"] = request.redDotRelationId;
        }
        if (!tea_util_1.default.isUnset(request.redDotType)) {
            body["redDotType"] = request.redDotType;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ModifyWorkbenchBadge",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/badges/modify`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ModifyWorkbenchBadgeResponse({}));
    }
    /**
     * @summary 工作台支持数字红点
     *
     * @param request ModifyWorkbenchBadgeRequest
     * @return ModifyWorkbenchBadgeResponse
     */
    async modifyWorkbenchBadge(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ModifyWorkbenchBadgeHeaders({});
        return await this.modifyWorkbenchBadgeWithOptions(request, headers, runtime);
    }
    /**
     * @summary 工作台组件授权范围查询
     *
     * @param headers QueryComponentScopesHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryComponentScopesResponse
     */
    async queryComponentScopesWithOptions(componentId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "QueryComponentScopes",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/components/${componentId}/scopes`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryComponentScopesResponse({}));
    }
    /**
     * @summary 工作台组件授权范围查询
     *
     * @return QueryComponentScopesResponse
     */
    async queryComponentScopes(componentId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryComponentScopesHeaders({});
        return await this.queryComponentScopesWithOptions(componentId, headers, runtime);
    }
    /**
     * @summary 查询快捷方式可见范围
     *
     * @param headers QueryShortcutScopesHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryShortcutScopesResponse
     */
    async queryShortcutScopesWithOptions(shortcutKey, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "QueryShortcutScopes",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/shortcuts/${shortcutKey}/scopes`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryShortcutScopesResponse({}));
    }
    /**
     * @summary 查询快捷方式可见范围
     *
     * @return QueryShortcutScopesResponse
     */
    async queryShortcutScopes(shortcutKey) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryShortcutScopesHeaders({});
        return await this.queryShortcutScopesWithOptions(shortcutKey, headers, runtime);
    }
    /**
     * @summary 工作台数字红点支持撤销已被删除的资源
     *
     * @param request UndoDeletionRequest
     * @param headers UndoDeletionHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UndoDeletionResponse
     */
    async undoDeletionWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizIdList)) {
            body["bizIdList"] = request.bizIdList;
        }
        if (!tea_util_1.default.isUnset(request.redDotRelationId)) {
            body["redDotRelationId"] = request.redDotRelationId;
        }
        if (!tea_util_1.default.isUnset(request.redDotType)) {
            body["redDotType"] = request.redDotType;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UndoDeletion",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/badges/undoDeleted`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UndoDeletionResponse({}));
    }
    /**
     * @summary 工作台数字红点支持撤销已被删除的资源
     *
     * @param request UndoDeletionRequest
     * @return UndoDeletionResponse
     */
    async undoDeletion(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UndoDeletionHeaders({});
        return await this.undoDeletionWithOptions(request, headers, runtime);
    }
    /**
     * @summary 更新自定义工作台页面可见性
     *
     * @param request UpdateDingPortalPageScopeRequest
     * @param headers UpdateDingPortalPageScopeHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateDingPortalPageScopeResponse
     */
    async updateDingPortalPageScopeWithOptions(pageUuid, appUuid, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.allVisible)) {
            body["allVisible"] = request.allVisible;
        }
        if (!tea_util_1.default.isUnset(request.deptIds)) {
            body["deptIds"] = request.deptIds;
        }
        if (!tea_util_1.default.isUnset(request.roleIds)) {
            body["roleIds"] = request.roleIds;
        }
        if (!tea_util_1.default.isUnset(request.userids)) {
            body["userids"] = request.userids;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateDingPortalPageScope",
            version: "workbench_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/workbench/dingPortals/${appUuid}/pageScopes/${pageUuid}`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateDingPortalPageScopeResponse({}));
    }
    /**
     * @summary 更新自定义工作台页面可见性
     *
     * @param request UpdateDingPortalPageScopeRequest
     * @return UpdateDingPortalPageScopeResponse
     */
    async updateDingPortalPageScope(pageUuid, appUuid, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateDingPortalPageScopeHeaders({});
        return await this.updateDingPortalPageScopeWithOptions(pageUuid, appUuid, request, headers, runtime);
    }
}
exports.default = Client;
