// @alicloud/dingtalk@2.1.16/dist/transcribe_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePermissionForUsersRequestMembers = exports.RemovePermissionResponseBodyData = exports.RemovePermissionRequestMembers = exports.GetTranscribeBriefResponseBodyData = exports.UpdatePermissionForUsersResponse = exports.UpdatePermissionForUsersResponseBody = exports.UpdatePermissionForUsersRequest = exports.UpdatePermissionForUsersHeaders = exports.RemovePermissionResponse = exports.RemovePermissionResponseBody = exports.RemovePermissionRequest = exports.RemovePermissionHeaders = exports.GetTranscribeBriefResponse = exports.GetTranscribeBriefResponseBody = exports.GetTranscribeBriefHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class GetTranscribeBriefHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscribeBriefHeaders = GetTranscribeBriefHeaders;
class GetTranscribeBriefResponseBody extends $tea.Model {
    static names() {
        return {
            data: 'data',
            statusCode: 'statusCode',
            success: 'success',
        };
    }
    static types() {
        return {
            data: GetTranscribeBriefResponseBodyData,
            statusCode: 'number',
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscribeBriefResponseBody = GetTranscribeBriefResponseBody;
class GetTranscribeBriefResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetTranscribeBriefResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscribeBriefResponse = GetTranscribeBriefResponse;
class RemovePermissionHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemovePermissionHeaders = RemovePermissionHeaders;
class RemovePermissionRequest extends $tea.Model {
    static names() {
        return {
            bizType: 'bizType',
            members: 'members',
            taskCreator: 'taskCreator',
            taskId: 'taskId',
        };
    }
    static types() {
        return {
            bizType: 'number',
            members: { 'type': 'array', 'itemType': RemovePermissionRequestMembers },
            taskCreator: 'number',
            taskId: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemovePermissionRequest = RemovePermissionRequest;
class RemovePermissionResponseBody extends $tea.Model {
    static names() {
        return {
            data: 'data',
            statusCode: 'statusCode',
            success: 'success',
        };
    }
    static types() {
        return {
            data: RemovePermissionResponseBodyData,
            statusCode: 'number',
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemovePermissionResponseBody = RemovePermissionResponseBody;
class RemovePermissionResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RemovePermissionResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemovePermissionResponse = RemovePermissionResponse;
class UpdatePermissionForUsersHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdatePermissionForUsersHeaders = UpdatePermissionForUsersHeaders;
class UpdatePermissionForUsersRequest extends $tea.Model {
    static names() {
        return {
            bizType: 'bizType',
            members: 'members',
            taskCreator: 'taskCreator',
            operatorUid: 'operatorUid',
        };
    }
    static types() {
        return {
            bizType: 'number',
            members: { 'type': 'array', 'itemType': UpdatePermissionForUsersRequestMembers },
            taskCreator: 'number',
            operatorUid: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdatePermissionForUsersRequest = UpdatePermissionForUsersRequest;
class UpdatePermissionForUsersResponseBody extends $tea.Model {
    static names() {
        return {
            isSuccess: 'isSuccess',
        };
    }
    static types() {
        return {
            isSuccess: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdatePermissionForUsersResponseBody = UpdatePermissionForUsersResponseBody;
class UpdatePermissionForUsersResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdatePermissionForUsersResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdatePermissionForUsersResponse = UpdatePermissionForUsersResponse;
class GetTranscribeBriefResponseBodyData extends $tea.Model {
    static names() {
        return {
            bizType: 'bizType',
        };
    }
    static types() {
        return {
            bizType: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscribeBriefResponseBodyData = GetTranscribeBriefResponseBodyData;
class RemovePermissionRequestMembers extends $tea.Model {
    static names() {
        return {
            memberId: 'memberId',
            memberType: 'memberType',
            policyType: 'policyType',
        };
    }
    static types() {
        return {
            memberId: 'number',
            memberType: 'string',
            policyType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemovePermissionRequestMembers = RemovePermissionRequestMembers;
class RemovePermissionResponseBodyData extends $tea.Model {
    static names() {
        return {
            failNameList: 'failNameList',
        };
    }
    static types() {
        return {
            failNameList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemovePermissionResponseBodyData = RemovePermissionResponseBodyData;
class UpdatePermissionForUsersRequestMembers extends $tea.Model {
    static names() {
        return {
            memberId: 'memberId',
            memberType: 'memberType',
            policyType: 'policyType',
        };
    }
    static types() {
        return {
            memberId: 'number',
            memberType: 'string',
            policyType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdatePermissionForUsersRequestMembers = UpdatePermissionForUsersRequestMembers;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 获取闪记任务的概要信息
     *
     * @param headers GetTranscribeBriefHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetTranscribeBriefResponse
     */
    async getTranscribeBriefWithOptions(taskUuid, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "GetTranscribeBrief",
            version: "transcribe_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/transcribe/tasks/${taskUuid}/briefInfos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetTranscribeBriefResponse({}));
    }
    /**
     * @summary 获取闪记任务的概要信息
     *
     * @return GetTranscribeBriefResponse
     */
    async getTranscribeBrief(taskUuid) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetTranscribeBriefHeaders({});
        return await this.getTranscribeBriefWithOptions(taskUuid, headers, runtime);
    }
    /**
     * @summary 移除指定用户对闪记任务的权限
     *
     * @param request RemovePermissionRequest
     * @param headers RemovePermissionHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RemovePermissionResponse
     */
    async removePermissionWithOptions(taskUuid, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizType)) {
            body["bizType"] = request.bizType;
        }
        if (!tea_util_1.default.isUnset(request.members)) {
            body["members"] = request.members;
        }
        if (!tea_util_1.default.isUnset(request.taskCreator)) {
            body["taskCreator"] = request.taskCreator;
        }
        if (!tea_util_1.default.isUnset(request.taskId)) {
            body["taskId"] = request.taskId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "RemovePermission",
            version: "transcribe_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/transcribe/tasks/${taskUuid}/permissions/remove`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RemovePermissionResponse({}));
    }
    /**
     * @summary 移除指定用户对闪记任务的权限
     *
     * @param request RemovePermissionRequest
     * @return RemovePermissionResponse
     */
    async removePermission(taskUuid, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RemovePermissionHeaders({});
        return await this.removePermissionWithOptions(taskUuid, request, headers, runtime);
    }
    /**
     * @summary 针对指定的闪记，修改或者授予指定用户权限
     *
     * @param request UpdatePermissionForUsersRequest
     * @param headers UpdatePermissionForUsersHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdatePermissionForUsersResponse
     */
    async updatePermissionForUsersWithOptions(taskUuid, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.operatorUid)) {
            query["operatorUid"] = request.operatorUid;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizType)) {
            body["bizType"] = request.bizType;
        }
        if (!tea_util_1.default.isUnset(request.members)) {
            body["members"] = request.members;
        }
        if (!tea_util_1.default.isUnset(request.taskCreator)) {
            body["taskCreator"] = request.taskCreator;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdatePermissionForUsers",
            version: "transcribe_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/transcribe/tasks/${taskUuid}/permissions`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdatePermissionForUsersResponse({}));
    }
    /**
     * @summary 针对指定的闪记，修改或者授予指定用户权限
     *
     * @param request UpdatePermissionForUsersRequest
     * @return UpdatePermissionForUsersResponse
     */
    async updatePermissionForUsers(taskUuid, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdatePermissionForUsersHeaders({});
        return await this.updatePermissionForUsersWithOptions(taskUuid, request, headers, runtime);
    }
}
exports.default = Client;
