// @alicloud/dingtalk@2.1.16/dist/connector_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateConnectorRequest = exports.UpdateConnectorHeaders = exports.UpdateActionResponse = exports.UpdateActionResponseBody = exports.UpdateActionRequest = exports.UpdateActionHeaders = exports.SyncDataResponse = exports.SyncDataResponseBody = exports.SyncDataRequest = exports.SyncDataHeaders = exports.SearchConnectorsResponse = exports.SearchConnectorsResponseBody = exports.SearchConnectorsRequest = exports.SearchConnectorsHeaders = exports.SearchActionsResponse = exports.SearchActionsResponseBody = exports.SearchActionsRequest = exports.SearchActionsHeaders = exports.PullDataByPkResponse = exports.PullDataByPkResponseBody = exports.PullDataByPkRequest = exports.PullDataByPkHeaders = exports.PullDataByPageResponse = exports.PullDataByPageResponseBody = exports.PullDataByPageRequest = exports.PullDataByPageHeaders = exports.InvokeInstanceResponse = exports.InvokeInstanceResponseBody = exports.InvokeInstanceRequest = exports.InvokeInstanceHeaders = exports.GetActionDetailResponse = exports.GetActionDetailResponseBody = exports.GetActionDetailRequest = exports.GetActionDetailHeaders = exports.CreateTriggerResponse = exports.CreateTriggerResponseBody = exports.CreateTriggerRequest = exports.CreateTriggerHeaders = exports.CreateInvocableInstanceResponse = exports.CreateInvocableInstanceResponseBody = exports.CreateInvocableInstanceRequest = exports.CreateInvocableInstanceHeaders = exports.CreateConnectorResponse = exports.CreateConnectorResponseBody = exports.CreateConnectorRequest = exports.CreateConnectorHeaders = exports.CreateActionResponse = exports.CreateActionResponseBody = exports.CreateActionRequest = exports.CreateActionHeaders = void 0;
exports.UpdateTriggerResponseBodyItem = exports.UpdateTriggerRequestTriggerInfo = exports.UpdateConnectorResponseBodyItem = exports.UpdateConnectorRequestConnectorInfo = exports.UpdateActionResponseBodyItem = exports.UpdateActionRequestActionInfo = exports.UpdateActionRequestActionInfoOutputMappingConfig = exports.UpdateActionRequestActionInfoOutputDataRules = exports.UpdateActionRequestActionInfoInputMappingConfig = exports.SyncDataResponseBodyList = exports.SyncDataRequestTriggerDataList = exports.SearchConnectorsResponseBodyList = exports.SearchActionsResponseBodyList = exports.PullDataByPageResponseBodyList = exports.GetActionDetailResponseBodyIntegrationConfig = exports.GetActionDetailResponseBodyIntegrationConfigProps = exports.GetActionDetailResponseBodyIntegrationConfigCategoryNames = exports.CreateTriggerResponseBodyItem = exports.CreateTriggerRequestTriggerInfo = exports.CreateConnectorResponseBodyItem = exports.CreateConnectorRequestConnectorInfo = exports.CreateActionResponseBodyItem = exports.CreateActionRequestActionInfo = exports.CreateActionRequestActionInfoOutputMappingConfig = exports.CreateActionRequestActionInfoOutputDataRules = exports.CreateActionRequestActionInfoInputMappingConfig = exports.UpdateTriggerResponse = exports.UpdateTriggerResponseBody = exports.UpdateTriggerRequest = exports.UpdateTriggerHeaders = exports.UpdateConnectorResponse = exports.UpdateConnectorResponseBody = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CreateActionHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateActionHeaders = CreateActionHeaders;
class CreateActionRequest extends $tea.Model {
    static names() {
        return {
            actionInfo: 'actionInfo',
            integratorFlag: 'integratorFlag',
        };
    }
    static types() {
        return {
            actionInfo: { 'type': 'array', 'itemType': CreateActionRequestActionInfo },
            integratorFlag: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateActionRequest = CreateActionRequest;
class CreateActionResponseBody extends $tea.Model {
    static names() {
        return {
            item: 'item',
        };
    }
    static types() {
        return {
            item: { 'type': 'array', 'itemType': CreateActionResponseBodyItem },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateActionResponseBody = CreateActionResponseBody;
class CreateActionResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateActionResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateActionResponse = CreateActionResponse;
class CreateConnectorHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateConnectorHeaders = CreateConnectorHeaders;
class CreateConnectorRequest extends $tea.Model {
    static names() {
        return {
            connectorInfo: 'connectorInfo',
            integratorFlag: 'integratorFlag',
        };
    }
    static types() {
        return {
            connectorInfo: { 'type': 'array', 'itemType': CreateConnectorRequestConnectorInfo },
            integratorFlag: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateConnectorRequest = CreateConnectorRequest;
class CreateConnectorResponseBody extends $tea.Model {
    static names() {
        return {
            item: 'item',
        };
    }
    static types() {
        return {
            item: { 'type': 'array', 'itemType': CreateConnectorResponseBodyItem },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateConnectorResponseBody = CreateConnectorResponseBody;
class CreateConnectorResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateConnectorResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateConnectorResponse = CreateConnectorResponse;
class CreateInvocableInstanceHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateInvocableInstanceHeaders = CreateInvocableInstanceHeaders;
class CreateInvocableInstanceRequest extends $tea.Model {
    static names() {
        return {
            connectAssetUri: 'connectAssetUri',
            instanceKey: 'instanceKey',
        };
    }
    static types() {
        return {
            connectAssetUri: 'string',
            instanceKey: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateInvocableInstanceRequest = CreateInvocableInstanceRequest;
class CreateInvocableInstanceResponseBody extends $tea.Model {
    static names() {
        return {
            connectAssetUri: 'connectAssetUri',
            versionId: 'versionId',
        };
    }
    static types() {
        return {
            connectAssetUri: 'string',
            versionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateInvocableInstanceResponseBody = CreateInvocableInstanceResponseBody;
class CreateInvocableInstanceResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateInvocableInstanceResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateInvocableInstanceResponse = CreateInvocableInstanceResponse;
class CreateTriggerHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateTriggerHeaders = CreateTriggerHeaders;
class CreateTriggerRequest extends $tea.Model {
    static names() {
        return {
            integratorFlag: 'integratorFlag',
            triggerInfo: 'triggerInfo',
        };
    }
    static types() {
        return {
            integratorFlag: 'string',
            triggerInfo: { 'type': 'array', 'itemType': CreateTriggerRequestTriggerInfo },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateTriggerRequest = CreateTriggerRequest;
class CreateTriggerResponseBody extends $tea.Model {
    static names() {
        return {
            item: 'item',
        };
    }
    static types() {
        return {
            item: { 'type': 'array', 'itemType': CreateTriggerResponseBodyItem },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateTriggerResponseBody = CreateTriggerResponseBody;
class CreateTriggerResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateTriggerResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateTriggerResponse = CreateTriggerResponse;
class GetActionDetailHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetActionDetailHeaders = GetActionDetailHeaders;
class GetActionDetailRequest extends $tea.Model {
    static names() {
        return {
            connectAssetUri: 'connectAssetUri',
        };
    }
    static types() {
        return {
            connectAssetUri: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetActionDetailRequest = GetActionDetailRequest;
class GetActionDetailResponseBody extends $tea.Model {
    static names() {
        return {
            connectAssetUri: 'connectAssetUri',
            inputSchema: 'inputSchema',
            integrationConfig: 'integrationConfig',
            name: 'name',
            outputSchema: 'outputSchema',
            refId: 'refId',
            refProviderCorpId: 'refProviderCorpId',
            refType: 'refType',
        };
    }
    static types() {
        return {
            connectAssetUri: 'string',
            inputSchema: 'string',
            integrationConfig: GetActionDetailResponseBodyIntegrationConfig,
            name: 'string',
            outputSchema: 'string',
            refId: 'string',
            refProviderCorpId: 'string',
            refType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetActionDetailResponseBody = GetActionDetailResponseBody;
class GetActionDetailResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetActionDetailResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetActionDetailResponse = GetActionDetailResponse;
class InvokeInstanceHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.InvokeInstanceHeaders = InvokeInstanceHeaders;
class InvokeInstanceRequest extends $tea.Model {
    static names() {
        return {
            connectAssetUri: 'connectAssetUri',
            inputJsonString: 'inputJsonString',
            instanceKey: 'instanceKey',
        };
    }
    static types() {
        return {
            connectAssetUri: 'string',
            inputJsonString: 'string',
            instanceKey: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.InvokeInstanceRequest = InvokeInstanceRequest;
class InvokeInstanceResponseBody extends $tea.Model {
    static names() {
        return {
            cost: 'cost',
            errorCode: 'errorCode',
            errorMessage: 'errorMessage',
            instanceId: 'instanceId',
            outputJson: 'outputJson',
        };
    }
    static types() {
        return {
            cost: 'number',
            errorCode: 'string',
            errorMessage: 'string',
            instanceId: 'string',
            outputJson: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.InvokeInstanceResponseBody = InvokeInstanceResponseBody;
class InvokeInstanceResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: InvokeInstanceResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.InvokeInstanceResponse = InvokeInstanceResponse;
class PullDataByPageHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PullDataByPageHeaders = PullDataByPageHeaders;
class PullDataByPageRequest extends $tea.Model {
    static names() {
        return {
            appId: 'appId',
            dataModelId: 'dataModelId',
            datetimeFilterField: 'datetimeFilterField',
            maxDatetime: 'maxDatetime',
            maxResults: 'maxResults',
            minDatetime: 'minDatetime',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            appId: 'string',
            dataModelId: 'string',
            datetimeFilterField: 'string',
            maxDatetime: 'number',
            maxResults: 'number',
            minDatetime: 'number',
            nextToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PullDataByPageRequest = PullDataByPageRequest;
class PullDataByPageResponseBody extends $tea.Model {
    static names() {
        return {
            list: 'list',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            list: { 'type': 'array', 'itemType': PullDataByPageResponseBodyList },
            maxResults: 'number',
            nextToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PullDataByPageResponseBody = PullDataByPageResponseBody;
class PullDataByPageResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PullDataByPageResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PullDataByPageResponse = PullDataByPageResponse;
class PullDataByPkHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PullDataByPkHeaders = PullDataByPkHeaders;
class PullDataByPkRequest extends $tea.Model {
    static names() {
        return {
            appId: 'appId',
            primaryKey: 'primaryKey',
        };
    }
    static types() {
        return {
            appId: 'string',
            primaryKey: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PullDataByPkRequest = PullDataByPkRequest;
class PullDataByPkResponseBody extends $tea.Model {
    static names() {
        return {
            dataCreateAppId: 'dataCreateAppId',
            dataCreateAppType: 'dataCreateAppType',
            dataGmtCreate: 'dataGmtCreate',
            dataGmtModified: 'dataGmtModified',
            dataModifiedAppId: 'dataModifiedAppId',
            dataModifiedAppType: 'dataModifiedAppType',
            jsonData: 'jsonData',
        };
    }
    static types() {
        return {
            dataCreateAppId: 'string',
            dataCreateAppType: 'string',
            dataGmtCreate: 'number',
            dataGmtModified: 'number',
            dataModifiedAppId: 'string',
            dataModifiedAppType: 'string',
            jsonData: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PullDataByPkResponseBody = PullDataByPkResponseBody;
class PullDataByPkResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PullDataByPkResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PullDataByPkResponse = PullDataByPkResponse;
class SearchActionsHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchActionsHeaders = SearchActionsHeaders;
class SearchActionsRequest extends $tea.Model {
    static names() {
        return {
            connectorId: 'connectorId',
            connectorProviderCorpId: 'connectorProviderCorpId',
            integrationTypes: 'integrationTypes',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            connectorId: 'string',
            connectorProviderCorpId: 'string',
            integrationTypes: { 'type': 'array', 'itemType': 'string' },
            maxResults: 'number',
            nextToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchActionsRequest = SearchActionsRequest;
class SearchActionsResponseBody extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            nextToken: 'nextToken',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': SearchActionsResponseBodyList },
            nextToken: 'string',
            totalCount: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchActionsResponseBody = SearchActionsResponseBody;
class SearchActionsResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SearchActionsResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchActionsResponse = SearchActionsResponse;
class SearchConnectorsHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchConnectorsHeaders = SearchConnectorsHeaders;
class SearchConnectorsRequest extends $tea.Model {
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            type: 'type',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'string',
            type: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchConnectorsRequest = SearchConnectorsRequest;
class SearchConnectorsResponseBody extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            nextToken: 'nextToken',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': SearchConnectorsResponseBodyList },
            nextToken: 'string',
            totalCount: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchConnectorsResponseBody = SearchConnectorsResponseBody;
class SearchConnectorsResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SearchConnectorsResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchConnectorsResponse = SearchConnectorsResponse;
class SyncDataHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncDataHeaders = SyncDataHeaders;
class SyncDataRequest extends $tea.Model {
    static names() {
        return {
            appId: 'appId',
            triggerDataList: 'triggerDataList',
        };
    }
    static types() {
        return {
            appId: 'string',
            triggerDataList: { 'type': 'array', 'itemType': SyncDataRequestTriggerDataList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncDataRequest = SyncDataRequest;
class SyncDataResponseBody extends $tea.Model {
    static names() {
        return {
            list: 'list',
        };
    }
    static types() {
        return {
            list: { 'type': 'array', 'itemType': SyncDataResponseBodyList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncDataResponseBody = SyncDataResponseBody;
class SyncDataResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SyncDataResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncDataResponse = SyncDataResponse;
class UpdateActionHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateActionHeaders = UpdateActionHeaders;
class UpdateActionRequest extends $tea.Model {
    static names() {
        return {
            actionInfo: 'actionInfo',
            integratorFlag: 'integratorFlag',
        };
    }
    static types() {
        return {
            actionInfo: { 'type': 'array', 'itemType': UpdateActionRequestActionInfo },
            integratorFlag: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateActionRequest = UpdateActionRequest;
class UpdateActionResponseBody extends $tea.Model {
    static names() {
        return {
            item: 'item',
        };
    }
    static types() {
        return {
            item: { 'type': 'array', 'itemType': UpdateActionResponseBodyItem },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateActionResponseBody = UpdateActionResponseBody;
class UpdateActionResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateActionResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateActionResponse = UpdateActionResponse;
class UpdateConnectorHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateConnectorHeaders = UpdateConnectorHeaders;
class UpdateConnectorRequest extends $tea.Model {
    static names() {
        return {
            connectorInfo: 'connectorInfo',
            integratorFlag: 'integratorFlag',
        };
    }
    static types() {
        return {
            connectorInfo: { 'type': 'array', 'itemType': UpdateConnectorRequestConnectorInfo },
            integratorFlag: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateConnectorRequest = UpdateConnectorRequest;
class UpdateConnectorResponseBody extends $tea.Model {
    static names() {
        return {
            item: 'item',
        };
    }
    static types() {
        return {
            item: { 'type': 'array', 'itemType': UpdateConnectorResponseBodyItem },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateConnectorResponseBody = UpdateConnectorResponseBody;
class UpdateConnectorResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateConnectorResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateConnectorResponse = UpdateConnectorResponse;
class UpdateTriggerHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateTriggerHeaders = UpdateTriggerHeaders;
class UpdateTriggerRequest extends $tea.Model {
    static names() {
        return {
            integratorFlag: 'integratorFlag',
            triggerInfo: 'triggerInfo',
        };
    }
    static types() {
        return {
            integratorFlag: 'string',
            triggerInfo: { 'type': 'array', 'itemType': UpdateTriggerRequestTriggerInfo },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateTriggerRequest = UpdateTriggerRequest;
class UpdateTriggerResponseBody extends $tea.Model {
    static names() {
        return {
            item: 'item',
        };
    }
    static types() {
        return {
            item: { 'type': 'array', 'itemType': UpdateTriggerResponseBodyItem },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateTriggerResponseBody = UpdateTriggerResponseBody;
class UpdateTriggerResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateTriggerResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateTriggerResponse = UpdateTriggerResponse;
class CreateActionRequestActionInfoInputMappingConfig extends $tea.Model {
    static names() {
        return {
            customSchemaMapping: 'customSchemaMapping',
            rules: 'rules',
        };
    }
    static types() {
        return {
            customSchemaMapping: 'string',
            rules: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateActionRequestActionInfoInputMappingConfig = CreateActionRequestActionInfoInputMappingConfig;
class CreateActionRequestActionInfoOutputDataRules extends $tea.Model {
    static names() {
        return {
            expectValue: 'expectValue',
            operate: 'operate',
            propertyPath: 'propertyPath',
        };
    }
    static types() {
        return {
            expectValue: 'string',
            operate: 'string',
            propertyPath: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateActionRequestActionInfoOutputDataRules = CreateActionRequestActionInfoOutputDataRules;
class CreateActionRequestActionInfoOutputMappingConfig extends $tea.Model {
    static names() {
        return {
            customSchemaMapping: 'customSchemaMapping',
            rules: 'rules',
        };
    }
    static types() {
        return {
            customSchemaMapping: 'string',
            rules: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateActionRequestActionInfoOutputMappingConfig = CreateActionRequestActionInfoOutputMappingConfig;
class CreateActionRequestActionInfo extends $tea.Model {
    static names() {
        return {
            apiPath: 'apiPath',
            description: 'description',
            dingConnectorId: 'dingConnectorId',
            inputMappingConfig: 'inputMappingConfig',
            inputSchema: 'inputSchema',
            integratorActionId: 'integratorActionId',
            integratorConnectorId: 'integratorConnectorId',
            name: 'name',
            outputDataRules: 'outputDataRules',
            outputMappingConfig: 'outputMappingConfig',
            outputSchema: 'outputSchema',
        };
    }
    static types() {
        return {
            apiPath: 'string',
            description: 'string',
            dingConnectorId: 'string',
            inputMappingConfig: CreateActionRequestActionInfoInputMappingConfig,
            inputSchema: 'string',
            integratorActionId: 'string',
            integratorConnectorId: 'string',
            name: 'string',
            outputDataRules: { 'type': 'array', 'itemType': CreateActionRequestActionInfoOutputDataRules },
            outputMappingConfig: CreateActionRequestActionInfoOutputMappingConfig,
            outputSchema: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateActionRequestActionInfo = CreateActionRequestActionInfo;
class CreateActionResponseBodyItem extends $tea.Model {
    static names() {
        return {
            dingActionId: 'dingActionId',
            dingConnectorId: 'dingConnectorId',
            integratorActionId: 'integratorActionId',
            integratorConnectorId: 'integratorConnectorId',
            subErrCode: 'subErrCode',
            subErrMsg: 'subErrMsg',
            success: 'success',
        };
    }
    static types() {
        return {
            dingActionId: 'string',
            dingConnectorId: 'string',
            integratorActionId: 'string',
            integratorConnectorId: 'string',
            subErrCode: 'string',
            subErrMsg: 'string',
            success: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateActionResponseBodyItem = CreateActionResponseBodyItem;
class CreateConnectorRequestConnectorInfo extends $tea.Model {
    static names() {
        return {
            apiDomain: 'apiDomain',
            apiSecret: 'apiSecret',
            appId: 'appId',
            authValueEnv: 'authValueEnv',
            description: 'description',
            domainEnv: 'domainEnv',
            iconMediaId: 'iconMediaId',
            integratorConnectorId: 'integratorConnectorId',
            name: 'name',
        };
    }
    static types() {
        return {
            apiDomain: 'string',
            apiSecret: 'string',
            appId: 'number',
            authValueEnv: 'boolean',
            description: 'string',
            domainEnv: 'boolean',
            iconMediaId: 'string',
            integratorConnectorId: 'string',
            name: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateConnectorRequestConnectorInfo = CreateConnectorRequestConnectorInfo;
class CreateConnectorResponseBodyItem extends $tea.Model {
    static names() {
        return {
            dingConnectorId: 'dingConnectorId',
            integratorConnectorId: 'integratorConnectorId',
            subErrCode: 'subErrCode',
            subErrMsg: 'subErrMsg',
            success: 'success',
        };
    }
    static types() {
        return {
            dingConnectorId: 'string',
            integratorConnectorId: 'string',
            subErrCode: 'string',
            subErrMsg: 'string',
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateConnectorResponseBodyItem = CreateConnectorResponseBodyItem;
class CreateTriggerRequestTriggerInfo extends $tea.Model {
    static names() {
        return {
            description: 'description',
            dingConnectorId: 'dingConnectorId',
            inputSchema: 'inputSchema',
            integratorConnectorId: 'integratorConnectorId',
            integratorTriggerId: 'integratorTriggerId',
            name: 'name',
        };
    }
    static types() {
        return {
            description: 'string',
            dingConnectorId: 'string',
            inputSchema: 'string',
            integratorConnectorId: 'string',
            integratorTriggerId: 'string',
            name: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateTriggerRequestTriggerInfo = CreateTriggerRequestTriggerInfo;
class CreateTriggerResponseBodyItem extends $tea.Model {
    static names() {
        return {
            dingConnectorId: 'dingConnectorId',
            dingTriggerId: 'dingTriggerId',
            integratorConnectorId: 'integratorConnectorId',
            integratorTriggerId: 'integratorTriggerId',
            subErrCode: 'subErrCode',
            subErrMsg: 'subErrMsg',
            success: 'success',
        };
    }
    static types() {
        return {
            dingConnectorId: 'string',
            dingTriggerId: 'string',
            integratorConnectorId: 'string',
            integratorTriggerId: 'string',
            subErrCode: 'string',
            subErrMsg: 'string',
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateTriggerResponseBodyItem = CreateTriggerResponseBodyItem;
class GetActionDetailResponseBodyIntegrationConfigCategoryNames extends $tea.Model {
    static names() {
        return {
            value: 'value',
        };
    }
    static types() {
        return {
            value: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetActionDetailResponseBodyIntegrationConfigCategoryNames = GetActionDetailResponseBodyIntegrationConfigCategoryNames;
class GetActionDetailResponseBodyIntegrationConfigProps extends $tea.Model {
    static names() {
        return {
            key: 'key',
            value: 'value',
        };
    }
    static types() {
        return {
            key: 'string',
            value: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetActionDetailResponseBodyIntegrationConfigProps = GetActionDetailResponseBodyIntegrationConfigProps;
class GetActionDetailResponseBodyIntegrationConfig extends $tea.Model {
    static names() {
        return {
            categoryNames: 'categoryNames',
            entityName: 'entityName',
            props: 'props',
        };
    }
    static types() {
        return {
            categoryNames: { 'type': 'array', 'itemType': GetActionDetailResponseBodyIntegrationConfigCategoryNames },
            entityName: 'string',
            props: { 'type': 'array', 'itemType': GetActionDetailResponseBodyIntegrationConfigProps },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetActionDetailResponseBodyIntegrationConfig = GetActionDetailResponseBodyIntegrationConfig;
class PullDataByPageResponseBodyList extends $tea.Model {
    static names() {
        return {
            dataCreateAppId: 'dataCreateAppId',
            dataCreateAppType: 'dataCreateAppType',
            dataGmtCreate: 'dataGmtCreate',
            dataGmtModified: 'dataGmtModified',
            dataModifiedAppId: 'dataModifiedAppId',
            dataModifiedAppType: 'dataModifiedAppType',
            jsonData: 'jsonData',
        };
    }
    static types() {
        return {
            dataCreateAppId: 'string',
            dataCreateAppType: 'string',
            dataGmtCreate: 'number',
            dataGmtModified: 'number',
            dataModifiedAppId: 'string',
            dataModifiedAppType: 'string',
            jsonData: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PullDataByPageResponseBodyList = PullDataByPageResponseBodyList;
class SearchActionsResponseBodyList extends $tea.Model {
    static names() {
        return {
            authorityUrl: 'authorityUrl',
            authorized: 'authorized',
            connectAssetUri: 'connectAssetUri',
            connectorId: 'connectorId',
            description: 'description',
            icon: 'icon',
            id: 'id',
            integrationType: 'integrationType',
            name: 'name',
            providerCorpId: 'providerCorpId',
        };
    }
    static types() {
        return {
            authorityUrl: 'string',
            authorized: 'boolean',
            connectAssetUri: 'string',
            connectorId: 'string',
            description: 'string',
            icon: 'string',
            id: 'string',
            integrationType: 'string',
            name: 'string',
            providerCorpId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchActionsResponseBodyList = SearchActionsResponseBodyList;
class SearchConnectorsResponseBodyList extends $tea.Model {
    static names() {
        return {
            description: 'description',
            icon: 'icon',
            id: 'id',
            name: 'name',
            providerCorpId: 'providerCorpId',
        };
    }
    static types() {
        return {
            description: 'string',
            icon: 'string',
            id: 'string',
            name: 'string',
            providerCorpId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchConnectorsResponseBodyList = SearchConnectorsResponseBodyList;
class SyncDataRequestTriggerDataList extends $tea.Model {
    static names() {
        return {
            action: 'action',
            customTriggerId: 'customTriggerId',
            dataGmtCreate: 'dataGmtCreate',
            dataGmtModified: 'dataGmtModified',
            integrationObject: 'integrationObject',
            jsonData: 'jsonData',
            triggerCondition: 'triggerCondition',
            triggerId: 'triggerId',
        };
    }
    static types() {
        return {
            action: 'string',
            customTriggerId: 'string',
            dataGmtCreate: 'number',
            dataGmtModified: 'number',
            integrationObject: 'string',
            jsonData: 'string',
            triggerCondition: 'string',
            triggerId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncDataRequestTriggerDataList = SyncDataRequestTriggerDataList;
class SyncDataResponseBodyList extends $tea.Model {
    static names() {
        return {
            bizPrimaryKey: 'bizPrimaryKey',
            subErrCode: 'subErrCode',
            subErrMsg: 'subErrMsg',
            success: 'success',
            triggerId: 'triggerId',
        };
    }
    static types() {
        return {
            bizPrimaryKey: 'string',
            subErrCode: 'string',
            subErrMsg: 'string',
            success: 'boolean',
            triggerId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SyncDataResponseBodyList = SyncDataResponseBodyList;
class UpdateActionRequestActionInfoInputMappingConfig extends $tea.Model {
    static names() {
        return {
            customSchemaMapping: 'customSchemaMapping',
            rules: 'rules',
        };
    }
    static types() {
        return {
            customSchemaMapping: 'string',
            rules: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateActionRequestActionInfoInputMappingConfig = UpdateActionRequestActionInfoInputMappingConfig;
class UpdateActionRequestActionInfoOutputDataRules extends $tea.Model {
    static names() {
        return {
            expectValue: 'expectValue',
            operate: 'operate',
            propertyPath: 'propertyPath',
        };
    }
    static types() {
        return {
            expectValue: 'string',
            operate: 'string',
            propertyPath: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateActionRequestActionInfoOutputDataRules = UpdateActionRequestActionInfoOutputDataRules;
class UpdateActionRequestActionInfoOutputMappingConfig extends $tea.Model {
    static names() {
        return {
            customSchemaMapping: 'customSchemaMapping',
            rules: 'rules',
        };
    }
    static types() {
        return {
            customSchemaMapping: 'string',
            rules: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateActionRequestActionInfoOutputMappingConfig = UpdateActionRequestActionInfoOutputMappingConfig;
class UpdateActionRequestActionInfo extends $tea.Model {
    static names() {
        return {
            apiPath: 'apiPath',
            description: 'description',
            dingActionId: 'dingActionId',
            dingConnectorId: 'dingConnectorId',
            inputMappingConfig: 'inputMappingConfig',
            inputSchema: 'inputSchema',
            integratorActionId: 'integratorActionId',
            integratorConnectorId: 'integratorConnectorId',
            name: 'name',
            outputDataRules: 'outputDataRules',
            outputMappingConfig: 'outputMappingConfig',
            outputSchema: 'outputSchema',
        };
    }
    static types() {
        return {
            apiPath: 'string',
            description: 'string',
            dingActionId: 'string',
            dingConnectorId: 'string',
            inputMappingConfig: UpdateActionRequestActionInfoInputMappingConfig,
            inputSchema: 'string',
            integratorActionId: 'string',
            integratorConnectorId: 'string',
            name: 'string',
            outputDataRules: { 'type': 'array', 'itemType': UpdateActionRequestActionInfoOutputDataRules },
            outputMappingConfig: UpdateActionRequestActionInfoOutputMappingConfig,
            outputSchema: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateActionRequestActionInfo = UpdateActionRequestActionInfo;
class UpdateActionResponseBodyItem extends $tea.Model {
    static names() {
        return {
            dingActionId: 'dingActionId',
            dingConnectorId: 'dingConnectorId',
            integratorActionId: 'integratorActionId',
            integratorConnectorId: 'integratorConnectorId',
            subErrCode: 'subErrCode',
            subErrMsg: 'subErrMsg',
            success: 'success',
        };
    }
    static types() {
        return {
            dingActionId: 'string',
            dingConnectorId: 'string',
            integratorActionId: 'string',
            integratorConnectorId: 'string',
            subErrCode: 'string',
            subErrMsg: 'string',
            success: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateActionResponseBodyItem = UpdateActionResponseBodyItem;
class UpdateConnectorRequestConnectorInfo extends $tea.Model {
    static names() {
        return {
            apiDomain: 'apiDomain',
            apiSecret: 'apiSecret',
            appId: 'appId',
            authValueEnv: 'authValueEnv',
            description: 'description',
            dingConnectorId: 'dingConnectorId',
            domainEnv: 'domainEnv',
            iconMediaId: 'iconMediaId',
            integratorConnectorId: 'integratorConnectorId',
            name: 'name',
        };
    }
    static types() {
        return {
            apiDomain: 'string',
            apiSecret: 'string',
            appId: 'number',
            authValueEnv: 'boolean',
            description: 'string',
            dingConnectorId: 'string',
            domainEnv: 'boolean',
            iconMediaId: 'string',
            integratorConnectorId: 'string',
            name: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateConnectorRequestConnectorInfo = UpdateConnectorRequestConnectorInfo;
class UpdateConnectorResponseBodyItem extends $tea.Model {
    static names() {
        return {
            dingConnectorId: 'dingConnectorId',
            integratorConnectorId: 'integratorConnectorId',
            subErrCode: 'subErrCode',
            subErrMsg: 'subErrMsg',
            success: 'success',
        };
    }
    static types() {
        return {
            dingConnectorId: 'string',
            integratorConnectorId: 'string',
            subErrCode: 'string',
            subErrMsg: 'string',
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateConnectorResponseBodyItem = UpdateConnectorResponseBodyItem;
class UpdateTriggerRequestTriggerInfo extends $tea.Model {
    static names() {
        return {
            description: 'description',
            dingConnectorId: 'dingConnectorId',
            dingTriggerId: 'dingTriggerId',
            inputSchema: 'inputSchema',
            integratorConnectorId: 'integratorConnectorId',
            integratorTriggerId: 'integratorTriggerId',
            name: 'name',
        };
    }
    static types() {
        return {
            description: 'string',
            dingConnectorId: 'string',
            dingTriggerId: 'string',
            inputSchema: 'string',
            integratorConnectorId: 'string',
            integratorTriggerId: 'string',
            name: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateTriggerRequestTriggerInfo = UpdateTriggerRequestTriggerInfo;
class UpdateTriggerResponseBodyItem extends $tea.Model {
    static names() {
        return {
            dingConnectorId: 'dingConnectorId',
            dingTriggerId: 'dingTriggerId',
            integratorConnectorId: 'integratorConnectorId',
            integratorTriggerId: 'integratorTriggerId',
            subErrCode: 'subErrCode',
            subErrMsg: 'subErrMsg',
            success: 'success',
        };
    }
    static types() {
        return {
            dingConnectorId: 'string',
            dingTriggerId: 'string',
            integratorConnectorId: 'string',
            integratorTriggerId: 'string',
            subErrCode: 'string',
            subErrMsg: 'string',
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateTriggerResponseBodyItem = UpdateTriggerResponseBodyItem;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 
     *
     * @param request CreateActionRequest
     * @param headers CreateActionHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateActionResponse
     */
    async createActionWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.actionInfo)) {
            body["actionInfo"] = request.actionInfo;
        }
        if (!tea_util_1.default.isUnset(request.integratorFlag)) {
            body["integratorFlag"] = request.integratorFlag;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateAction",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/actions`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateActionResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CreateActionRequest
     * @return CreateActionResponse
     */
    async createAction(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateActionHeaders({});
        return await this.createActionWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CreateConnectorRequest
     * @param headers CreateConnectorHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateConnectorResponse
     */
    async createConnectorWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.connectorInfo)) {
            body["connectorInfo"] = request.connectorInfo;
        }
        if (!tea_util_1.default.isUnset(request.integratorFlag)) {
            body["integratorFlag"] = request.integratorFlag;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateConnector",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/connectors`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateConnectorResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CreateConnectorRequest
     * @return CreateConnectorResponse
     */
    async createConnector(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateConnectorHeaders({});
        return await this.createConnectorWithOptions(request, headers, runtime);
    }
    /**
     * @summary /
     *
     * @param request CreateInvocableInstanceRequest
     * @param headers CreateInvocableInstanceHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateInvocableInstanceResponse
     */
    async createInvocableInstanceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.connectAssetUri)) {
            body["connectAssetUri"] = request.connectAssetUri;
        }
        if (!tea_util_1.default.isUnset(request.instanceKey)) {
            body["instanceKey"] = request.instanceKey;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateInvocableInstance",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/instances`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateInvocableInstanceResponse({}));
    }
    /**
     * @summary /
     *
     * @param request CreateInvocableInstanceRequest
     * @return CreateInvocableInstanceResponse
     */
    async createInvocableInstance(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateInvocableInstanceHeaders({});
        return await this.createInvocableInstanceWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CreateTriggerRequest
     * @param headers CreateTriggerHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateTriggerResponse
     */
    async createTriggerWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.integratorFlag)) {
            body["integratorFlag"] = request.integratorFlag;
        }
        if (!tea_util_1.default.isUnset(request.triggerInfo)) {
            body["triggerInfo"] = request.triggerInfo;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateTrigger",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/triggers`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateTriggerResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CreateTriggerRequest
     * @return CreateTriggerResponse
     */
    async createTrigger(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateTriggerHeaders({});
        return await this.createTriggerWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetActionDetailRequest
     * @param headers GetActionDetailHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetActionDetailResponse
     */
    async getActionDetailWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.connectAssetUri)) {
            body["connectAssetUri"] = request.connectAssetUri;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "GetActionDetail",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/assets/actions/details/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetActionDetailResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetActionDetailRequest
     * @return GetActionDetailResponse
     */
    async getActionDetail(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetActionDetailHeaders({});
        return await this.getActionDetailWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request InvokeInstanceRequest
     * @param headers InvokeInstanceHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return InvokeInstanceResponse
     */
    async invokeInstanceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.connectAssetUri)) {
            body["connectAssetUri"] = request.connectAssetUri;
        }
        if (!tea_util_1.default.isUnset(request.inputJsonString)) {
            body["inputJsonString"] = request.inputJsonString;
        }
        if (!tea_util_1.default.isUnset(request.instanceKey)) {
            body["instanceKey"] = request.instanceKey;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "InvokeInstance",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/instances/invoke`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new InvokeInstanceResponse({}));
    }
    /**
     * @summary 
     *
     * @param request InvokeInstanceRequest
     * @return InvokeInstanceResponse
     */
    async invokeInstance(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new InvokeInstanceHeaders({});
        return await this.invokeInstanceWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request PullDataByPageRequest
     * @param headers PullDataByPageHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return PullDataByPageResponse
     */
    async pullDataByPageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.appId)) {
            query["appId"] = request.appId;
        }
        if (!tea_util_1.default.isUnset(request.dataModelId)) {
            query["dataModelId"] = request.dataModelId;
        }
        if (!tea_util_1.default.isUnset(request.datetimeFilterField)) {
            query["datetimeFilterField"] = request.datetimeFilterField;
        }
        if (!tea_util_1.default.isUnset(request.maxDatetime)) {
            query["maxDatetime"] = request.maxDatetime;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.minDatetime)) {
            query["minDatetime"] = request.minDatetime;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "PullDataByPage",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/data`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PullDataByPageResponse({}));
    }
    /**
     * @summary 
     *
     * @param request PullDataByPageRequest
     * @return PullDataByPageResponse
     */
    async pullDataByPage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PullDataByPageHeaders({});
        return await this.pullDataByPageWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request PullDataByPkRequest
     * @param headers PullDataByPkHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return PullDataByPkResponse
     */
    async pullDataByPkWithOptions(dataModelId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.appId)) {
            query["appId"] = request.appId;
        }
        if (!tea_util_1.default.isUnset(request.primaryKey)) {
            query["primaryKey"] = request.primaryKey;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "PullDataByPk",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/data/${dataModelId}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PullDataByPkResponse({}));
    }
    /**
     * @summary 
     *
     * @param request PullDataByPkRequest
     * @return PullDataByPkResponse
     */
    async pullDataByPk(dataModelId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PullDataByPkHeaders({});
        return await this.pullDataByPkWithOptions(dataModelId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request SearchActionsRequest
     * @param headers SearchActionsHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SearchActionsResponse
     */
    async searchActionsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.connectorId)) {
            body["connectorId"] = request.connectorId;
        }
        if (!tea_util_1.default.isUnset(request.connectorProviderCorpId)) {
            body["connectorProviderCorpId"] = request.connectorProviderCorpId;
        }
        if (!tea_util_1.default.isUnset(request.integrationTypes)) {
            body["integrationTypes"] = request.integrationTypes;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            body["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            body["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SearchActions",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/assets/actions/search`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SearchActionsResponse({}));
    }
    /**
     * @summary 
     *
     * @param request SearchActionsRequest
     * @return SearchActionsResponse
     */
    async searchActions(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SearchActionsHeaders({});
        return await this.searchActionsWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request SearchConnectorsRequest
     * @param headers SearchConnectorsHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SearchConnectorsResponse
     */
    async searchConnectorsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.type)) {
            query["type"] = request.type;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "SearchConnectors",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/assets/connectors`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SearchConnectorsResponse({}));
    }
    /**
     * @summary 
     *
     * @param request SearchConnectorsRequest
     * @return SearchConnectorsResponse
     */
    async searchConnectors(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SearchConnectorsHeaders({});
        return await this.searchConnectorsWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request SyncDataRequest
     * @param headers SyncDataHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SyncDataResponse
     */
    async syncDataWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appId)) {
            body["appId"] = request.appId;
        }
        if (!tea_util_1.default.isUnset(request.triggerDataList)) {
            body["triggerDataList"] = request.triggerDataList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SyncData",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/triggers/data/sync`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SyncDataResponse({}));
    }
    /**
     * @summary 
     *
     * @param request SyncDataRequest
     * @return SyncDataResponse
     */
    async syncData(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SyncDataHeaders({});
        return await this.syncDataWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateActionRequest
     * @param headers UpdateActionHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateActionResponse
     */
    async updateActionWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.actionInfo)) {
            body["actionInfo"] = request.actionInfo;
        }
        if (!tea_util_1.default.isUnset(request.integratorFlag)) {
            body["integratorFlag"] = request.integratorFlag;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateAction",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/actions`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateActionResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateActionRequest
     * @return UpdateActionResponse
     */
    async updateAction(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateActionHeaders({});
        return await this.updateActionWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateConnectorRequest
     * @param headers UpdateConnectorHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateConnectorResponse
     */
    async updateConnectorWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.connectorInfo)) {
            body["connectorInfo"] = request.connectorInfo;
        }
        if (!tea_util_1.default.isUnset(request.integratorFlag)) {
            body["integratorFlag"] = request.integratorFlag;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateConnector",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/connectors`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateConnectorResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateConnectorRequest
     * @return UpdateConnectorResponse
     */
    async updateConnector(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateConnectorHeaders({});
        return await this.updateConnectorWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateTriggerRequest
     * @param headers UpdateTriggerHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateTriggerResponse
     */
    async updateTriggerWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.integratorFlag)) {
            body["integratorFlag"] = request.integratorFlag;
        }
        if (!tea_util_1.default.isUnset(request.triggerInfo)) {
            body["triggerInfo"] = request.triggerInfo;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateTrigger",
            version: "connector_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/connector/triggers`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateTriggerResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateTriggerRequest
     * @return UpdateTriggerResponse
     */
    async updateTrigger(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateTriggerHeaders({});
        return await this.updateTriggerWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
