// @alicloud/dingtalk@2.1.16/dist/blackboard_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryBlackboardReadUnReadResponseBodyUsers = exports.QueryBlackboardSpaceResponse = exports.QueryBlackboardSpaceResponseBody = exports.QueryBlackboardSpaceRequest = exports.QueryBlackboardSpaceHeaders = exports.QueryBlackboardReadUnReadResponse = exports.QueryBlackboardReadUnReadResponseBody = exports.QueryBlackboardReadUnReadRequest = exports.QueryBlackboardReadUnReadHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class QueryBlackboardReadUnReadHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBlackboardReadUnReadHeaders = QueryBlackboardReadUnReadHeaders;
class QueryBlackboardReadUnReadRequest extends $tea.Model {
    static names() {
        return {
            blackboardId: 'blackboardId',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            operationUserId: 'operationUserId',
        };
    }
    static types() {
        return {
            blackboardId: 'string',
            maxResults: 'number',
            nextToken: 'string',
            operationUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBlackboardReadUnReadRequest = QueryBlackboardReadUnReadRequest;
class QueryBlackboardReadUnReadResponseBody extends $tea.Model {
    static names() {
        return {
            nextToken: 'nextToken',
            users: 'users',
        };
    }
    static types() {
        return {
            nextToken: 'string',
            users: { 'type': 'array', 'itemType': QueryBlackboardReadUnReadResponseBodyUsers },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBlackboardReadUnReadResponseBody = QueryBlackboardReadUnReadResponseBody;
class QueryBlackboardReadUnReadResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryBlackboardReadUnReadResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBlackboardReadUnReadResponse = QueryBlackboardReadUnReadResponse;
class QueryBlackboardSpaceHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBlackboardSpaceHeaders = QueryBlackboardSpaceHeaders;
class QueryBlackboardSpaceRequest extends $tea.Model {
    static names() {
        return {
            operationUserId: 'operationUserId',
        };
    }
    static types() {
        return {
            operationUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBlackboardSpaceRequest = QueryBlackboardSpaceRequest;
class QueryBlackboardSpaceResponseBody extends $tea.Model {
    static names() {
        return {
            spaceId: 'spaceId',
        };
    }
    static types() {
        return {
            spaceId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBlackboardSpaceResponseBody = QueryBlackboardSpaceResponseBody;
class QueryBlackboardSpaceResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryBlackboardSpaceResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBlackboardSpaceResponse = QueryBlackboardSpaceResponse;
class QueryBlackboardReadUnReadResponseBodyUsers extends $tea.Model {
    static names() {
        return {
            read: 'read',
            readTimestamp: 'readTimestamp',
            userId: 'userId',
        };
    }
    static types() {
        return {
            read: 'string',
            readTimestamp: 'number',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBlackboardReadUnReadResponseBodyUsers = QueryBlackboardReadUnReadResponseBodyUsers;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 查询公告已读未读人员列表
     *
     * @param request QueryBlackboardReadUnReadRequest
     * @param headers QueryBlackboardReadUnReadHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryBlackboardReadUnReadResponse
     */
    async queryBlackboardReadUnReadWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.blackboardId)) {
            query["blackboardId"] = request.blackboardId;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.operationUserId)) {
            query["operationUserId"] = request.operationUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryBlackboardReadUnRead",
            version: "blackboard_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/blackboard/readers`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryBlackboardReadUnReadResponse({}));
    }
    /**
     * @summary 查询公告已读未读人员列表
     *
     * @param request QueryBlackboardReadUnReadRequest
     * @return QueryBlackboardReadUnReadResponse
     */
    async queryBlackboardReadUnRead(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryBlackboardReadUnReadHeaders({});
        return await this.queryBlackboardReadUnReadWithOptions(request, headers, runtime);
    }
    /**
     * @summary 获取公告钉盘空间信息
     *
     * @param request QueryBlackboardSpaceRequest
     * @param headers QueryBlackboardSpaceHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return QueryBlackboardSpaceResponse
     */
    async queryBlackboardSpaceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.operationUserId)) {
            query["operationUserId"] = request.operationUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryBlackboardSpace",
            version: "blackboard_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/blackboard/spaces`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryBlackboardSpaceResponse({}));
    }
    /**
     * @summary 获取公告钉盘空间信息
     *
     * @param request QueryBlackboardSpaceRequest
     * @return QueryBlackboardSpaceResponse
     */
    async queryBlackboardSpace(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryBlackboardSpaceHeaders({});
        return await this.queryBlackboardSpaceWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
