// @alicloud/dingtalk@2.1.16/dist/micro_app_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAppRoleScopeByRoleIdHeaders = exports.GetAppResourceUseInfoResponse = exports.GetAppResourceUseInfoRequest = exports.GetAppResourceUseInfoHeaders = exports.GetApaasAppResponse = exports.GetApaasAppResponseBody = exports.GetApaasAppHeaders = exports.DeleteInnerAppResponse = exports.DeleteInnerAppResponseBody = exports.DeleteInnerAppRequest = exports.DeleteInnerAppHeaders = exports.DeleteAppRoleResponse = exports.DeleteAppRoleResponseBody = exports.DeleteAppRoleRequest = exports.DeleteAppRoleHeaders = exports.CreateInnerAppResponse = exports.CreateInnerAppResponseBody = exports.CreateInnerAppRequest = exports.CreateInnerAppHeaders = exports.CreateApaasAppResponse = exports.CreateApaasAppResponseBody = exports.CreateApaasAppRequest = exports.CreateApaasAppHeaders = exports.AyunTestOnlineResponse = exports.AyunTestOnlineResponseBody = exports.AyunTestResponse = exports.AyunTestResponseBody = exports.AppStatusManagerTestResponse = exports.AppStatusManagerTestResponseBody = exports.AppStatusManagerTestRequest = exports.AnheiTestNineResponse = exports.AnheiTestNineResponseBody = exports.AnheiTestBResponse = exports.AnheiTestBResponseBody = exports.AnheiTest888Response = exports.AnheiTest888ResponseBody = exports.AnheiPResponse = exports.AnheiPResponseBody = exports.AddMemberToAppRoleResponse = exports.AddMemberToAppRoleResponseBody = exports.AddMemberToAppRoleRequest = exports.AddMemberToAppRoleHeaders = exports.AddAppToWorkBenchGroupResponse = exports.AddAppToWorkBenchGroupResponseBody = exports.AddAppToWorkBenchGroupRequest = exports.AddAppToWorkBenchGroupHeaders = exports.AddAppRolesToMemberResponse = exports.AddAppRolesToMemberResponseBody = exports.AddAppRolesToMemberRequest = exports.AddAppRolesToMemberHeaders = void 0;
exports.RebuildRoleScopeForAppRoleResponse = exports.RebuildRoleScopeForAppRoleResponseBody = exports.RebuildRoleScopeForAppRoleRequest = exports.RebuildRoleScopeForAppRoleHeaders = exports.PublishInnerAppVersionResponse = exports.PublishInnerAppVersionResponseBody = exports.PublishInnerAppVersionRequest = exports.PublishInnerAppVersionHeaders = exports.PageInnerAppHistoryVersionResponse = exports.PageInnerAppHistoryVersionResponseBody = exports.PageInnerAppHistoryVersionRequest = exports.PageInnerAppHistoryVersionHeaders = exports.ListUserVilebleAppResponse = exports.ListUserVilebleAppResponseBody = exports.ListUserVilebleAppHeaders = exports.ListRoleInfoByUserResponse = exports.ListRoleInfoByUserResponseBody = exports.ListRoleInfoByUserHeaders = exports.ListInnerAppVersionResponse = exports.ListInnerAppVersionResponseBody = exports.ListInnerAppVersionHeaders = exports.ListInnerAppResponse = exports.ListInnerAppResponseBody = exports.ListInnerAppRequest = exports.ListInnerAppHeaders = exports.ListAppRoleScopesResponse = exports.ListAppRoleScopesResponseBody = exports.ListAppRoleScopesRequest = exports.ListAppRoleScopesHeaders = exports.ListAllInnerAppsResponse = exports.ListAllInnerAppsResponseBody = exports.ListAllInnerAppsHeaders = exports.ListAllAppResponse = exports.ListAllAppResponseBody = exports.ListAllAppHeaders = exports.GetUserAppDevAccessResponse = exports.GetUserAppDevAccessResponseBody = exports.GetUserAppDevAccessHeaders = exports.GetMicroAppUserAccessResponse = exports.GetMicroAppUserAccessResponseBody = exports.GetMicroAppUserAccessHeaders = exports.GetMicroAppScopeResponse = exports.GetMicroAppScopeResponseBody = exports.GetMicroAppScopeHeaders = exports.GetInnerAppResponse = exports.GetInnerAppResponseBody = exports.GetInnerAppRequest = exports.GetInnerAppHeaders = exports.GetAppRoleScopeByRoleIdResponse = exports.GetAppRoleScopeByRoleIdResponseBody = void 0;
exports.PageInnerAppHistoryVersionResponseBodyMiniAppVersionList = exports.ListUserVilebleAppResponseBodyAppList = exports.ListRoleInfoByUserResponseBodyResult = exports.ListInnerAppVersionResponseBodyAppVersionList = exports.ListInnerAppResponseBodyAppList = exports.ListAppRoleScopesResponseBodyDataList = exports.ListAllInnerAppsResponseBodyAppList = exports.ListAllAppResponseBodyAppList = exports.GetMicroAppScopeResponseBodyResult = exports.GetAppResourceUseInfoResponseBody = exports.AddAppRolesToMemberResponseBodyResult = exports.AddAppRolesToMemberRequestRoleList = exports.UpdateInnerAppResponse = exports.UpdateInnerAppResponseBody = exports.UpdateInnerAppRequest = exports.UpdateInnerAppHeaders = exports.UpdateAppRoleInfoResponse = exports.UpdateAppRoleInfoResponseBody = exports.UpdateAppRoleInfoRequest = exports.UpdateAppRoleInfoHeaders = exports.UpdateApaasAppResponse = exports.UpdateApaasAppResponseBody = exports.UpdateApaasAppRequest = exports.UpdateApaasAppHeaders = exports.SetMicroAppScopeResponse = exports.SetMicroAppScopeResponseBody = exports.SetMicroAppScopeRequest = exports.SetMicroAppScopeHeaders = exports.RollbackInnerAppVersionResponse = exports.RollbackInnerAppVersionResponseBody = exports.RollbackInnerAppVersionRequest = exports.RollbackInnerAppVersionHeaders = exports.RemoveMemberForAppRoleResponse = exports.RemoveMemberForAppRoleResponseBody = exports.RemoveMemberForAppRoleRequest = exports.RemoveMemberForAppRoleHeaders = exports.RemoveApaasAppResponse = exports.RemoveApaasAppResponseBody = exports.RemoveApaasAppRequest = exports.RemoveApaasAppHeaders = exports.RegisterCustomAppRoleResponse = exports.RegisterCustomAppRoleResponseBody = exports.RegisterCustomAppRoleRequest = exports.RegisterCustomAppRoleHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AddAppRolesToMemberHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppRolesToMemberHeaders = AddAppRolesToMemberHeaders;
class AddAppRolesToMemberRequest extends $tea.Model {
    static names() {
        return {
            memberId: 'memberId',
            memberType: 'memberType',
            opUserId: 'opUserId',
            roleList: 'roleList',
        };
    }
    static types() {
        return {
            memberId: 'string',
            memberType: 'string',
            opUserId: 'string',
            roleList: { 'type': 'array', 'itemType': AddAppRolesToMemberRequestRoleList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppRolesToMemberRequest = AddAppRolesToMemberRequest;
class AddAppRolesToMemberResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': AddAppRolesToMemberResponseBodyResult },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppRolesToMemberResponseBody = AddAppRolesToMemberResponseBody;
class AddAppRolesToMemberResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddAppRolesToMemberResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppRolesToMemberResponse = AddAppRolesToMemberResponse;
class AddAppToWorkBenchGroupHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppToWorkBenchGroupHeaders = AddAppToWorkBenchGroupHeaders;
class AddAppToWorkBenchGroupRequest extends $tea.Model {
    static names() {
        return {
            componentId: 'componentId',
            ecologicalCorpId: 'ecologicalCorpId',
            opUnionId: 'opUnionId',
        };
    }
    static types() {
        return {
            componentId: 'string',
            ecologicalCorpId: 'string',
            opUnionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppToWorkBenchGroupRequest = AddAppToWorkBenchGroupRequest;
class AddAppToWorkBenchGroupResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppToWorkBenchGroupResponseBody = AddAppToWorkBenchGroupResponseBody;
class AddAppToWorkBenchGroupResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddAppToWorkBenchGroupResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppToWorkBenchGroupResponse = AddAppToWorkBenchGroupResponse;
class AddMemberToAppRoleHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddMemberToAppRoleHeaders = AddMemberToAppRoleHeaders;
class AddMemberToAppRoleRequest extends $tea.Model {
    static names() {
        return {
            deptIdList: 'deptIdList',
            opUserId: 'opUserId',
            scopeVersion: 'scopeVersion',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            deptIdList: { 'type': 'array', 'itemType': 'number' },
            opUserId: 'string',
            scopeVersion: 'number',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddMemberToAppRoleRequest = AddMemberToAppRoleRequest;
class AddMemberToAppRoleResponseBody extends $tea.Model {
    static names() {
        return {
            latestScopeVersion: 'latestScopeVersion',
        };
    }
    static types() {
        return {
            latestScopeVersion: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddMemberToAppRoleResponseBody = AddMemberToAppRoleResponseBody;
class AddMemberToAppRoleResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddMemberToAppRoleResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddMemberToAppRoleResponse = AddMemberToAppRoleResponse;
class AnheiPResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AnheiPResponseBody = AnheiPResponseBody;
class AnheiPResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AnheiPResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AnheiPResponse = AnheiPResponse;
class AnheiTest888ResponseBody extends $tea.Model {
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AnheiTest888ResponseBody = AnheiTest888ResponseBody;
class AnheiTest888Response extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AnheiTest888ResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AnheiTest888Response = AnheiTest888Response;
class AnheiTestBResponseBody extends $tea.Model {
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AnheiTestBResponseBody = AnheiTestBResponseBody;
class AnheiTestBResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AnheiTestBResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AnheiTestBResponse = AnheiTestBResponse;
class AnheiTestNineResponseBody extends $tea.Model {
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AnheiTestNineResponseBody = AnheiTestNineResponseBody;
class AnheiTestNineResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AnheiTestNineResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AnheiTestNineResponse = AnheiTestNineResponse;
class AppStatusManagerTestRequest extends $tea.Model {
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AppStatusManagerTestRequest = AppStatusManagerTestRequest;
class AppStatusManagerTestResponseBody extends $tea.Model {
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AppStatusManagerTestResponseBody = AppStatusManagerTestResponseBody;
class AppStatusManagerTestResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AppStatusManagerTestResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AppStatusManagerTestResponse = AppStatusManagerTestResponse;
class AyunTestResponseBody extends $tea.Model {
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AyunTestResponseBody = AyunTestResponseBody;
class AyunTestResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AyunTestResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AyunTestResponse = AyunTestResponse;
class AyunTestOnlineResponseBody extends $tea.Model {
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AyunTestOnlineResponseBody = AyunTestOnlineResponseBody;
class AyunTestOnlineResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AyunTestOnlineResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AyunTestOnlineResponse = AyunTestOnlineResponse;
class CreateApaasAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateApaasAppHeaders = CreateApaasAppHeaders;
class CreateApaasAppRequest extends $tea.Model {
    static names() {
        return {
            appDesc: 'appDesc',
            appIcon: 'appIcon',
            appName: 'appName',
            bizAppId: 'bizAppId',
            homepageEditLink: 'homepageEditLink',
            homepageLink: 'homepageLink',
            isShortCut: 'isShortCut',
            ompLink: 'ompLink',
            opUserId: 'opUserId',
            pcHomepageEditLink: 'pcHomepageEditLink',
            pcHomepageLink: 'pcHomepageLink',
            templateKey: 'templateKey',
        };
    }
    static types() {
        return {
            appDesc: 'string',
            appIcon: 'string',
            appName: 'string',
            bizAppId: 'string',
            homepageEditLink: 'string',
            homepageLink: 'string',
            isShortCut: 'number',
            ompLink: 'string',
            opUserId: 'string',
            pcHomepageEditLink: 'string',
            pcHomepageLink: 'string',
            templateKey: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateApaasAppRequest = CreateApaasAppRequest;
class CreateApaasAppResponseBody extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
            bizAppId: 'bizAppId',
        };
    }
    static types() {
        return {
            agentId: 'number',
            bizAppId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateApaasAppResponseBody = CreateApaasAppResponseBody;
class CreateApaasAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateApaasAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateApaasAppResponse = CreateApaasAppResponse;
class CreateInnerAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateInnerAppHeaders = CreateInnerAppHeaders;
class CreateInnerAppRequest extends $tea.Model {
    static names() {
        return {
            desc: 'desc',
            developType: 'developType',
            homepageLink: 'homepageLink',
            icon: 'icon',
            ipWhiteList: 'ipWhiteList',
            name: 'name',
            ompLink: 'ompLink',
            opUnionId: 'opUnionId',
            pcHomepageLink: 'pcHomepageLink',
            scopeType: 'scopeType',
        };
    }
    static types() {
        return {
            desc: 'string',
            developType: 'number',
            homepageLink: 'string',
            icon: 'string',
            ipWhiteList: { 'type': 'array', 'itemType': 'string' },
            name: 'string',
            ompLink: 'string',
            opUnionId: 'string',
            pcHomepageLink: 'string',
            scopeType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateInnerAppRequest = CreateInnerAppRequest;
class CreateInnerAppResponseBody extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
            appKey: 'appKey',
            appSecret: 'appSecret',
        };
    }
    static types() {
        return {
            agentId: 'number',
            appKey: 'string',
            appSecret: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateInnerAppResponseBody = CreateInnerAppResponseBody;
class CreateInnerAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateInnerAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateInnerAppResponse = CreateInnerAppResponse;
class DeleteAppRoleHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteAppRoleHeaders = DeleteAppRoleHeaders;
class DeleteAppRoleRequest extends $tea.Model {
    static names() {
        return {
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            opUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteAppRoleRequest = DeleteAppRoleRequest;
class DeleteAppRoleResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteAppRoleResponseBody = DeleteAppRoleResponseBody;
class DeleteAppRoleResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteAppRoleResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteAppRoleResponse = DeleteAppRoleResponse;
class DeleteInnerAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteInnerAppHeaders = DeleteInnerAppHeaders;
class DeleteInnerAppRequest extends $tea.Model {
    static names() {
        return {
            opUnionId: 'opUnionId',
        };
    }
    static types() {
        return {
            opUnionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteInnerAppRequest = DeleteInnerAppRequest;
class DeleteInnerAppResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteInnerAppResponseBody = DeleteInnerAppResponseBody;
class DeleteInnerAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteInnerAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteInnerAppResponse = DeleteInnerAppResponse;
class GetApaasAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetApaasAppHeaders = GetApaasAppHeaders;
class GetApaasAppResponseBody extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
            bizAppId: 'bizAppId',
            publishStatus: 'publishStatus',
        };
    }
    static types() {
        return {
            agentId: 'number',
            bizAppId: 'string',
            publishStatus: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetApaasAppResponseBody = GetApaasAppResponseBody;
class GetApaasAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetApaasAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetApaasAppResponse = GetApaasAppResponse;
class GetAppResourceUseInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAppResourceUseInfoHeaders = GetAppResourceUseInfoHeaders;
class GetAppResourceUseInfoRequest extends $tea.Model {
    static names() {
        return {
            benefitCode: 'benefitCode',
            endTime: 'endTime',
            periodType: 'periodType',
            startTime: 'startTime',
        };
    }
    static types() {
        return {
            benefitCode: 'string',
            endTime: 'string',
            periodType: 'string',
            startTime: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAppResourceUseInfoRequest = GetAppResourceUseInfoRequest;
class GetAppResourceUseInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: { 'type': 'array', 'itemType': GetAppResourceUseInfoResponseBody },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAppResourceUseInfoResponse = GetAppResourceUseInfoResponse;
class GetAppRoleScopeByRoleIdHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAppRoleScopeByRoleIdHeaders = GetAppRoleScopeByRoleIdHeaders;
class GetAppRoleScopeByRoleIdResponseBody extends $tea.Model {
    static names() {
        return {
            canManageRole: 'canManageRole',
            deptIdList: 'deptIdList',
            roleId: 'roleId',
            roleName: 'roleName',
            scopeType: 'scopeType',
            scopeVersion: 'scopeVersion',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            canManageRole: 'boolean',
            deptIdList: { 'type': 'array', 'itemType': 'number' },
            roleId: 'number',
            roleName: 'string',
            scopeType: 'string',
            scopeVersion: 'string',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAppRoleScopeByRoleIdResponseBody = GetAppRoleScopeByRoleIdResponseBody;
class GetAppRoleScopeByRoleIdResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetAppRoleScopeByRoleIdResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAppRoleScopeByRoleIdResponse = GetAppRoleScopeByRoleIdResponse;
class GetInnerAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetInnerAppHeaders = GetInnerAppHeaders;
class GetInnerAppRequest extends $tea.Model {
    static names() {
        return {
            ecologicalCorpId: 'ecologicalCorpId',
            opUnionId: 'opUnionId',
        };
    }
    static types() {
        return {
            ecologicalCorpId: 'string',
            opUnionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetInnerAppRequest = GetInnerAppRequest;
class GetInnerAppResponseBody extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
            appKey: 'appKey',
            appSecret: 'appSecret',
            desc: 'desc',
            homepageLink: 'homepageLink',
            icon: 'icon',
            ipWhiteList: 'ipWhiteList',
            name: 'name',
            ompLink: 'ompLink',
            pcHomepageLink: 'pcHomepageLink',
        };
    }
    static types() {
        return {
            agentId: 'number',
            appKey: 'string',
            appSecret: 'string',
            desc: 'string',
            homepageLink: 'string',
            icon: 'string',
            ipWhiteList: { 'type': 'array', 'itemType': 'string' },
            name: 'string',
            ompLink: 'string',
            pcHomepageLink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetInnerAppResponseBody = GetInnerAppResponseBody;
class GetInnerAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetInnerAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetInnerAppResponse = GetInnerAppResponse;
class GetMicroAppScopeHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMicroAppScopeHeaders = GetMicroAppScopeHeaders;
class GetMicroAppScopeResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetMicroAppScopeResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMicroAppScopeResponseBody = GetMicroAppScopeResponseBody;
class GetMicroAppScopeResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetMicroAppScopeResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMicroAppScopeResponse = GetMicroAppScopeResponse;
class GetMicroAppUserAccessHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMicroAppUserAccessHeaders = GetMicroAppUserAccessHeaders;
class GetMicroAppUserAccessResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMicroAppUserAccessResponseBody = GetMicroAppUserAccessResponseBody;
class GetMicroAppUserAccessResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetMicroAppUserAccessResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMicroAppUserAccessResponse = GetMicroAppUserAccessResponse;
class GetUserAppDevAccessHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAppDevAccessHeaders = GetUserAppDevAccessHeaders;
class GetUserAppDevAccessResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAppDevAccessResponseBody = GetUserAppDevAccessResponseBody;
class GetUserAppDevAccessResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetUserAppDevAccessResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAppDevAccessResponse = GetUserAppDevAccessResponse;
class ListAllAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAllAppHeaders = ListAllAppHeaders;
class ListAllAppResponseBody extends $tea.Model {
    static names() {
        return {
            appList: 'appList',
        };
    }
    static types() {
        return {
            appList: { 'type': 'array', 'itemType': ListAllAppResponseBodyAppList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAllAppResponseBody = ListAllAppResponseBody;
class ListAllAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListAllAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAllAppResponse = ListAllAppResponse;
class ListAllInnerAppsHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAllInnerAppsHeaders = ListAllInnerAppsHeaders;
class ListAllInnerAppsResponseBody extends $tea.Model {
    static names() {
        return {
            appList: 'appList',
        };
    }
    static types() {
        return {
            appList: { 'type': 'array', 'itemType': ListAllInnerAppsResponseBodyAppList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAllInnerAppsResponseBody = ListAllInnerAppsResponseBody;
class ListAllInnerAppsResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListAllInnerAppsResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAllInnerAppsResponse = ListAllInnerAppsResponse;
class ListAppRoleScopesHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAppRoleScopesHeaders = ListAppRoleScopesHeaders;
class ListAppRoleScopesRequest extends $tea.Model {
    static names() {
        return {
            nextToken: 'nextToken',
            size: 'size',
        };
    }
    static types() {
        return {
            nextToken: 'number',
            size: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAppRoleScopesRequest = ListAppRoleScopesRequest;
class ListAppRoleScopesResponseBody extends $tea.Model {
    static names() {
        return {
            dataList: 'dataList',
            hasMore: 'hasMore',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            dataList: { 'type': 'array', 'itemType': ListAppRoleScopesResponseBodyDataList },
            hasMore: 'boolean',
            nextToken: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAppRoleScopesResponseBody = ListAppRoleScopesResponseBody;
class ListAppRoleScopesResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListAppRoleScopesResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAppRoleScopesResponse = ListAppRoleScopesResponse;
class ListInnerAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListInnerAppHeaders = ListInnerAppHeaders;
class ListInnerAppRequest extends $tea.Model {
    static names() {
        return {
            ecologicalCorpId: 'ecologicalCorpId',
        };
    }
    static types() {
        return {
            ecologicalCorpId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListInnerAppRequest = ListInnerAppRequest;
class ListInnerAppResponseBody extends $tea.Model {
    static names() {
        return {
            appList: 'appList',
        };
    }
    static types() {
        return {
            appList: { 'type': 'array', 'itemType': ListInnerAppResponseBodyAppList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListInnerAppResponseBody = ListInnerAppResponseBody;
class ListInnerAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListInnerAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListInnerAppResponse = ListInnerAppResponse;
class ListInnerAppVersionHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListInnerAppVersionHeaders = ListInnerAppVersionHeaders;
class ListInnerAppVersionResponseBody extends $tea.Model {
    static names() {
        return {
            appVersionList: 'appVersionList',
        };
    }
    static types() {
        return {
            appVersionList: { 'type': 'array', 'itemType': ListInnerAppVersionResponseBodyAppVersionList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListInnerAppVersionResponseBody = ListInnerAppVersionResponseBody;
class ListInnerAppVersionResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListInnerAppVersionResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListInnerAppVersionResponse = ListInnerAppVersionResponse;
class ListRoleInfoByUserHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListRoleInfoByUserHeaders = ListRoleInfoByUserHeaders;
class ListRoleInfoByUserResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': ListRoleInfoByUserResponseBodyResult },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListRoleInfoByUserResponseBody = ListRoleInfoByUserResponseBody;
class ListRoleInfoByUserResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListRoleInfoByUserResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListRoleInfoByUserResponse = ListRoleInfoByUserResponse;
class ListUserVilebleAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUserVilebleAppHeaders = ListUserVilebleAppHeaders;
class ListUserVilebleAppResponseBody extends $tea.Model {
    static names() {
        return {
            appList: 'appList',
        };
    }
    static types() {
        return {
            appList: { 'type': 'array', 'itemType': ListUserVilebleAppResponseBodyAppList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUserVilebleAppResponseBody = ListUserVilebleAppResponseBody;
class ListUserVilebleAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListUserVilebleAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUserVilebleAppResponse = ListUserVilebleAppResponse;
class PageInnerAppHistoryVersionHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageInnerAppHistoryVersionHeaders = PageInnerAppHistoryVersionHeaders;
class PageInnerAppHistoryVersionRequest extends $tea.Model {
    static names() {
        return {
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
        };
    }
    static types() {
        return {
            pageNumber: 'number',
            pageSize: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageInnerAppHistoryVersionRequest = PageInnerAppHistoryVersionRequest;
class PageInnerAppHistoryVersionResponseBody extends $tea.Model {
    static names() {
        return {
            miniAppVersionList: 'miniAppVersionList',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            miniAppVersionList: { 'type': 'array', 'itemType': PageInnerAppHistoryVersionResponseBodyMiniAppVersionList },
            totalCount: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageInnerAppHistoryVersionResponseBody = PageInnerAppHistoryVersionResponseBody;
class PageInnerAppHistoryVersionResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PageInnerAppHistoryVersionResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageInnerAppHistoryVersionResponse = PageInnerAppHistoryVersionResponse;
class PublishInnerAppVersionHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PublishInnerAppVersionHeaders = PublishInnerAppVersionHeaders;
class PublishInnerAppVersionRequest extends $tea.Model {
    static names() {
        return {
            appVersionId: 'appVersionId',
            miniAppOnPc: 'miniAppOnPc',
            opUnionId: 'opUnionId',
            publishType: 'publishType',
        };
    }
    static types() {
        return {
            appVersionId: 'number',
            miniAppOnPc: 'boolean',
            opUnionId: 'string',
            publishType: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PublishInnerAppVersionRequest = PublishInnerAppVersionRequest;
class PublishInnerAppVersionResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PublishInnerAppVersionResponseBody = PublishInnerAppVersionResponseBody;
class PublishInnerAppVersionResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PublishInnerAppVersionResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PublishInnerAppVersionResponse = PublishInnerAppVersionResponse;
class RebuildRoleScopeForAppRoleHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RebuildRoleScopeForAppRoleHeaders = RebuildRoleScopeForAppRoleHeaders;
class RebuildRoleScopeForAppRoleRequest extends $tea.Model {
    static names() {
        return {
            deptIdList: 'deptIdList',
            opUserId: 'opUserId',
            scopeType: 'scopeType',
            scopeVersion: 'scopeVersion',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            deptIdList: { 'type': 'array', 'itemType': 'number' },
            opUserId: 'string',
            scopeType: 'string',
            scopeVersion: 'number',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RebuildRoleScopeForAppRoleRequest = RebuildRoleScopeForAppRoleRequest;
class RebuildRoleScopeForAppRoleResponseBody extends $tea.Model {
    static names() {
        return {
            latestScopeVersion: 'latestScopeVersion',
        };
    }
    static types() {
        return {
            latestScopeVersion: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RebuildRoleScopeForAppRoleResponseBody = RebuildRoleScopeForAppRoleResponseBody;
class RebuildRoleScopeForAppRoleResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RebuildRoleScopeForAppRoleResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RebuildRoleScopeForAppRoleResponse = RebuildRoleScopeForAppRoleResponse;
class RegisterCustomAppRoleHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RegisterCustomAppRoleHeaders = RegisterCustomAppRoleHeaders;
class RegisterCustomAppRoleRequest extends $tea.Model {
    static names() {
        return {
            canManageRole: 'canManageRole',
            opUserId: 'opUserId',
            roleName: 'roleName',
        };
    }
    static types() {
        return {
            canManageRole: 'boolean',
            opUserId: 'string',
            roleName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RegisterCustomAppRoleRequest = RegisterCustomAppRoleRequest;
class RegisterCustomAppRoleResponseBody extends $tea.Model {
    static names() {
        return {
            roleId: 'roleId',
            scopeVersion: 'scopeVersion',
        };
    }
    static types() {
        return {
            roleId: 'number',
            scopeVersion: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RegisterCustomAppRoleResponseBody = RegisterCustomAppRoleResponseBody;
class RegisterCustomAppRoleResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RegisterCustomAppRoleResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RegisterCustomAppRoleResponse = RegisterCustomAppRoleResponse;
class RemoveApaasAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveApaasAppHeaders = RemoveApaasAppHeaders;
class RemoveApaasAppRequest extends $tea.Model {
    static names() {
        return {
            bizAppId: 'bizAppId',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            bizAppId: 'string',
            opUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveApaasAppRequest = RemoveApaasAppRequest;
class RemoveApaasAppResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveApaasAppResponseBody = RemoveApaasAppResponseBody;
class RemoveApaasAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RemoveApaasAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveApaasAppResponse = RemoveApaasAppResponse;
class RemoveMemberForAppRoleHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveMemberForAppRoleHeaders = RemoveMemberForAppRoleHeaders;
class RemoveMemberForAppRoleRequest extends $tea.Model {
    static names() {
        return {
            deptIdList: 'deptIdList',
            opUserId: 'opUserId',
            scopeVersion: 'scopeVersion',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            deptIdList: { 'type': 'array', 'itemType': 'number' },
            opUserId: 'string',
            scopeVersion: 'number',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveMemberForAppRoleRequest = RemoveMemberForAppRoleRequest;
class RemoveMemberForAppRoleResponseBody extends $tea.Model {
    static names() {
        return {
            latestScopeVersion: 'latestScopeVersion',
        };
    }
    static types() {
        return {
            latestScopeVersion: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveMemberForAppRoleResponseBody = RemoveMemberForAppRoleResponseBody;
class RemoveMemberForAppRoleResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RemoveMemberForAppRoleResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RemoveMemberForAppRoleResponse = RemoveMemberForAppRoleResponse;
class RollbackInnerAppVersionHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RollbackInnerAppVersionHeaders = RollbackInnerAppVersionHeaders;
class RollbackInnerAppVersionRequest extends $tea.Model {
    static names() {
        return {
            appVersionId: 'appVersionId',
            opUnionId: 'opUnionId',
        };
    }
    static types() {
        return {
            appVersionId: 'number',
            opUnionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RollbackInnerAppVersionRequest = RollbackInnerAppVersionRequest;
class RollbackInnerAppVersionResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RollbackInnerAppVersionResponseBody = RollbackInnerAppVersionResponseBody;
class RollbackInnerAppVersionResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RollbackInnerAppVersionResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.RollbackInnerAppVersionResponse = RollbackInnerAppVersionResponse;
class SetMicroAppScopeHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SetMicroAppScopeHeaders = SetMicroAppScopeHeaders;
class SetMicroAppScopeRequest extends $tea.Model {
    static names() {
        return {
            addDeptIds: 'addDeptIds',
            addRoleIds: 'addRoleIds',
            addUserIds: 'addUserIds',
            delDeptIds: 'delDeptIds',
            delRoleIds: 'delRoleIds',
            delUserIds: 'delUserIds',
            onlyAdminVisible: 'onlyAdminVisible',
        };
    }
    static types() {
        return {
            addDeptIds: { 'type': 'array', 'itemType': 'number' },
            addRoleIds: { 'type': 'array', 'itemType': 'number' },
            addUserIds: { 'type': 'array', 'itemType': 'string' },
            delDeptIds: { 'type': 'array', 'itemType': 'number' },
            delRoleIds: { 'type': 'array', 'itemType': 'number' },
            delUserIds: { 'type': 'array', 'itemType': 'string' },
            onlyAdminVisible: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SetMicroAppScopeRequest = SetMicroAppScopeRequest;
class SetMicroAppScopeResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SetMicroAppScopeResponseBody = SetMicroAppScopeResponseBody;
class SetMicroAppScopeResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SetMicroAppScopeResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SetMicroAppScopeResponse = SetMicroAppScopeResponse;
class UpdateApaasAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateApaasAppHeaders = UpdateApaasAppHeaders;
class UpdateApaasAppRequest extends $tea.Model {
    static names() {
        return {
            appIcon: 'appIcon',
            appName: 'appName',
            appStatus: 'appStatus',
            bizAppId: 'bizAppId',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            appIcon: 'string',
            appName: 'string',
            appStatus: 'number',
            bizAppId: 'string',
            opUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateApaasAppRequest = UpdateApaasAppRequest;
class UpdateApaasAppResponseBody extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
            bizAppId: 'bizAppId',
        };
    }
    static types() {
        return {
            agentId: 'number',
            bizAppId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateApaasAppResponseBody = UpdateApaasAppResponseBody;
class UpdateApaasAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateApaasAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateApaasAppResponse = UpdateApaasAppResponse;
class UpdateAppRoleInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateAppRoleInfoHeaders = UpdateAppRoleInfoHeaders;
class UpdateAppRoleInfoRequest extends $tea.Model {
    static names() {
        return {
            canManageRole: 'canManageRole',
            newRoleName: 'newRoleName',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            canManageRole: 'boolean',
            newRoleName: 'string',
            opUserId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateAppRoleInfoRequest = UpdateAppRoleInfoRequest;
class UpdateAppRoleInfoResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateAppRoleInfoResponseBody = UpdateAppRoleInfoResponseBody;
class UpdateAppRoleInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateAppRoleInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateAppRoleInfoResponse = UpdateAppRoleInfoResponse;
class UpdateInnerAppHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateInnerAppHeaders = UpdateInnerAppHeaders;
class UpdateInnerAppRequest extends $tea.Model {
    static names() {
        return {
            desc: 'desc',
            homepageLink: 'homepageLink',
            icon: 'icon',
            ipWhiteList: 'ipWhiteList',
            name: 'name',
            ompLink: 'ompLink',
            opUnionId: 'opUnionId',
            pcHomepageLink: 'pcHomepageLink',
        };
    }
    static types() {
        return {
            desc: 'string',
            homepageLink: 'string',
            icon: 'string',
            ipWhiteList: { 'type': 'array', 'itemType': 'string' },
            name: 'string',
            ompLink: 'string',
            opUnionId: 'string',
            pcHomepageLink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateInnerAppRequest = UpdateInnerAppRequest;
class UpdateInnerAppResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateInnerAppResponseBody = UpdateInnerAppResponseBody;
class UpdateInnerAppResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateInnerAppResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateInnerAppResponse = UpdateInnerAppResponse;
class AddAppRolesToMemberRequestRoleList extends $tea.Model {
    static names() {
        return {
            roleId: 'roleId',
            scopeVersion: 'scopeVersion',
        };
    }
    static types() {
        return {
            roleId: 'number',
            scopeVersion: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppRolesToMemberRequestRoleList = AddAppRolesToMemberRequestRoleList;
class AddAppRolesToMemberResponseBodyResult extends $tea.Model {
    static names() {
        return {
            latestScopeVersion: 'latestScopeVersion',
            roleId: 'roleId',
            subErrorCode: 'subErrorCode',
            subErrorMsg: 'subErrorMsg',
            success: 'success',
        };
    }
    static types() {
        return {
            latestScopeVersion: 'number',
            roleId: 'number',
            subErrorCode: 'string',
            subErrorMsg: 'string',
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddAppRolesToMemberResponseBodyResult = AddAppRolesToMemberResponseBodyResult;
class GetAppResourceUseInfoResponseBody extends $tea.Model {
    static names() {
        return {
            period: 'period',
            usedNum: 'usedNum',
            quotaNum: 'quotaNum',
        };
    }
    static types() {
        return {
            period: 'string',
            usedNum: 'number',
            quotaNum: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAppResourceUseInfoResponseBody = GetAppResourceUseInfoResponseBody;
class GetMicroAppScopeResponseBodyResult extends $tea.Model {
    static names() {
        return {
            deptIds: 'deptIds',
            onlyAdminVisible: 'onlyAdminVisible',
            roleIds: 'roleIds',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            deptIds: { 'type': 'array', 'itemType': 'number' },
            onlyAdminVisible: 'boolean',
            roleIds: { 'type': 'array', 'itemType': 'number' },
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMicroAppScopeResponseBodyResult = GetMicroAppScopeResponseBodyResult;
class ListAllAppResponseBodyAppList extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
            appId: 'appId',
            appStatus: 'appStatus',
            desc: 'desc',
            developType: 'developType',
            homepageLink: 'homepageLink',
            icon: 'icon',
            name: 'name',
            ompLink: 'ompLink',
            pcHomepageLink: 'pcHomepageLink',
        };
    }
    static types() {
        return {
            agentId: 'number',
            appId: 'number',
            appStatus: 'number',
            desc: 'string',
            developType: 'number',
            homepageLink: 'string',
            icon: 'string',
            name: 'string',
            ompLink: 'string',
            pcHomepageLink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAllAppResponseBodyAppList = ListAllAppResponseBodyAppList;
class ListAllInnerAppsResponseBodyAppList extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
            appId: 'appId',
            appStatus: 'appStatus',
            desc: 'desc',
            developType: 'developType',
            homepageLink: 'homepageLink',
            icon: 'icon',
            name: 'name',
            ompLink: 'ompLink',
            pcHomepageLink: 'pcHomepageLink',
        };
    }
    static types() {
        return {
            agentId: 'number',
            appId: 'number',
            appStatus: 'number',
            desc: 'string',
            developType: 'number',
            homepageLink: 'string',
            icon: 'string',
            name: 'string',
            ompLink: 'string',
            pcHomepageLink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAllInnerAppsResponseBodyAppList = ListAllInnerAppsResponseBodyAppList;
class ListAppRoleScopesResponseBodyDataList extends $tea.Model {
    static names() {
        return {
            canManageRole: 'canManageRole',
            deptIdList: 'deptIdList',
            roleId: 'roleId',
            roleName: 'roleName',
            scopeType: 'scopeType',
            scopeVersion: 'scopeVersion',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            canManageRole: 'boolean',
            deptIdList: { 'type': 'array', 'itemType': 'number' },
            roleId: 'number',
            roleName: 'string',
            scopeType: 'string',
            scopeVersion: 'number',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAppRoleScopesResponseBodyDataList = ListAppRoleScopesResponseBodyDataList;
class ListInnerAppResponseBodyAppList extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
            desc: 'desc',
            homepageLink: 'homepageLink',
            icon: 'icon',
            name: 'name',
            ompLink: 'ompLink',
            pcHomepageLink: 'pcHomepageLink',
        };
    }
    static types() {
        return {
            agentId: 'number',
            desc: 'string',
            homepageLink: 'string',
            icon: 'string',
            name: 'string',
            ompLink: 'string',
            pcHomepageLink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListInnerAppResponseBodyAppList = ListInnerAppResponseBodyAppList;
class ListInnerAppVersionResponseBodyAppVersionList extends $tea.Model {
    static names() {
        return {
            appVersion: 'appVersion',
            appVersionId: 'appVersionId',
            appVersionType: 'appVersionType',
            createTime: 'createTime',
            entranceLink: 'entranceLink',
            miniAppId: 'miniAppId',
            miniAppOnPc: 'miniAppOnPc',
            modifyTime: 'modifyTime',
        };
    }
    static types() {
        return {
            appVersion: 'string',
            appVersionId: 'number',
            appVersionType: 'number',
            createTime: 'string',
            entranceLink: 'string',
            miniAppId: 'string',
            miniAppOnPc: 'boolean',
            modifyTime: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListInnerAppVersionResponseBodyAppVersionList = ListInnerAppVersionResponseBodyAppVersionList;
class ListRoleInfoByUserResponseBodyResult extends $tea.Model {
    static names() {
        return {
            canManageRole: 'canManageRole',
            roleId: 'roleId',
            roleName: 'roleName',
        };
    }
    static types() {
        return {
            canManageRole: 'boolean',
            roleId: 'number',
            roleName: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListRoleInfoByUserResponseBodyResult = ListRoleInfoByUserResponseBodyResult;
class ListUserVilebleAppResponseBodyAppList extends $tea.Model {
    static names() {
        return {
            agentId: 'agentId',
            appId: 'appId',
            appStatus: 'appStatus',
            desc: 'desc',
            developType: 'developType',
            homepageLink: 'homepageLink',
            icon: 'icon',
            name: 'name',
            ompLink: 'ompLink',
            pcHomepageLink: 'pcHomepageLink',
        };
    }
    static types() {
        return {
            agentId: 'number',
            appId: 'number',
            appStatus: 'number',
            desc: 'string',
            developType: 'number',
            homepageLink: 'string',
            icon: 'string',
            name: 'string',
            ompLink: 'string',
            pcHomepageLink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListUserVilebleAppResponseBodyAppList = ListUserVilebleAppResponseBodyAppList;
class PageInnerAppHistoryVersionResponseBodyMiniAppVersionList extends $tea.Model {
    static names() {
        return {
            appVersion: 'appVersion',
            appVersionId: 'appVersionId',
            appVersionType: 'appVersionType',
            createTime: 'createTime',
            miniAppId: 'miniAppId',
            miniAppOnPc: 'miniAppOnPc',
            modifyTime: 'modifyTime',
        };
    }
    static types() {
        return {
            appVersion: 'string',
            appVersionId: 'number',
            appVersionType: 'number',
            createTime: 'string',
            miniAppId: 'string',
            miniAppOnPc: 'boolean',
            modifyTime: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.PageInnerAppHistoryVersionResponseBodyMiniAppVersionList = PageInnerAppHistoryVersionResponseBodyMiniAppVersionList;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._signatureAlgorithm = "v2";
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * @summary 
     *
     * @param request AddAppRolesToMemberRequest
     * @param headers AddAppRolesToMemberHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddAppRolesToMemberResponse
     */
    async addAppRolesToMemberWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.memberId)) {
            body["memberId"] = request.memberId;
        }
        if (!tea_util_1.default.isUnset(request.memberType)) {
            body["memberType"] = request.memberType;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.roleList)) {
            body["roleList"] = request.roleList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddAppRolesToMember",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/members/roles`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddAppRolesToMemberResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddAppRolesToMemberRequest
     * @return AddAppRolesToMemberResponse
     */
    async addAppRolesToMember(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddAppRolesToMemberHeaders({});
        return await this.addAppRolesToMemberWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request AddAppToWorkBenchGroupRequest
     * @param headers AddAppToWorkBenchGroupHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddAppToWorkBenchGroupResponse
     */
    async addAppToWorkBenchGroupWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.componentId)) {
            body["componentId"] = request.componentId;
        }
        if (!tea_util_1.default.isUnset(request.ecologicalCorpId)) {
            body["ecologicalCorpId"] = request.ecologicalCorpId;
        }
        if (!tea_util_1.default.isUnset(request.opUnionId)) {
            body["opUnionId"] = request.opUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddAppToWorkBenchGroup",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/addToWorkBenchGroup`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddAppToWorkBenchGroupResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddAppToWorkBenchGroupRequest
     * @return AddAppToWorkBenchGroupResponse
     */
    async addAppToWorkBenchGroup(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddAppToWorkBenchGroupHeaders({});
        return await this.addAppToWorkBenchGroupWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request AddMemberToAppRoleRequest
     * @param headers AddMemberToAppRoleHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return AddMemberToAppRoleResponse
     */
    async addMemberToAppRoleWithOptions(agentId, roleId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.deptIdList)) {
            body["deptIdList"] = request.deptIdList;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.scopeVersion)) {
            body["scopeVersion"] = request.scopeVersion;
        }
        if (!tea_util_1.default.isUnset(request.userIdList)) {
            body["userIdList"] = request.userIdList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddMemberToAppRole",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/roles/${roleId}/members`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddMemberToAppRoleResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AddMemberToAppRoleRequest
     * @return AddMemberToAppRoleResponse
     */
    async addMemberToAppRole(agentId, roleId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddMemberToAppRoleHeaders({});
        return await this.addMemberToAppRoleWithOptions(agentId, roleId, request, headers, runtime);
    }
    /**
     * @summary AnheiP
     *
     * @param headers map
     * @param runtime runtime options for this request RuntimeOptions
     * @return AnheiPResponse
     */
    async anheiPWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "AnheiP",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/anheiP`,
            method: "GET",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AnheiPResponse({}));
    }
    /**
     * @summary AnheiP
     *
     * @return AnheiPResponse
     */
    async anheiP() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.anheiPWithOptions(headers, runtime);
    }
    /**
     * @summary AnheiTest888
     *
     * @param headers map
     * @param runtime runtime options for this request RuntimeOptions
     * @return AnheiTest888Response
     */
    async anheiTest888WithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "AnheiTest888",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/anheiTest888`,
            method: "POST",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AnheiTest888Response({}));
    }
    /**
     * @summary AnheiTest888
     *
     * @return AnheiTest888Response
     */
    async anheiTest888() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.anheiTest888WithOptions(headers, runtime);
    }
    /**
     * @summary AnheiTestB
     *
     * @param headers map
     * @param runtime runtime options for this request RuntimeOptions
     * @return AnheiTestBResponse
     */
    async anheiTestBWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "AnheiTestB",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/anheiTestB`,
            method: "PUT",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AnheiTestBResponse({}));
    }
    /**
     * @summary AnheiTestB
     *
     * @return AnheiTestBResponse
     */
    async anheiTestB() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.anheiTestBWithOptions(headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers map
     * @param runtime runtime options for this request RuntimeOptions
     * @return AnheiTestNineResponse
     */
    async anheiTestNineWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "AnheiTestNine",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/anheiTestNine`,
            method: "POST",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AnheiTestNineResponse({}));
    }
    /**
     * @summary 
     *
     * @return AnheiTestNineResponse
     */
    async anheiTestNine() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.anheiTestNineWithOptions(headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request AppStatusManagerTestRequest
     * @param headers map
     * @param runtime runtime options for this request RuntimeOptions
     * @return AppStatusManagerTestResponse
     */
    async appStatusManagerTestWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.requestId)) {
            query["requestId"] = request.requestId;
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "AppStatusManagerTest",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/manager/test`,
            method: "GET",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AppStatusManagerTestResponse({}));
    }
    /**
     * @summary 
     *
     * @param request AppStatusManagerTestRequest
     * @return AppStatusManagerTestResponse
     */
    async appStatusManagerTest(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.appStatusManagerTestWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers map
     * @param runtime runtime options for this request RuntimeOptions
     * @return AyunTestResponse
     */
    async ayunTestWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "AyunTest",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/ayun/test`,
            method: "GET",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AyunTestResponse({}));
    }
    /**
     * @summary 
     *
     * @return AyunTestResponse
     */
    async ayunTest() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.ayunTestWithOptions(headers, runtime);
    }
    /**
     * @summary openAPI
     *
     * @param headers map
     * @param runtime runtime options for this request RuntimeOptions
     * @return AyunTestOnlineResponse
     */
    async ayunTestOnlineWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "AyunTestOnline",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/ayunTest`,
            method: "GET",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AyunTestOnlineResponse({}));
    }
    /**
     * @summary openAPI
     *
     * @return AyunTestOnlineResponse
     */
    async ayunTestOnline() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.ayunTestOnlineWithOptions(headers, runtime);
    }
    /**
     * @summary Apaas
     *
     * @param request CreateApaasAppRequest
     * @param headers CreateApaasAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateApaasAppResponse
     */
    async createApaasAppWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appDesc)) {
            body["appDesc"] = request.appDesc;
        }
        if (!tea_util_1.default.isUnset(request.appIcon)) {
            body["appIcon"] = request.appIcon;
        }
        if (!tea_util_1.default.isUnset(request.appName)) {
            body["appName"] = request.appName;
        }
        if (!tea_util_1.default.isUnset(request.bizAppId)) {
            body["bizAppId"] = request.bizAppId;
        }
        if (!tea_util_1.default.isUnset(request.homepageEditLink)) {
            body["homepageEditLink"] = request.homepageEditLink;
        }
        if (!tea_util_1.default.isUnset(request.homepageLink)) {
            body["homepageLink"] = request.homepageLink;
        }
        if (!tea_util_1.default.isUnset(request.isShortCut)) {
            body["isShortCut"] = request.isShortCut;
        }
        if (!tea_util_1.default.isUnset(request.ompLink)) {
            body["ompLink"] = request.ompLink;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.pcHomepageEditLink)) {
            body["pcHomepageEditLink"] = request.pcHomepageEditLink;
        }
        if (!tea_util_1.default.isUnset(request.pcHomepageLink)) {
            body["pcHomepageLink"] = request.pcHomepageLink;
        }
        if (!tea_util_1.default.isUnset(request.templateKey)) {
            body["templateKey"] = request.templateKey;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateApaasApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apaasApps`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateApaasAppResponse({}));
    }
    /**
     * @summary Apaas
     *
     * @param request CreateApaasAppRequest
     * @return CreateApaasAppResponse
     */
    async createApaasApp(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateApaasAppHeaders({});
        return await this.createApaasAppWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request CreateInnerAppRequest
     * @param headers CreateInnerAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return CreateInnerAppResponse
     */
    async createInnerAppWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.desc)) {
            body["desc"] = request.desc;
        }
        if (!tea_util_1.default.isUnset(request.developType)) {
            body["developType"] = request.developType;
        }
        if (!tea_util_1.default.isUnset(request.homepageLink)) {
            body["homepageLink"] = request.homepageLink;
        }
        if (!tea_util_1.default.isUnset(request.icon)) {
            body["icon"] = request.icon;
        }
        if (!tea_util_1.default.isUnset(request.ipWhiteList)) {
            body["ipWhiteList"] = request.ipWhiteList;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.ompLink)) {
            body["ompLink"] = request.ompLink;
        }
        if (!tea_util_1.default.isUnset(request.opUnionId)) {
            body["opUnionId"] = request.opUnionId;
        }
        if (!tea_util_1.default.isUnset(request.pcHomepageLink)) {
            body["pcHomepageLink"] = request.pcHomepageLink;
        }
        if (!tea_util_1.default.isUnset(request.scopeType)) {
            body["scopeType"] = request.scopeType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateInnerApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateInnerAppResponse({}));
    }
    /**
     * @summary 
     *
     * @param request CreateInnerAppRequest
     * @return CreateInnerAppResponse
     */
    async createInnerApp(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateInnerAppHeaders({});
        return await this.createInnerAppWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request DeleteAppRoleRequest
     * @param headers DeleteAppRoleHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeleteAppRoleResponse
     */
    async deleteAppRoleWithOptions(agentId, roleId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DeleteAppRole",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/roles/${roleId}`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteAppRoleResponse({}));
    }
    /**
     * @summary 
     *
     * @param request DeleteAppRoleRequest
     * @return DeleteAppRoleResponse
     */
    async deleteAppRole(agentId, roleId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteAppRoleHeaders({});
        return await this.deleteAppRoleWithOptions(agentId, roleId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request DeleteInnerAppRequest
     * @param headers DeleteInnerAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return DeleteInnerAppResponse
     */
    async deleteInnerAppWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.opUnionId)) {
            query["opUnionId"] = request.opUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DeleteInnerApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteInnerAppResponse({}));
    }
    /**
     * @summary 
     *
     * @param request DeleteInnerAppRequest
     * @return DeleteInnerAppResponse
     */
    async deleteInnerApp(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteInnerAppHeaders({});
        return await this.deleteInnerAppWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary Apaas
     *
     * @param headers GetApaasAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetApaasAppResponse
     */
    async getApaasAppWithOptions(bizAppId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "GetApaasApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apaasApps/${bizAppId}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetApaasAppResponse({}));
    }
    /**
     * @summary Apaas
     *
     * @return GetApaasAppResponse
     */
    async getApaasApp(bizAppId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetApaasAppHeaders({});
        return await this.getApaasAppWithOptions(bizAppId, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request GetAppResourceUseInfoRequest
     * @param headers GetAppResourceUseInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetAppResourceUseInfoResponse
     */
    async getAppResourceUseInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.benefitCode)) {
            query["benefitCode"] = request.benefitCode;
        }
        if (!tea_util_1.default.isUnset(request.endTime)) {
            query["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.periodType)) {
            query["periodType"] = request.periodType;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            query["startTime"] = request.startTime;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetAppResourceUseInfo",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/resources/useInfos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "array",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetAppResourceUseInfoResponse({}));
    }
    /**
     * @summary 
     *
     * @param request GetAppResourceUseInfoRequest
     * @return GetAppResourceUseInfoResponse
     */
    async getAppResourceUseInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetAppResourceUseInfoHeaders({});
        return await this.getAppResourceUseInfoWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers GetAppRoleScopeByRoleIdHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetAppRoleScopeByRoleIdResponse
     */
    async getAppRoleScopeByRoleIdWithOptions(agentId, roleId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "GetAppRoleScopeByRoleId",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/roles/${roleId}/scopes`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetAppRoleScopeByRoleIdResponse({}));
    }
    /**
     * @summary 
     *
     * @return GetAppRoleScopeByRoleIdResponse
     */
    async getAppRoleScopeByRoleId(agentId, roleId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetAppRoleScopeByRoleIdHeaders({});
        return await this.getAppRoleScopeByRoleIdWithOptions(agentId, roleId, headers, runtime);
    }
    /**
     * @summary H5
     *
     * @param request GetInnerAppRequest
     * @param headers GetInnerAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetInnerAppResponse
     */
    async getInnerAppWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.ecologicalCorpId)) {
            query["ecologicalCorpId"] = request.ecologicalCorpId;
        }
        if (!tea_util_1.default.isUnset(request.opUnionId)) {
            query["opUnionId"] = request.opUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetInnerApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetInnerAppResponse({}));
    }
    /**
     * @summary H5
     *
     * @param request GetInnerAppRequest
     * @return GetInnerAppResponse
     */
    async getInnerApp(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetInnerAppHeaders({});
        return await this.getInnerAppWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers GetMicroAppScopeHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetMicroAppScopeResponse
     */
    async getMicroAppScopeWithOptions(agentId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "GetMicroAppScope",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/scopes`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetMicroAppScopeResponse({}));
    }
    /**
     * @summary 
     *
     * @return GetMicroAppScopeResponse
     */
    async getMicroAppScope(agentId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetMicroAppScopeHeaders({});
        return await this.getMicroAppScopeWithOptions(agentId, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers GetMicroAppUserAccessHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetMicroAppUserAccessResponse
     */
    async getMicroAppUserAccessWithOptions(agentId, userId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "GetMicroAppUserAccess",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/users/${userId}/adminAccess`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetMicroAppUserAccessResponse({}));
    }
    /**
     * @summary 
     *
     * @return GetMicroAppUserAccessResponse
     */
    async getMicroAppUserAccess(agentId, userId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetMicroAppUserAccessHeaders({});
        return await this.getMicroAppUserAccessWithOptions(agentId, userId, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers GetUserAppDevAccessHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return GetUserAppDevAccessResponse
     */
    async getUserAppDevAccessWithOptions(userId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "GetUserAppDevAccess",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/users/${userId}/devAccesses`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetUserAppDevAccessResponse({}));
    }
    /**
     * @summary 
     *
     * @return GetUserAppDevAccessResponse
     */
    async getUserAppDevAccess(userId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUserAppDevAccessHeaders({});
        return await this.getUserAppDevAccessWithOptions(userId, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers ListAllAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListAllAppResponse
     */
    async listAllAppWithOptions(headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "ListAllApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/allApps`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListAllAppResponse({}));
    }
    /**
     * @summary 
     *
     * @return ListAllAppResponse
     */
    async listAllApp() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListAllAppHeaders({});
        return await this.listAllAppWithOptions(headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers ListAllInnerAppsHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListAllInnerAppsResponse
     */
    async listAllInnerAppsWithOptions(headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "ListAllInnerApps",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/allInnerApps`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListAllInnerAppsResponse({}));
    }
    /**
     * @summary 
     *
     * @return ListAllInnerAppsResponse
     */
    async listAllInnerApps() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListAllInnerAppsHeaders({});
        return await this.listAllInnerAppsWithOptions(headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request ListAppRoleScopesRequest
     * @param headers ListAppRoleScopesHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListAppRoleScopesResponse
     */
    async listAppRoleScopesWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.size)) {
            query["size"] = request.size;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListAppRoleScopes",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/roles`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListAppRoleScopesResponse({}));
    }
    /**
     * @summary 
     *
     * @param request ListAppRoleScopesRequest
     * @return ListAppRoleScopesResponse
     */
    async listAppRoleScopes(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListAppRoleScopesHeaders({});
        return await this.listAppRoleScopesWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary H5
     *
     * @param request ListInnerAppRequest
     * @param headers ListInnerAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListInnerAppResponse
     */
    async listInnerAppWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.ecologicalCorpId)) {
            query["ecologicalCorpId"] = request.ecologicalCorpId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListInnerApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListInnerAppResponse({}));
    }
    /**
     * @summary H5
     *
     * @param request ListInnerAppRequest
     * @return ListInnerAppResponse
     */
    async listInnerApp(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListInnerAppHeaders({});
        return await this.listInnerAppWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers ListInnerAppVersionHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListInnerAppVersionResponse
     */
    async listInnerAppVersionWithOptions(agentId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "ListInnerAppVersion",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/innerMiniApps/${agentId}/versions`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListInnerAppVersionResponse({}));
    }
    /**
     * @summary 
     *
     * @return ListInnerAppVersionResponse
     */
    async listInnerAppVersion(agentId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListInnerAppVersionHeaders({});
        return await this.listInnerAppVersionWithOptions(agentId, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers ListRoleInfoByUserHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListRoleInfoByUserResponse
     */
    async listRoleInfoByUserWithOptions(agentId, userId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "ListRoleInfoByUser",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/users/${userId}/roles`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListRoleInfoByUserResponse({}));
    }
    /**
     * @summary 
     *
     * @return ListRoleInfoByUserResponse
     */
    async listRoleInfoByUser(agentId, userId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListRoleInfoByUserHeaders({});
        return await this.listRoleInfoByUserWithOptions(agentId, userId, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param headers ListUserVilebleAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return ListUserVilebleAppResponse
     */
    async listUserVilebleAppWithOptions(userId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "ListUserVilebleApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/users/${userId}/apps`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListUserVilebleAppResponse({}));
    }
    /**
     * @summary 
     *
     * @return ListUserVilebleAppResponse
     */
    async listUserVilebleApp(userId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListUserVilebleAppHeaders({});
        return await this.listUserVilebleAppWithOptions(userId, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request PageInnerAppHistoryVersionRequest
     * @param headers PageInnerAppHistoryVersionHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return PageInnerAppHistoryVersionResponse
     */
    async pageInnerAppHistoryVersionWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            query["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            query["pageSize"] = request.pageSize;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "PageInnerAppHistoryVersion",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/innerMiniApps/${agentId}/historyVersions`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PageInnerAppHistoryVersionResponse({}));
    }
    /**
     * @summary 
     *
     * @param request PageInnerAppHistoryVersionRequest
     * @return PageInnerAppHistoryVersionResponse
     */
    async pageInnerAppHistoryVersion(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PageInnerAppHistoryVersionHeaders({});
        return await this.pageInnerAppHistoryVersionWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request PublishInnerAppVersionRequest
     * @param headers PublishInnerAppVersionHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return PublishInnerAppVersionResponse
     */
    async publishInnerAppVersionWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appVersionId)) {
            body["appVersionId"] = request.appVersionId;
        }
        if (!tea_util_1.default.isUnset(request.miniAppOnPc)) {
            body["miniAppOnPc"] = request.miniAppOnPc;
        }
        if (!tea_util_1.default.isUnset(request.opUnionId)) {
            body["opUnionId"] = request.opUnionId;
        }
        if (!tea_util_1.default.isUnset(request.publishType)) {
            body["publishType"] = request.publishType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PublishInnerAppVersion",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/innerMiniApps/${agentId}/versions/publish`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PublishInnerAppVersionResponse({}));
    }
    /**
     * @summary 
     *
     * @param request PublishInnerAppVersionRequest
     * @return PublishInnerAppVersionResponse
     */
    async publishInnerAppVersion(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PublishInnerAppVersionHeaders({});
        return await this.publishInnerAppVersionWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request RebuildRoleScopeForAppRoleRequest
     * @param headers RebuildRoleScopeForAppRoleHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RebuildRoleScopeForAppRoleResponse
     */
    async rebuildRoleScopeForAppRoleWithOptions(agentId, roleId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.deptIdList)) {
            body["deptIdList"] = request.deptIdList;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.scopeType)) {
            body["scopeType"] = request.scopeType;
        }
        if (!tea_util_1.default.isUnset(request.scopeVersion)) {
            body["scopeVersion"] = request.scopeVersion;
        }
        if (!tea_util_1.default.isUnset(request.userIdList)) {
            body["userIdList"] = request.userIdList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "RebuildRoleScopeForAppRole",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/roles/${roleId}/scopes/rebuild`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RebuildRoleScopeForAppRoleResponse({}));
    }
    /**
     * @summary 
     *
     * @param request RebuildRoleScopeForAppRoleRequest
     * @return RebuildRoleScopeForAppRoleResponse
     */
    async rebuildRoleScopeForAppRole(agentId, roleId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RebuildRoleScopeForAppRoleHeaders({});
        return await this.rebuildRoleScopeForAppRoleWithOptions(agentId, roleId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request RegisterCustomAppRoleRequest
     * @param headers RegisterCustomAppRoleHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RegisterCustomAppRoleResponse
     */
    async registerCustomAppRoleWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.canManageRole)) {
            body["canManageRole"] = request.canManageRole;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.roleName)) {
            body["roleName"] = request.roleName;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "RegisterCustomAppRole",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/roles`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RegisterCustomAppRoleResponse({}));
    }
    /**
     * @summary 
     *
     * @param request RegisterCustomAppRoleRequest
     * @return RegisterCustomAppRoleResponse
     */
    async registerCustomAppRole(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RegisterCustomAppRoleHeaders({});
        return await this.registerCustomAppRoleWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary apaas
     *
     * @param request RemoveApaasAppRequest
     * @param headers RemoveApaasAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RemoveApaasAppResponse
     */
    async removeApaasAppWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizAppId)) {
            body["bizAppId"] = request.bizAppId;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "RemoveApaasApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apaasApps/remove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RemoveApaasAppResponse({}));
    }
    /**
     * @summary apaas
     *
     * @param request RemoveApaasAppRequest
     * @return RemoveApaasAppResponse
     */
    async removeApaasApp(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RemoveApaasAppHeaders({});
        return await this.removeApaasAppWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request RemoveMemberForAppRoleRequest
     * @param headers RemoveMemberForAppRoleHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RemoveMemberForAppRoleResponse
     */
    async removeMemberForAppRoleWithOptions(agentId, roleId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.deptIdList)) {
            body["deptIdList"] = request.deptIdList;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.scopeVersion)) {
            body["scopeVersion"] = request.scopeVersion;
        }
        if (!tea_util_1.default.isUnset(request.userIdList)) {
            body["userIdList"] = request.userIdList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "RemoveMemberForAppRole",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/roles/${roleId}/members/batchRemove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RemoveMemberForAppRoleResponse({}));
    }
    /**
     * @summary 
     *
     * @param request RemoveMemberForAppRoleRequest
     * @return RemoveMemberForAppRoleResponse
     */
    async removeMemberForAppRole(agentId, roleId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RemoveMemberForAppRoleHeaders({});
        return await this.removeMemberForAppRoleWithOptions(agentId, roleId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request RollbackInnerAppVersionRequest
     * @param headers RollbackInnerAppVersionHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return RollbackInnerAppVersionResponse
     */
    async rollbackInnerAppVersionWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appVersionId)) {
            body["appVersionId"] = request.appVersionId;
        }
        if (!tea_util_1.default.isUnset(request.opUnionId)) {
            body["opUnionId"] = request.opUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "RollbackInnerAppVersion",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/innerMiniApps/${agentId}/versions/rollback`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RollbackInnerAppVersionResponse({}));
    }
    /**
     * @summary 
     *
     * @param request RollbackInnerAppVersionRequest
     * @return RollbackInnerAppVersionResponse
     */
    async rollbackInnerAppVersion(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RollbackInnerAppVersionHeaders({});
        return await this.rollbackInnerAppVersionWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request SetMicroAppScopeRequest
     * @param headers SetMicroAppScopeHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return SetMicroAppScopeResponse
     */
    async setMicroAppScopeWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.addDeptIds)) {
            body["addDeptIds"] = request.addDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.addRoleIds)) {
            body["addRoleIds"] = request.addRoleIds;
        }
        if (!tea_util_1.default.isUnset(request.addUserIds)) {
            body["addUserIds"] = request.addUserIds;
        }
        if (!tea_util_1.default.isUnset(request.delDeptIds)) {
            body["delDeptIds"] = request.delDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.delRoleIds)) {
            body["delRoleIds"] = request.delRoleIds;
        }
        if (!tea_util_1.default.isUnset(request.delUserIds)) {
            body["delUserIds"] = request.delUserIds;
        }
        if (!tea_util_1.default.isUnset(request.onlyAdminVisible)) {
            body["onlyAdminVisible"] = request.onlyAdminVisible;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SetMicroAppScope",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/scopes`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SetMicroAppScopeResponse({}));
    }
    /**
     * @summary 
     *
     * @param request SetMicroAppScopeRequest
     * @return SetMicroAppScopeResponse
     */
    async setMicroAppScope(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SetMicroAppScopeHeaders({});
        return await this.setMicroAppScopeWithOptions(agentId, request, headers, runtime);
    }
    /**
     * @summary apaas
     *
     * @param request UpdateApaasAppRequest
     * @param headers UpdateApaasAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateApaasAppResponse
     */
    async updateApaasAppWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appIcon)) {
            body["appIcon"] = request.appIcon;
        }
        if (!tea_util_1.default.isUnset(request.appName)) {
            body["appName"] = request.appName;
        }
        if (!tea_util_1.default.isUnset(request.appStatus)) {
            body["appStatus"] = request.appStatus;
        }
        if (!tea_util_1.default.isUnset(request.bizAppId)) {
            body["bizAppId"] = request.bizAppId;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateApaasApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apaasApps`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateApaasAppResponse({}));
    }
    /**
     * @summary apaas
     *
     * @param request UpdateApaasAppRequest
     * @return UpdateApaasAppResponse
     */
    async updateApaasApp(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateApaasAppHeaders({});
        return await this.updateApaasAppWithOptions(request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateAppRoleInfoRequest
     * @param headers UpdateAppRoleInfoHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateAppRoleInfoResponse
     */
    async updateAppRoleInfoWithOptions(agentId, roleId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.canManageRole)) {
            body["canManageRole"] = request.canManageRole;
        }
        if (!tea_util_1.default.isUnset(request.newRoleName)) {
            body["newRoleName"] = request.newRoleName;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateAppRoleInfo",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}/roles/${roleId}`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateAppRoleInfoResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateAppRoleInfoRequest
     * @return UpdateAppRoleInfoResponse
     */
    async updateAppRoleInfo(agentId, roleId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateAppRoleInfoHeaders({});
        return await this.updateAppRoleInfoWithOptions(agentId, roleId, request, headers, runtime);
    }
    /**
     * @summary 
     *
     * @param request UpdateInnerAppRequest
     * @param headers UpdateInnerAppHeaders
     * @param runtime runtime options for this request RuntimeOptions
     * @return UpdateInnerAppResponse
     */
    async updateInnerAppWithOptions(agentId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.desc)) {
            body["desc"] = request.desc;
        }
        if (!tea_util_1.default.isUnset(request.homepageLink)) {
            body["homepageLink"] = request.homepageLink;
        }
        if (!tea_util_1.default.isUnset(request.icon)) {
            body["icon"] = request.icon;
        }
        if (!tea_util_1.default.isUnset(request.ipWhiteList)) {
            body["ipWhiteList"] = request.ipWhiteList;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.ompLink)) {
            body["ompLink"] = request.ompLink;
        }
        if (!tea_util_1.default.isUnset(request.opUnionId)) {
            body["opUnionId"] = request.opUnionId;
        }
        if (!tea_util_1.default.isUnset(request.pcHomepageLink)) {
            body["pcHomepageLink"] = request.pcHomepageLink;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateInnerApp",
            version: "microApp_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/microApp/apps/${agentId}`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateInnerAppResponse({}));
    }
    /**
     * @summary 
     *
     * @param request UpdateInnerAppRequest
     * @return UpdateInnerAppResponse
     */
    async updateInnerApp(agentId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateInnerAppHeaders({});
        return await this.updateInnerAppWithOptions(agentId, request, headers, runtime);
    }
}
exports.default = Client;
